---
title: "Comparación de la respuesta de anticuerpos ante _Plasmodium vivax_ mediante Microarreglos de proteínas"
author: "Andree Valle Campos"
date: '`r Sys.Date()`'
output:
  #html_document:
  #pdf_document:
  html_notebook:
    toc: yes
    toc_depth: 6
    toc_float:
      collapsed: yes
    #theme: united
    code_folding: "hide"
    #fig_caption: TRUE
    #number_sections: TRUE
bibliography: malaria.bib
link-citations: yes
#csl: american-medical-association.csl
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, #fig.path = "01-",
                      warning = FALSE)
# knitr::opts_knit$set(root.dir = '../.')
options(width = 90) # expand limits of CONSOLE output
```

## Microarray Data Analysis

+ __UPDATE:__ 
    - Ir al contraste de los __resultados__ con la __lista de Ag de N6__. [Ver aquí.](#n6-subset)

### Dependencies

This document has the following dependencies:
```{r, warning=FALSE, message=FALSE}
library("tidyverse")
library("haven")
library("broom")
library("biobroom")
library("ggrepel")
library("forcats")
library("stringr")

library("Rmisc")       #multiploting ggplots

library("Biobase")     #ExpressionSet
library("genefilter")  #DataCondensation
library("limma")       #DifferentialAnalysisOfGenes
library("NMF")         #AnnotatedHeatmaps!!!

#library(PerformanceAnalytics) #GRAPHICAL CORRELATIONS
#library(beeswarm)
#library(vioplot)
#library(beanplot)
#library(htmlTable)  #HTMLtables -minimalistic- without 'ResultsAsis'
#library(knitr)        #better it seems

library(patchwork)

theme_set(theme_bw())
```


### Input

```{r}
raw<-read.csv("data-raw/RawData.csv")
```

```{r, eval=FALSE}
raw %>% View()
raw %>% dplyr::count(Spot.Type)
raw %>% dim()
raw %>% glimpse()
raw %>% 
  as_tibble() %>% 
  select(-(Index:Description)) %>% 
  colnames() %>% enframe() %>% 
  mutate(start=
           case_when(
             str_starts(value,"L")~str_replace(value,"(...).+", "\\1"),
             str_starts(value,"P")~str_replace(value,"(....).+", "\\1"), 
             TRUE ~ "non")) %>% 
  dplyr::count(start)
# samples si tiene ubicacion de los 8 controles
# las lecturas de Ab no estan en el archivo de valores crudos
# pero si estan en los archivo excel
readr::read_csv("data-raw/samples.csv") %>% 
  filter(Study=="Controls") %>% 
  select(-Study)
readr::read_csv("data-raw/samples.csv") %>% 
  # rownames_to_column() %>% 
  # as_tibble() %>% 
  # dplyr::count(Study)
  # dplyr::count(Study,Group)
  select(value=Sample.ID) %>% 
  mutate(start=
           case_when(
             str_starts(value,"L")~str_replace(value,"(...).+", "\\1"),
             str_starts(value,"PQ")~str_replace(value,"(....).+", "\\1"), 
             TRUE ~ "non")) %>% 
  dplyr::count(start)
```

### 1. Tidy up

#### ivtt antigens

```{r}
anti <- raw %>% 
  dplyr::filter(Spot.Type=="IVTT.AG") %>% #head()
  select(ID,11:ncol(.)) %>% #head()
  gather(sample_name, expression, -ID) %>% 
  mutate(expression=as.numeric(expression)) %>% 
  reshape2::acast(ID ~ sample_name,
                  value.var = "expression") #%>% class()

# dimensión
dim(anti)
# seis lecturas por proteína de las 8 primeras muestras
head(anti[,1:8])
```

#### ivtt controls

```{r}
ctrl <- raw %>% 
  dplyr::filter(Spot.Type=="IVTT.CTRL") %>% #head()
  select(ID,11:ncol(.)) %>% #head()
  summarise_if(is.numeric,funs(median)) %>% # MEDIAN to NORMALIZE against noDNA controls
  mutate(ID="noDNA") %>% 
  gather(sample_name, expression, -ID) %>% 
  mutate(expression=as.numeric(expression)) %>% 
  reshape2::acast(ID ~ sample_name,
                  value.var = "expression") #%>% class()

# dimensión
dim(ctrl)
# seis lecturas por proteína de las 8 primeras muestras
ctrl[,1:8]
```

```{r}
ctrm <- NULL # crear objeto vacío
# loop para generar una matriz de las dimensiones de `anti`
for(i in 1:nrow(anti)){
  ctrm <- rbind(ctrm,ctrl)
}
# dimensiones de la matriz con la mediana de ctrls `noDNA` creada
dim(ctrm)
# matriz con la mediana de ctrls `noDNA` para las 8 primeras muestras
head(ctrm[,1:8]) #%>% class()
```

#### purified proteins

```{r}
pure <- raw %>% 
  dplyr::filter(Spot.Type=="PURIFIED.PROTEIN") %>% #dim() #dplyr::count(Species)
  #separate(Description,c("gene", "conc"), sep=",",remove = FALSE) %>% #dplyr::count(gene)
  #select(ID,Species,gene,14:ncol(.)) %>% 
  select(ID,11:ncol(.)) %>% 
  gather(sample_name, expression, -ID) %>% 
  mutate(expression=as.numeric(expression)) %>% 
  reshape2::acast(ID ~ sample_name,
                  value.var = "expression") #%>% class()
```

#### gene names

```{r}
# generate a list (df) of gene names and gene ID's
ln <- raw %>% 
  dplyr::filter(Spot.Type=="IVTT.AG") %>% #head()
  select(ID,Gene.ID,Description,Species)

lg <- ln %>% #dplyr::count(Gene.ID)
  group_by(Gene.ID,Species) %>% dplyr::count() %>% #filter(Species=="Pv") %>% arrange(desc(n))
  ungroup()

readr::write_csv(lg %>% select(Gene.ID), "data/04-listgen.csv")
```

#### feature data

```{r}
feat <- raw %>% 
  dplyr::filter(Spot.Type=="IVTT.AG") %>%
  
  #DO NOT WORKS!
  #dplyr::mutate(ID=as.character(ID)) %>% 
  #dplyr::mutate(ID= stringr::str_replace(ID, "(.....)_(\\d{6})_(.+)","\\3"))
  #separate(ID,c("id.sp","id.cod","id.res"),sep = "_",remove = FALSE) %>% 
  #separate(id.cod,c("id.cod","id.num")) %>% 
  #unite(id.num,id.num,id.res)
  
  #THIS WORKS!
  #dplyr::mutate(num=seq(from=33, to=32+n())) %>% #dplyr::count(num) %>% dplyr::count(n)
  #unite(Description,Description,num,sep=" _") %>%
  #dplyr::mutate(Description=as.factor(Description)) %>% 
  
  #select(ID,1:10) %>% #head()
  select(ID,Gene.ID,Description,Species) %>% #head()
  as.data.frame() %>% 
  column_to_rownames(var="ID")#%>% class()

feat_d <- new("AnnotatedDataFrame", data=feat)
```

#### phenotype data

- __join__ `sample.csv` with `03-sevrcov.rds` and `primaquine` sample data
- __create__ categorical variables from numerical: `edad` and `episodio_previo_num`
    + `cut()`advantage: automatic factor labels.

```{r}
sevdb_3 <- readRDS("data/03-sevrcov.rds") %>% #dplyr::count(episodio_previo)
  dplyr::rename(Sample.ID="codigo") %>% 
  mutate(edad_CAT=cut(edad,c(0,18,40,65,Inf)),
         expo_CAT=cut(episodio_previo_num,c(-Inf,0,1,4,Inf))) %>%  #%>%
  select(Sample.ID,sev_WHO_num,episodio_previo_num,#en_zona_endemica,#edad,#,parasitemia
         sev_WHO:expo_CAT,-Study#,-episodio_previo_num
         )
```

```{r, message=FALSE}
pridb <- #readr::read_csv("analysis/more/ADi-NAMRU6_Data/samples.csv") %>% ## BUSCAR ORIGINAL!!!
  readr::read_csv("data-raw/ADi-NAMRU6_Data-samples.csv") %>% ## ENCONTRÉ ORIGINAL
  filter(Study=="Primaquine") %>% 
  select(Sample.ID,Cat.Age.WHO:Anemia.HTO)
```

```{r, message=FALSE}
pheno <- readr::read_csv("data-raw/samples.csv") %>% #dplyr::count(Sample.Type)
  dplyr::filter(Sample.Type!="Control") %>% #str()
  select(Sample.ID, 1:9, -Sample.Type,-Subject.ID) %>% #dplyr::count(Filename)
  dplyr::arrange(Sample.ID) %>%
  full_join(sevdb_3,by = "Sample.ID") %>% 
  full_join(pridb,by = "Sample.ID") %>% 
  dplyr::arrange(Sample.ID) %>% 
  # # manual imputation of one observation # warning! (post-hoc decision)
  # # filter(!is.na(sev_WHO) & is.na(episodio_previo)) %>% glimpse()
  # mutate(episodio_previo=as.character(episodio_previo)) %>% 
  # mutate(episodio_previo=case_when(
  #   Sample.ID=="LIM2017"~"sin",
  #   TRUE~episodio_previo)) %>% 
  # mutate(episodio_previo=as.factor(episodio_previo)) %>% 
  # # filter(Sample.ID=="LIM2017") %>% glimpse()
  as.data.frame() %>% 
  column_to_rownames(var="Sample.ID")
  

pheno_d <- new("AnnotatedDataFrame", data=pheno)

#colnames(norx)
```

### 2. Normalization + Transformation

```{r log-norm}
# custom function
# definir resultado para los valores menores o iguales a cero:
log2.NA = function(x) {log2(ifelse(x>0, x, NA))}

# normalización con respecto a los controles `noDNA` c/ transformación log2
norm <- log2.NA(anti/ctrm)
head(norm[,1:6])
```

```{r}
test <- anti/ctrm 
```

```{r ratio-test, eval=FALSE, echo=FALSE}
test <- anti/ctrm 

test %>% as.data.frame() %>% 
  rownames_to_column() %>% 
  gather(sample_name,expression,-rowname) %>% 
  filter(expression <= 0)

anti %>% as.data.frame() %>% 
  rownames_to_column() %>% 
  gather(sample_name,expression,-rowname) %>% 
  dplyr::filter(rowname=="PVX_113590_2o2.849",sample_name=="LIM2077") # 21

norm %>% as.data.frame() %>% 
  rownames_to_column() %>% 
  gather(sample_name,expression,-rowname) %>% 
  dplyr::filter(rowname=="PVX_113590_2o2.849",sample_name=="LIM2077") # 21
```

```{r vsn-norm}

```

#### outlier

```{r}

ndb <- norm %>% as.data.frame() %>% 
  gather(sample_name,expression)

p <- ndb %>% 
  full_join(ndb %>% 
              group_by(sample_name) %>% 
              dplyr::summarise(avg=mean(expression)) %>% 
              ungroup(), 
            by="sample_name") %>% 
  dplyr::arrange(avg) %>% 
  ggplot(aes(reorder(sample_name,avg,order = TRUE),expression)) +
  geom_boxplot() +
  theme(axis.text.x = element_blank()) +
  labs(title="Outlier visualization")
```

```{r}
# EDIT specific value under CONDITIONAL STATEMENT: 
# source: https://github.com/tidyverse/dplyr/issues/425
norx <- norm %>% as.data.frame() %>% 
  rownames_to_column() %>% 
  gather(sample_name,expression,-rowname) %>%
  #dplyr::filter(expression < -5) #%>% dim()
  mutate(expression= ifelse(expression < -5, NA, expression)) %>% # RULE TO EDIT a value of a column!
  #dplyr::filter(rowname=="PVX_111175.544",sample_name=="LIM2017") # 21
  reshape2::acast(rowname ~ sample_name,
                  value.var = "expression") #%>% class()
 
#norm["PVX_111175.544","LIM2017"] <- NA ## messy modification

#anti %>% as.data.frame() %>% 
#  rownames_to_column() %>% 
#  gather(sample_name,expression,-rowname) %>% 
#  dplyr::filter(rowname=="PVX_111175.544",sample_name=="LIM2017") # 21
#
#ctrl %>% as.data.frame() %>% 
#  rownames_to_column() %>% 
#  gather(sample_name,expression,-rowname) %>% 
#  dplyr::filter(sample_name=="LIM2017") # 13550.5
```

```{r, fig.height=4, fig.width=15}
q <- norx %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  full_join(ndb %>% 
              group_by(sample_name) %>% 
              dplyr::summarise(avg=mean(expression)) %>% 
              ungroup(), 
            by="sample_name") %>% 
  dplyr::arrange(avg) %>% 
  ggplot(aes(reorder(sample_name,avg,order = TRUE),expression)) +
  geom_boxplot() +
  theme(axis.text.x = element_blank()) +
  labs(title="Conditional edition of a single element")

Rmisc::multiplot(p,q,cols = 2)
```

### 3. ExpressionSet

- __arrange__ `norx` rownames wrt `feat_d` ones.
- __create__ the expression set

```{r}
head(norx[,1:5])
head(norx[feat_d %>% rownames(),1:5])
```

```{r}
eset <- ExpressionSet(assayData = norx[feat_d %>% rownames(),], 
                      phenoData = pheno_d,
                      featureData = feat_d)
eset
```

#### subset eset 
**per Pv/Pf chip and Pq/Sev experiment**
```{r}
#eset
# ExpressionSet subseted by both SAMPLES and FEATURE covariates!!!!
##
eset.VIVAX.PQ<-eset[featureData(eset)$Species=="Pv",
                    phenoData(eset)$Study=="Primaquine"]
#eset.VIVAX.PQ
#
eset.FALCIP.PQ<-eset[featureData(eset)$Species=="Pf",
                     phenoData(eset)$Study=="Primaquine"]
#eset.FALCIP.PQ
##
eset.VIVAX.SEV<-eset[featureData(eset)$Species=="Pv",
                     phenoData(eset)$Study!="Primaquine"]
eset.VIVAX.SEV
#
eset.FALCIP.SEV<-eset[featureData(eset)$Species=="Pf",
                      phenoData(eset)$Study!="Primaquine"]
#eset.FALCIP.SEV
##
```

```{r, eval=FALSE}
summary(pData(eset))
varMetadata(eset)
table(pData(eset)$Study)
table(pData(eset)$Group)
```

#### specific pData

```{r}
pData(eset.VIVAX.SEV) <- pData(eset.VIVAX.SEV) %>% 
  select(Filename:expo_CAT,-Filename,-Slide,-Pad,-Probing.Day)

eset.VIVAX.SEV
```

```{r}
pData(eset.VIVAX.PQ) <- pData(eset.VIVAX.PQ) %>% 
  select(Study,Group,Weight:Anemia.HTO)

#eset.VIVAX.PQ
```

### 4. Filtering

```{r, echo=TRUE}
#
#eset             # 1014 features X 200 samples   (10%= 20)
#
#eset.VIVAX.PQ    # 515 features X 140 samples    (10%= 14)
#eset.FALCIP.PQ   # 499 features X 140 samples    (10%= 14)
#eset.VIVAX.SEV   # 515 features X 60 samples     (10%= 6)
#eset.FALCIP.SEV  # 499 features X 60 samples     (10%= 6)
#
############################## [START] GENEFILTER for each DATASET (n=4)
f1 <- kOverA(20,1)                           ## expression measure above 1 in at least 20 samples
ffun <- filterfun(f1)
wh1 <- genefilter(exprs(eset),ffun)  # WHOLE data set
#sum(wh1) # 526 -> 818 ---------------------------------------------------->>>> (ALL of this are WITHOUT mean centering)
#
# GENERATE the NEW ExpressionSet
eset.FILTER<-eset[wh1,]
#eset.FILTER

############################## GENEFILTER for VIVAX PRIMAQUINE
f2 <- kOverA(14,1)                           ## expression measure above 1 in at least 14 samples
ffun2 <- filterfun(f2)
wh2 <- genefilter(exprs(eset.VIVAX.PQ),ffun2)  # WHOLE data set
#sum(wh2) # 252 -> 397
#
# GENERATE the NEW ExpressionSet
eset.VIVAX.PQ.FILTER<-eset.VIVAX.PQ[wh2,]
#eset.VIVAX.PQ.FILTER

############################## GENEFILTER for FALCIPARUM PRIMAQUINE
f2 <- kOverA(14,1)                           ## expression measure above 1 in at least 14 samples
ffun2 <- filterfun(f2)
wh3 <- genefilter(exprs(eset.FALCIP.PQ),ffun2)  # WHOLE data set
#sum(wh3) # 279 -> 430
#
# GENERATE the NEW ExpressionSet
eset.FALCIP.PQ.FILTER<-eset.FALCIP.PQ[wh3,]
#eset.FALCIP.PQ.FILTER

############################## [DONE] GENEFILTER for VIVAX SEVERE
f3 <- kOverA(6,1)                           ## expression measure above 1 in at least 6 samples
ffun3 <- filterfun(f3)
wh4 <- genefilter(exprs(eset.VIVAX.SEV),ffun3)  # WHOLE data set
#sum(wh4) # 255 -> 394
#
# GENERATE the NEW ExpressionSet
eset.VIVAX.SEV.FILTER<-eset.VIVAX.SEV[wh4,]
eset.VIVAX.SEV.FILTER

############################## [DONE] GENEFILTER for FALCIPARUM SEVERE
f3 <- kOverA(6,1)                           ## expression measure above 1 in at least 6 samples
ffun3 <- filterfun(f3)
wh5 <- genefilter(exprs(eset.FALCIP.SEV),ffun3)  # WHOLE data set
#sum(wh5) # 280 -> 419
#
# GENERATE the NEW ExpressionSet
eset.FALCIP.SEV.FILTER<-eset.FALCIP.SEV[wh5,]
#eset.FALCIP.SEV.FILTER

############################## [END] OF GENEFILTER 
#
# results
#
#eset.FILTER

#eset.VIVAX.PQ.FILTER      # 252 /515 features X 140 samples
#eset.FALCIP.PQ.FILTER     # 279 /499 features X 140 samples
#eset.VIVAX.SEV.FILTER     # 255 /515 features X 60 samples
#eset.FALCIP.SEV.FILTER    # 280 /499 features X 60 samples
#
##########################################################

```

### 0. Descriptive Statistics

#### Preprocessing

##### Distributions per step

raw -> normalized -> transformed -> filtered

```{r, fig.width=12, fig.height=3.5}
a <- anti %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  dplyr::mutate(labl="Raw")
b <- test %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  filter(expression >= 0) %>% 
  dplyr::mutate(labl="Normalized")
c <- norx %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  dplyr::mutate(labl="Transformed")
d <- exprs(eset.VIVAX.SEV.FILTER) %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  dplyr::mutate(labl="Filtered*")

rbind(a,b,c,d) %>% 
  dplyr::mutate(labl=forcats::fct_relevel(labl,
                                          "Raw",
                                          "Normalized",
                                          "Transformed",
                                          "Filtered*")) %>% 
  ggplot(aes(expression)) +
  geom_histogram() + theme_bw() +
  #facet_grid(.~labl,scales = "free") +
  facet_wrap(~labl,nrow = 1,ncol = 4,scales = "free") +
  labs(title="Preprocessing: Data distribution per step",
       caption="*Samples by Severe Malaria Study\nand Probes by P. vivax antigens") +
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black",size = 12#, face = "bold"
                                    ))
```

```{r, fig.width=16, fig.height=4,eval=FALSE,echo=FALSE}
a <- anti %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  ggplot(aes(expression)) +
  geom_histogram() +
  labs(title="RAW") + theme_bw()
b <- test %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  filter(expression >= 0) %>% 
  ggplot(aes(expression)) +
  geom_histogram() +
  labs(title="NORMALIZED") + theme_bw()
c <- norx %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  ggplot(aes(expression)) +
  geom_histogram() +
  labs(title="TRANSFORMED") + theme_bw()
d <- exprs(eset.VIVAX.SEV.FILTER) %>% as.data.frame() %>% 
  gather(sample_name,expression) %>% 
  ggplot(aes(expression)) +
  geom_histogram() +
  labs(title="SEVERE VIVAX FILTERED DATA") + theme_bw()

Rmisc::multiplot(a,b,c,d,cols = 4)
```

*Of sample covariates, feature covariates and microarray whole dataset*

#### Sample covariates

- just show the results from `03-sevrcov.Rmd`
- compare against initial stratification and after update!

##### Reclassification

- Smith-Nuñez, correo __28oct2016__:
    + La base de datos epidemiológicos inicial pasó por __control de calidad y reingreso de fichas__.
    + __Acción:__
        - Contrastar y Crear una DB consenso,
        - Redefinir clasificación OMS de Malaria Severa,
        - Reclasificar muestras,
        - Filtrar muestras seleccionadas para el ensayo de Microarreglo de Proteínas.

```{r}
pData(eset) %>% 
  group_by(Group,sev_WHO) %>% dplyr::count()
```

```{r}
pData(eset) %>% 
  group_by(sev_WHO, episodio_previo) %>% dplyr::count()
```

#### Feature covariates

- maybe a graph of the slides with all the coordinates? :)

##### Load PlasmoDB Metadata

- __Write `data/04-listgen.csv`__ with all the gene names. [here](#gene-names)
- __Create PlasmoDB strategy__, select required features, and download data.
- __Tidy PlasmoDB data frame__ output:
    + __Programaticaly renamed all__ variable/column names using __regex__.

```{r, message=FALSE}
all <- readr::read_tsv("data/04-listgen.tsv") %>% #colnames() %>% #class()
  #make.names(unique = TRUE) %>% 
  rename_all(
      funs(
        #stringr::str_to_lower(.) %>%
        stringr::str_replace_all(., '\\s', '\\.') %>% 
        stringr::str_replace_all(., '\\[|\\]', '') %>% 
        stringr::str_replace_all(., '\\#.', '') %>% 
        stringr::str_replace_all(., 'Computed.', '') %>% 
        stringr::str_replace_all(., '\\/', '\\_')
      )
  ) %>% 
  select(-source_id,-X18) %>% 
  dplyr::rename(Gene.Name=Gene.Name.or.Symbol) %>% 
  dplyr::mutate_all(
    funs(
      stringr::str_replace_all(.,"N/A",replacement = NA_character_)
    )) %>% 
  dplyr::mutate(SignalP.Scores=stringr::str_replace(SignalP.Scores, "^NN.{0,}","SP")) %>% 
  dplyr::mutate(NonSyn_Syn.SNP.Ratio.All.Strains=as.numeric(NonSyn_Syn.SNP.Ratio.All.Strains),
                Total.SNPs.All.Strains=as.numeric(Total.SNPs.All.Strains),
                Transcript.Length=as.numeric(Transcript.Length),
                Ortholog.count=as.numeric(Ortholog.count)) %>% 
  #dplyr::mutate(SignalP=stringr::str_replace(SignalP.Scores, "^NN.{0,}","SP")) %>% 
  #group_by(SignalP.Scores,SignalP) %>% dplyr::count()
  dplyr::rename(SignalP=SignalP.Scores) %>% 
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_003775", "MSP4", Gene.Name)) %>%
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_003770", "MSP5", Gene.Name)) %>% 
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_097625", "MSP8", Gene.Name)) %>% 
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_114145", "MSP10", Gene.Name)) %>% 
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_116780", "SFT2", Gene.Name))
#lg
head(all)
#raw %>% filter(Gene.ID=="PVX_003775")#only_2o2 --> IS THIS SUGGESTING A MISTAKE IN ID TIPYING? --> COMPARE!
#raw %>% filter(stringr::str_detect(Description, "MSP7"))
```

```{r, message=FALSE}
full <- readr::read_tsv("data/04-listall.tsv") %>% #colnames() %>% #class()
  #make.names(unique = TRUE) %>% 
  rename_all(
      funs(
        #stringr::str_to_lower(.) %>%
        stringr::str_replace_all(., '\\s', '\\.') %>% 
        stringr::str_replace_all(., '\\[|\\]', '') %>% 
        stringr::str_replace_all(., '\\#.', '') %>% 
        stringr::str_replace_all(., 'Computed.', '') %>% 
        stringr::str_replace_all(., '\\/', '\\_')
      )
  ) %>% 
  select(-source_id,-X8) %>% 
  dplyr::rename(Gene.Name=Gene.Name.or.Symbol) %>% 
  dplyr::mutate_all(
    funs(
      stringr::str_replace_all(.,"N/A",replacement = NA_character_)
    )) #%>% 
  #dplyr::mutate(SignalP.Scores=stringr::str_replace(SignalP.Scores, "^NN.{0,}","SP")) %>% 
  #dplyr::mutate(NonSyn_Syn.SNP.Ratio.All.Strains=as.numeric(NonSyn_Syn.SNP.Ratio.All.Strains),
  #              Total.SNPs.All.Strains=as.numeric(Total.SNPs.All.Strains),
  #              Transcript.Length=as.numeric(Transcript.Length),
  #              Ortholog.count=as.numeric(Ortholog.count)) %>% 
  #dplyr::mutate(SignalP=stringr::str_replace(SignalP.Scores, "^NN.{0,}","SP")) %>% 
  #group_by(SignalP.Scores,SignalP) %>% dplyr::count()
  #dplyr::rename(SignalP=SignalP.Scores) %>% 
  #dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_003775", "MSP4", Gene.Name)) %>%
  #dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_003770", "MSP5", Gene.Name)) %>% 
  #dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_097625", "MSP8", Gene.Name)) %>% 
  #dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_114145", "MSP10", Gene.Name)) %>% 
  #dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_116780", "SFT2", Gene.Name))
#lg
head(full)
#raw %>% filter(Gene.ID=="PVX_003775")#only_2o2 --> IS THIS SUGGESTING A MISTAKE IN ID TIPYING? --> COMPARE!
#raw %>% filter(stringr::str_detect(Description, "MSP7"))
```

##### GO distribution

```{r,fig.height=5,fig.width=7.7}
all_go <- all %>% select(Gene.ID,Product.Description,GO.Components) %>% 
  separate(GO.Components, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,GO.Components,-Gene.ID,-Product.Description) %>% 
  filter(GO.Components!="NA") %>% #dplyr::count(GO.Components) %>% dplyr::arrange(desc(n)) #%>% filter(n>1)
  dplyr::mutate(summary.db="Pf/Pv500 microarray",
                protype= stringr::str_count(Product.Description, "hypothetical"),
                protype=as.factor(protype),
                protype= forcats::fct_recode(protype, "hypothetical"="1", "known"="0")) %>%
  select(Gene.ID,Product.Description,protype,summary.db,GO.Components) #%>% dplyr::count(protype)

full_go <- full %>% select(Gene.ID,Product.Description,GO.Components) %>% 
  separate(GO.Components, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,GO.Components,-Gene.ID,-Product.Description) %>% 
  filter(GO.Components!="NA") %>% #dplyr::count(GO.Components) %>% dplyr::arrange(desc(n)) #%>% filter(n>1)
  dplyr::mutate(summary.db="P. vivax whole genome",
                protype= stringr::str_count(Product.Description, "hypothetical"),
                protype=as.factor(protype),
                protype= forcats::fct_recode(protype, "hypothetical"="1", "known"="0")) %>%
  select(Gene.ID,Product.Description,protype,summary.db,GO.Components) #%>% dplyr::count(protype)

go_ord <- rbind(full_go,all_go) %>% #class()
  group_by(GO.Components,summary.db) %>% 
  dplyr::summarise(count=n()) %>% ungroup() %>% 
  dplyr::arrange(desc(count)) %>% 
  spread(summary.db,count) %>% 
  dplyr::rename(micro="Pf/Pv500 microarray",
                whole="P. vivax whole genome") %>% 
  dplyr::arrange(desc(micro)) %>% 
  replace_na(list(micro = 0)) %>% 
  filter(micro>1)

#rbind(a,b,c,d) 
rbind(full_go,all_go) %>% 
  inner_join(go_ord,by = "GO.Components") %>% 
  dplyr::arrange(desc(micro)) %>% 
  #replace_na(list(SignalP = "na")) %>% 
  ggplot(aes(#x=summary.db#, fill=GO.Components
             x=reorder(GO.Components,micro,order = TRUE), fill=protype
             )) +
  geom_bar(#position = "fill"
           #position = "identity"
           ) +
  facet_grid(.~summary.db#, scales = "free"#, space = "free"
             ) +#
  coord_flip() + theme_bw() +
  scale_fill_discrete(name="Protein",
                      labels=c("with annotation", "hypothetical")) +
  theme(legend.position=c(-.73, 0.04),#"bottom"
        legend.margin = margin(0,0,0,0),
        axis.title.y=element_blank()
        #legend.text = element_text(size = 8),
        #legend.title = element_text(size = 8)
        ) +
  labs(title="Gene Ontology: Pf/Pv500 microarray proteins",
       subtitle="Cellular Component prediction compared to P. vivax genome") +
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black"#,#size = 12, 
                                    #face = "bold"
                                    ))
```

```{r,fig.height=3.5,fig.width=6}
#rbind(a,b,c,d) 
rbind(full_go,all_go) %>% 
  inner_join(go_ord %>% 
               filter(micro>3),by = "GO.Components") %>% 
  dplyr::arrange(desc(micro)) %>% 
  #replace_na(list(SignalP = "na")) %>% 
  ggplot(aes(#x=summary.db#, fill=GO.Components
             x=reorder(GO.Components,micro,order = TRUE), fill=protype
             )) +
  geom_bar(#position = "fill"
           #position = "identity"
           ) +
  facet_grid(.~summary.db#, scales = "free"#, space = "free"
             ) +#
  coord_flip() + theme_bw() +
  scale_fill_discrete(name="Protein",
                      labels=c("with annotation", "hypothetical")) +
  theme(#legend.position=c(-.38, -0.04),#"bottom"
        legend.position=c(.85, 0.15),
        legend.margin = margin(0,0,0,0),
        axis.title.y=element_blank()
        #legend.text = element_text(size = 8),
        #legend.title = element_text(size = 8)
        ) +
  labs(title="Gene Ontology: Pf/Pv500 microarray proteins",
       subtitle="Predicted Cellular Components compared to P. vivax genome") +
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black"#,#size = 12, 
                                    #face = "bold"
                                    ))
```

```{r}
# rbind(full_go,all_go) %>% 
#   inner_join(go_ord,by = "GO.Components") %>% 
#   dplyr::arrange(desc(micro))

# go_ord %>% 
#   avallecam::print_inf()

# rbind(full_go,all_go) %>% #class()
#   group_by(GO.Components,summary.db) %>% 
#   dplyr::summarise(count=n()) %>% ungroup() %>% 
#   dplyr::arrange(desc(count))

library(compareGroups)
microarray_proteins <- rbind(full_go,all_go) %>% 
  inner_join(go_ord %>% 
               filter(micro>3),by = "GO.Components") %>% 
  dplyr::arrange(desc(micro)) %>% 
  janitor::clean_names() %>% 
  dplyr::mutate(go_components=fct_infreq(go_components)) %>% 
  dplyr::select(protype, summary_db, go_components) %>% 
  # dplyr::count(go_components) %>% 
  compareGroups(summary_db~., data=.,
                max.xlev = 30,
                chisq.test.perm = TRUE) %>% 
  createTable(digits = 1)

microarray_proteins %>% export2md()
microarray_proteins %>% 
  export2xls("table/04-tab01-microarray_proteins.xls")
```


#### Microarray data

##### Heteroskedasticity 
*pre- & post- transformation/normalization*

```{r, fig.height=3, fig.width=6}
m <- anti %>% as.data.frame() %>% 
  rownames_to_column() %>% 
  gather(sample_name, expression, -rowname) %>% 
  #mutate(expression= ifelse(expression < 30, NA, expression)) %>% 
  #dplyr::filter(rowname=="PVX_111175.544",sample_name=="LIM2017") # 21
  #dplyr::filter(sample_name=="PQSJ103")
  group_by(sample_name) %>% dplyr::summarise(mean= mean(expression),
                                             sd= sd(expression)) %>% 
  #dplyr::filter(mean<5000)
  dplyr::mutate(labl="Raw data")

n <- norx %>% as.data.frame() %>% 
  rownames_to_column() %>% 
  gather(sample_name, expression, -rowname) %>% 
  group_by(sample_name) %>% dplyr::summarise(mean= mean(expression),
                                             sd= sd(expression)) %>% 
  dplyr::mutate(labl="FOC Normalized data")

rbind(m,n) %>% 
  dplyr::mutate(labl=forcats::fct_relevel(labl,
                                          "Raw data",
                                          "FOC Normalized data")) %>% 
  ggplot(aes(mean,sd)) +
  geom_point() + theme_bw() + 
  geom_smooth(linetype=0) +
  facet_wrap(~labl,nrow = 1,ncol = 2,scales = "free") +
  labs(title="Mean-variance dependence") +
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black",size = 10#, face = "bold"
                                    ))
```


```{r, fig.height=3, fig.width=6,eval=FALSE,echo=FALSE}
m <- anti %>% as.data.frame() %>% 
  rownames_to_column() %>% 
  gather(sample_name, expression, -rowname) %>% 
  #mutate(expression= ifelse(expression < 30, NA, expression)) %>% 
  #dplyr::filter(rowname=="PVX_111175.544",sample_name=="LIM2017") # 21
  #dplyr::filter(sample_name=="PQSJ103")
  group_by(sample_name) %>% dplyr::summarise(mean= mean(expression),
                                             sd= sd(expression)) %>% 
  #dplyr::filter(mean<5000)
  ggplot(aes(mean,sd)) +
  geom_point() +
  #geom_smooth() +
  labs(title="Mean-variance dependence",
       subtitle="Raw data") + theme_bw()

n <- norx %>% as.data.frame() %>% 
  rownames_to_column() %>% 
  gather(sample_name, expression, -rowname) %>% 
  group_by(sample_name) %>% dplyr::summarise(mean= mean(expression),
                                             sd= sd(expression)) %>% 
  ggplot(aes(mean,sd)) +
  geom_point() +
  #geom_smooth() +
  labs(title="Mean-variance dependence",
       subtitle="FOC Normalization method") + theme_bw()

Rmisc::multiplot(m,n,cols = 2)
```

##### Validity test

*Using Pf IVTT and purified proteins*

As a reference, check Crompton, 2010 [@crompton2010] supplementary info [here](http://www.pnas.org/content/suppl/2010/03/25/1001323107.DCSupplemental/pnas.201001323SI.pdf).

All comparison are in raw scale, since normalization can not be done with the purified protein spots.

***

```{r, eval=FALSE}
rownames(pure)
```

__Non Present Reactive Targets__

_This was a Preliminary Cromprenhensive Screening?_

- __Load and tidy up__ the file sent in __28sep2015__ by Joe Campo.

- AMA1, ~~CSP, GEST, CelTOS~~
- Here, AMA1 and all the proteins in the list with a known _Gene Name_:

```{r}
np <- readxl::read_xlsx("data-raw/PfPv500_NonpresentReactiveTargets.xlsx") %>% 
  rename_all(
      funs(
        #stringr::str_to_lower(.) %>%
        stringr::str_replace_all(., '\\s', '\\.') %>% 
        stringr::str_replace_all(., '\\[|\\]', '')
      )
  )
#np %>% filter(Gene ID=="PVX_092275")
np %>% filter(stringr::str_detect(Product.Description, "AMA1"))
#np %>% filter(stringr::str_detect(Product.Description, "CSP"))
#np %>% filter(stringr::str_detect(Product.Description, "GEST"))
#np %>% filter(stringr::str_detect(Product.Description, "CelTOS"))
np %>% filter(stringr::str_detect(Product.Description, "\\("))
```

- In this list, __ADi__ reports a __total of `r nrow(np)`__ Non present Reactive Targets.

**Validate w.r.t Pf EBA175, MSP2, CSP**
```{r}
EBA175 <- as.data.frame(cbind(as.matrix(anti["PF3D7_0731500_e1s2.361",]), 
                              as.matrix(pure["EBA175, 0.3mg/mL.23",]), 
                              as.matrix(pure["EBA175, 0.1mg/mL.24",])
                              )
                        )
colnames(EBA175) <- c("PF3D7_0731500_e1s2.361", "EBA175_0.3.23", "EBA175_0.1.24")

# PF3D7_0930300_s1.67   falciparum      IVTT.AG PF3D7_0930300   merozoite surface protein 1 (MSP1)
MSP1s1 <- as.data.frame(cbind(as.matrix(anti["PF3D7_0930300_s1.67",]), 
                              as.matrix(pure["MSP1, 0.3mg/mL.601",]), 
                              as.matrix(pure["MSP1, 0.1mg/mL.602",])))
colnames(MSP1s1) <- c("PF3D7_0930300_s1.67", "MSP1_0.3.601", "MSP1_0.1.602")

# PF3D7_0930300_s2.941  falciparum      IVTT.AG PF3D7_0930300   merozoite surface protein 1 (MSP1)
MSP1s2 <- as.data.frame(cbind(as.matrix(anti["PF3D7_0930300_s2.941",]), 
                              as.matrix(pure["MSP1, 0.3mg/mL.601",]), 
                              as.matrix(pure["MSP1, 0.1mg/mL.602",])))
colnames(MSP1s2) <- c("PF3D7_0930300_s2.941", "MSP1_0.3.601", "MSP1_0.1.602")

# PF3D7_0206800.65      falciparum      IVTT.AG PF3D7_0206800   merozoite surface protein 2 (MSP2)
MSP2 <- as.data.frame(cbind(as.matrix(anti["PF3D7_0206800.65",]), 
                            as.matrix(pure["MSP2, 0.3mg/mL.890",]), 
                            as.matrix(pure["MSP2, 0.1mg/mL.891",])))
colnames(MSP2) <- c("PF3D7_0206800.65", "MSP2_0.3.890", "MSP2_0.1.891")

# PF3D7_0304600.932     falciparum      IVTT.AG PF3D7_0304600   circumsporozoite (CS) protein (CSP)
CSP <- as.data.frame(cbind(as.matrix(anti["PF3D7_0304600.932",]), 
                           as.matrix(pure["Pf CSP, 0.3mg/mL.894",]), 
                           as.matrix(pure["Pf CSP, 0.1mg/mL.895",])))
colnames(CSP) <- c("PF3D7_0304600.932", "PfCSP_0.3.894", "PfCSP_0.1.895")

# VIVAX AMA1
PvAMA1 <- as.data.frame(cbind(as.matrix(pure["Pvivax AMA1, 0.3mg/mL.603",]), 
                              as.matrix(pure["Pvivax AMA1, 0.1mg/mL.604",]), 
                              as.matrix(pure["Pvivax AMA1 Eoto monomer prep2, 0.3mg/mL.892",]), 
                              as.matrix(pure["Pvivax AMA1 Eoto monomer prep2, 0.1mg/mL.893",])))
colnames(PvAMA1) <- c("PvAMA1_0.3.603", "PvAMA1_0.1.604", 
                      "PvAMA1_Eoto_prep2_0.3.892", "PvAMA1_Eoto_prep2_0.1.893")

```

```{r, warning=FALSE, eval=FALSE, echo=FALSE}
cor(EBA175, method = "spearman")
cor(MSP2, method = "spearman")
cor(CSP, method = "spearman")
#PerformanceAnalytics::chart.Correlation(log(EBA175), method = "pearson")
PerformanceAnalytics::chart.Correlation(EBA175, method = "spearman")
PerformanceAnalytics::chart.Correlation(MSP2, method = "spearman")
PerformanceAnalytics::chart.Correlation(CSP, method = "spearman")
#PerformanceAnalytics::chart.Correlation(PvAMA1, method = "spearman")

#summary(lm(PF3D7_0731500_e1s2.361 ~ EBA175_0.3.23, EBA175))$r.squared
```

```{r}
f <- EBA175 %>% 
  cor.test(~ PF3D7_0731500_e1s2.361 + EBA175_0.3.23, 
           data = ., method = "spearman") %>% 
  broom::tidy() #%>% format(digits=2)

g <- MSP2 %>% 
  cor.test(~ PF3D7_0206800.65 + MSP2_0.1.891, 
           data = ., method = "spearman") %>% 
  broom::tidy() #%>% format(digits=2)

h <- CSP %>% 
  cor.test(~ PF3D7_0304600.932 + PfCSP_0.3.894, 
           data = ., method = "spearman") %>% 
  broom::tidy() #%>% format(digits=2)
```

```{r, fig.width=9, fig.height=3,eval=FALSE}
v11 <- EBA175 %>% 
  ggplot(aes(x=EBA175_0.3.23,y=PF3D7_0731500_e1s2.361)) + 
  geom_point() + #geom_smooth(method = "lm") +
  xlab("Ab reactivity to Purified EBA175") + ylab("Ab reactivity to IVTT EBA175") +
  labs(title="Validity test: PfEBA175", 
       subtitle= paste0(#"S=",f$statistic,", rho="
                        "rho=",f$estimate %>% format(digits=2),", P=",f$p.value %>% format(digits=2))#,
       #caption=paste0("method: ",f$method)
       ) + theme_bw()
  

v32 <- MSP2 %>% 
  ggplot(aes(x=MSP2_0.1.891,y=PF3D7_0206800.65)) + 
  geom_point() + #geom_smooth(method = "lm") +
  xlab("Ab reactivity to Purified MSP2") + ylab("Ab reactivity to IVTT MSP2") +
  labs(title="Validity test: PfMSP2", 
       subtitle= paste0(#"S=",g$statistic,", rho="
                        "rho=",g$estimate %>% format(digits=2),", P=",g$p.value %>% format(digits=2))#,
       #caption=paste0("method: ",g$method)
       ) + theme_bw()

v41 <- CSP %>% 
  ggplot(aes(x=PfCSP_0.3.894,y=PF3D7_0304600.932)) + 
  geom_point() + #geom_smooth(method = "lm") +
  xlab("Ab reactivity to Purified CSP") + ylab("Ab reactivity to IVTT CSP") +
  labs(title="Validity test: PfCSP", 
       subtitle= paste0(#"S=",h$statistic,", rho="
                        "rho=",h$estimate %>% format(digits=2),", P=",h$p.value %>% format(digits=2))#,
       #caption=paste0("method: ",h$method)
       ) + theme_bw()

multiplot(v11, v32, v41, cols= 3)
```

```{r, fig.width=8.5, fig.height=3}
a <- EBA175 %>% rownames_to_column() %>% 
  dplyr::mutate(antigen="Pf EBA175") %>% 
  select(rowname,antigen,x=EBA175_0.3.23,y=PF3D7_0731500_e1s2.361)
b <- MSP2 %>% rownames_to_column() %>% 
  dplyr::mutate(antigen="Pf MSP2") %>% 
  select(rowname,antigen,x=MSP2_0.1.891,y=PF3D7_0206800.65)
c <- CSP %>% rownames_to_column() %>% 
  dplyr::mutate(antigen="Pf CSP") %>% 
  select(rowname,antigen,x=PfCSP_0.3.894,y=PF3D7_0304600.932)

rbind(a,b,c) %>% 
  ggplot(aes(x,y)) +
  geom_point() + theme_bw() + #geom_smooth(method = "lm") +
  facet_grid(.~antigen#,scales = "free", space = "free"
             ) +
  geom_text(aes(x,y,label=lab),
            data = data_frame(x=0,y=50700,
                              lab=c(
                                paste0(#"S=",f$statistic,", rho="
                                  "rho=",f$estimate %>% format(digits=2),
                                  ", P",ifelse(f$p.value<0.001,"<0.001",
                                               paste0("=",f$p.value))),
                                paste0(#"S=",g$statistic,", rho="
                                  "rho=",g$estimate %>% format(digits=2),
                                  ", P",ifelse(g$p.value<0.001,"<0.001",
                                                paste0("=",g$p.value))),
                                paste0(#"S=",h$statistic,", rho="
                                  "rho=",h$estimate %>% format(digits=2),
                                  ", P",ifelse(h$p.value<0.001,"<0.001",
                                                paste0("=",h$p.value)))
                                                ),
                              antigen=c("Pf EBA175","Pf MSP2","Pf CSP")),
            vjust=.5,hjust=0.05,size=3.5) +
  xlab("Ab reactivity to Purified Antigen") + ylab("Ab reactivity to IVTT Antigen") +
  labs(title="Validity of the protein microarray assay") +
  #     subtitle= paste0(#"S=",g$statistic,", rho="
  #                      "rho=",g$estimate,", P=",g$p.value)#,
  #     #caption=paste0("method: ",g$method)
  #    ) + 
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black",size = 11#, face = "bold"
                                    ))

### TRY TO ADD MATHEMATICAL NOTATION
# https://blog.snap.uaf.edu/2013/03/25/mathematical-notation-in-r-plots/
# https://stackoverflow.com/questions/11408031/using-plotmath-symbol-in-ggplot2-geom-text-legend-is-altered-why
# https://rstudio-pubs-static.s3.amazonaws.com/136237_170402e5f0b54561bf7605bdea98267a.html
# http://ggplot2.tidyverse.org/reference/geom_text.html
# http://vis.supstat.com/2013/04/mathematical-annotation-in-r/
# https://stackoverflow.com/questions/4973898/combining-paste-and-expression-functions-in-plot-labels
```

```{r}
w <- cor(EBA175, method = "spearman") %>% as.data.frame() %>% 
  rownames_to_column(var="ID") %>% 
  slice(1) %>% 
  gather(pure.prot,rho,-ID) %>% 
  filter(rho!=1) %>% 
  inner_join(ln, by="ID")
x <- cor(MSP2, method = "spearman") %>% as.data.frame() %>% 
  rownames_to_column(var="ID") %>% 
  slice(1) %>% 
  gather(pure.prot,rho,-ID) %>% 
  filter(rho!=1) %>% 
  inner_join(ln, by="ID")
z <- cor(CSP, method = "spearman") %>% as.data.frame() %>% 
  rownames_to_column(var="ID") %>% 
  slice(1) %>% 
  gather(pure.prot,rho,-ID) %>% 
  filter(rho!=1) %>% 
  inner_join(ln, by="ID")

v <- rbind(w,x,z)#; max(v$rho); min(v$rho)
v
```

```{r}
# f
# g
# h
# list_validation <- c("0731500", "0206800", "0304600", "EBA", "MSP2", "CSP")
d <- dplyr::bind_cols(EBA175,
                      MSP1s1,MSP1s2,
                      PvAMA1,
                      MSP2,CSP) %>% 
  as_tibble() %>% 
  janitor::clean_names()

# one
# d %>% 
#   as.matrix() %>% 
#   Hmisc::rcorr() %>% 
#   broom::tidy()

#two
correlate_combo <- function(matrix_to_correlate) {
  d <- matrix_to_correlate
  
  var_pairs <- t(combn(names(d), 2)) %>%
  as_tibble() %>% 
  setNames(c("x", "y"))

var_pairs %>% 
  dplyr::mutate(r.test = purrr::map2(x, y, ~ stats::cor.test(d[[.x]], d[[.y]],
                                                             method = "spearman")),
                r.test = purrr::map(r.test, broom::tidy)) %>%
  tidyr::unnest(r.test) #%>% 
  # filter(str_detect(x,paste0(str_to_lower(list_validation),collapse = "|"))) %>% 
  # distinct(x,.keep_all = T) %>% 
  # arrange(desc(estimate)) %>% 
  # view()
}

rho_pvalue <- correlate_combo(matrix_to_correlate = EBA175) %>% slice(-3) %>% 
  bind_rows(correlate_combo(matrix_to_correlate = CSP) %>% slice(-3)) %>% 
  bind_rows(correlate_combo(matrix_to_correlate = MSP2) %>% slice(-3)) #%>% view()

rho_pvalue
rho_pvalue %>% 
  writexl::write_xlsx("table/04-tab02-validation_correlation.xlsx")

correlate_combo(matrix_to_correlate = PvAMA1) #%>% view()

#three
library(corrr)
d %>%
  correlate() %>%
  rplot() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
  # focus(mpg:drat, mirror = TRUE) %>%
  # network_plot()
```


__RANGO__ de valores `rho`: __`r format(min(v$rho),digits=2)` - `r max(v$rho) %>% format(digits=2)`__

### 5. Group comparison

#### 5.1 severe

```{r}
eset <- eset.VIVAX.SEV.FILTER
pData(eset) %>% glimpse()
```

##### Differential expression

###### Define the control group

```{r}
#eset$sev_WHO <- factor(eset$sev_WHO)

table(eset$sev_WHO) # debe ser: control (fct referencia) vs caso
eset$sev_WHO <- relevel(eset$sev_WHO, ref = "severo")
table(eset$sev_WHO) # debe ser: control (fct referencia) vs caso
```

###### Intervention [IMPORTANT]

An critical evaluation the p value histogram 
generated for the severe comparison
using moderated t-test 
showed that its distribution was irregular
(https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6164648/)
one suggested alternative was to use a wilcoxon rank sum test
instead od a t test
following lindeloev (https://lindeloev.github.io/tests-as-linear/#51_independent_t-test_and_mann-whitney_u)
we transformed the raw values of the matrix to rank values
to apply a linear model to data and remake a non-parametric test

chunk in eval = FALSE by default

```{r,eval=FALSE}
rank_to_eset <- function(eset) {
  
  esApply(X = eset,MARGIN = 1,FUN = rank) %>% t()
  
}

eset <- ExpressionSet(assayData = rank_to_eset(eset = eset), 
                      phenoData = eset@phenoData,
                      featureData = eset@featureData)

eset@assayData$exprs %>% head()
```


###### Matrix + Linear model + eBayes

```{r}
design <- model.matrix(~ eset$sev_WHO)
fit <- lmFit(eset, design)
eb <- eBayes(fit)
topTable(eb) %>% rownames_to_column()
```

```{r}
# tidy toptable
td <- topTable(eb, coef = 2,               # default: slope
         sort.by =NULL, resort.by =NULL, 
         #genelist = NULL,                 # no fit$gene
         number = Inf,                     # show all the hipothesis
         confint=0.95) %>% rownames_to_column() %>% as.tbl() %>% 
  dplyr::rename(ID=rowname) %>% 
  arrange(P.Value) %>% 
  dplyr::mutate(p.order = seq(n())) %>% 
  mutate(significance=if_else(adj.P.Val<0.05,
                              "FDR<0.05","not sig.")) %>% #dplyr::count(significance)
  inner_join(biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% #dplyr::count(sev_WHO)
               group_by(gene,sev_WHO) %>% dplyr::summarise(mean_group=mean(value)) %>% 
               ungroup() %>% 
               spread(sev_WHO,mean_group) %>% 
               #arrange(desc(no_severo)) %>% 
               dplyr::rename(ID=gene),
             by="ID")
```

```{r}
td %>% 
  filter(P.Value<0.05)
td %>% 
  filter(P.Value<0.05) %>% 
  serosurvey::unite_dotwhiskers(variable_dot = logFC,
                                variable_low = CI.L,
                                variable_upp = CI.R,
                                digits_dot = 2,
                                digits_low = 1,
                                digits_upp = 1,
                                decimal_to_percent = F) %>% 
  select(ID,Gene.ID,Description,
         AveExpr,no_severo,severo,
         logFCunite1_,P.Value,adj.P.Val,
         everything()) %>% 
  # view()
  writexl::write_xlsx("table/04-tab03-severe_compare-pvalue_results.xlsx")
  # write("table/04-table03-severe_compare-protein_features.xlsx")
```

##### Plots

###### p-value histogram

```{r, fig.height=3, fig.width=6}
ax <- td %>% ggplot(aes(P.Value)) +
  geom_histogram(breaks=seq(0, 1, by = .05)) +
  scale_x_continuous(breaks=seq(0, 1, by = .5)) +
  geom_vline(aes(xintercept=0.05), lty=3, col="red") +
  labs(title="P-value histogram",
       subtitle="Without multiple testing correction") + xlab("p.value") + theme_bw()
bx <-  td %>% ggplot(aes(adj.P.Val)) +
  geom_histogram(breaks=seq(0, 1, by = .05)) +
  scale_x_continuous(breaks=seq(0, 1, by = .5)) +
  geom_vline(aes(xintercept=0.05), lty=3, col="red") +
  coord_cartesian(xlim = c(0,1)) +
  labs(title="P-value histogram",
       subtitle="After multiple testing correction") + xlab("adjust p.value") + theme_bw()
Rmisc::multiplot(ax,bx,cols = 2)
```

###### volcano plot

```{r, fig.height=6, fig.width=3.4}
td %>% 
  mutate(unadj_significance=if_else(P.Value<0.05,
                              "P.Value<0.05","not sig.")) %>% #dplyr::count(significance)
  ggplot(aes(logFC,-log10(P.Value))) +
  geom_point(aes(colour=unadj_significance)) + 
  scale_color_manual(name="unadjust\nsignificance",
                     values=c("black","red")) + 
  theme_bw() +
  #geom_point(aes(colour=no_severo)) + viridis::scale_color_viridis() +
  #ggrepel::geom_text_repel(data = td %>% 
  #                           top_n(-10,P.Value) %>% 
  #                           inner_join(all, by="Gene.ID") %>% 
  #                           inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")
  #                           #filter(adj.P.Val<0.05 & logFC>1.5 & no_severo>1)# %>% 
  #                           #arrange(desc(no_severo)) %>% 
  #                           #top_n(10,no_severo)
  #                         ,
  #                         aes(label=Gene.Name), #force = 5,
  #                         #direction = "both",
  #                         box.padding = unit(5.5, "lines"),
  #                         point.padding = unit(.5, "lines")
  #                         ) + 
  theme(legend.position=c(.2, .91),#"bottom"
        legend.margin = margin(2,2,2,2)) +
  labs(title="Non-severe vs Severe",
       #title="No-severo vs Severo",
       subtitle="Effect size against statistical significance"#,
       #subtitle="Tamaño del efecto y significancia estadística",
       #caption="annotated features filtered by\nP.Value<0.01"
       )+
  geom_hline(aes(yintercept=-log10(0.05)), lty=3)+
  geom_vline(aes(xintercept=1), lty=3) +
  geom_vline(aes(xintercept=-1), lty=3) +
  coord_cartesian(xlim = c(-2, 2), ylim = c(0,15))
```

###### figure output 2

```{r, fig.height=6, fig.width=3}

adj_pval_limit <- td %>% 
  dplyr::count(adj.P.Val) %>% 
  arrange(adj.P.Val) %>% 
  top_n(n = -2,wt = adj.P.Val) %>% 
  pull(adj.P.Val) %>% 
  pluck(2) %>% 
  round(digits = 2)

severe_plot02_b <- td %>% 
  ggplot(aes(logFC,-log10(P.Value))) +
  geom_point(aes(colour=adj.P.Val)) + 
  # geom_point(aes(colour=significance)) + 
  # scale_color_viridis_c() +
  scale_color_binned(type = "viridis",show.limits = TRUE) +
  # guides(color = guide_bins(show.limits = TRUE))
  # scale_color_manual(values=c(#"red",
  #                             "black")) +
  #geom_point(aes(colour=con)) + viridis::scale_color_viridis() +
  ggrepel::geom_text_repel(data = td %>% 
                             # filter((adj.P.Val<0.30 #& abs(logFC)>0.6
                             filter((adj.P.Val<=adj_pval_limit #& abs(logFC)>0.6
                                     )# | 
                             # filter((adj.P.Val<0.05 & logFC>1 & con>1)# | 
                                      #(adj.P.Val<0.001 & #logFC>0.9 & 
                                      #   con>2.7) 
                                    ) %>% 
                             inner_join(all, by="Gene.ID") %>% 
                             inner_join(lg %>% 
                                          dplyr::rename(seg.num=n), 
                                        by="Gene.ID")  %>% 
                             mutate(newname=case_when(
                               is.na(Gene.Name) ~ Gene.ID,
                               TRUE ~ Gene.Name
                             )) #%>% 
                             #arrange(desc(con)) %>% 
                             #top_n(10,con)
                           ,
                           aes(label=newname), 
                           size = 3,
                           # force = 15,
                           # direction = "x",
                           # box.padding = unit(4, "lines"),
                           # point.padding = unit(.5, "lines")
                           ) + 
  theme_bw() +
  #theme(legend.position="bottom") +
  # theme(#legend.position=c(.18, .44),
  #       legend.position=c(.17, .92),
  #       legend.margin = margin(2,2,2,2)) +
  # ylab( expression(log[2]~transformed~normalized~MFI) ) +  
  labs(#title="Non-severe vs Severe",
    # x = "log fold change",
    x = expression(log[2]~fold~change),
    # y = "-log10(p)", 
    y = expression(-log[10](p)), 
    color = "BH\nadjusted"
    #title="No-severo vs Severo",
    # subtitle="Effect size against statistical significance"#,
    #subtitle="Tamaño del efecto y significancia estadística",
    #caption="annotated features filtered by\nP.Value<0.01"
  ) +
  # or (adj.P.Val<0.001 & con>AveExpr(top1))
  geom_hline(aes(yintercept=-log10(0.05)), lty=3)+
  geom_vline(aes(xintercept=-1), lty=3) +
  geom_vline(aes(xintercept=1), lty=3) #+
  # xlim(-15,15)
  #geom_vline(aes(xintercept=-1), lty=3) +
  # coord_cartesian(xlim = c(-2, 2), ylim = c(0,15)) 
# severe_plot02_b

td %>% 
  ggplot(aes(P.Value)) +
  geom_histogram(breaks=seq(0, 1, by = .05)) +
  scale_x_continuous(breaks=seq(0, 1, by = .2)) +
  geom_vline(aes(xintercept=0.05), lty=3, col="red") +
  labs(x = "p", y = "Frequency") +
  severe_plot02_b +
  patchwork::plot_annotation(#title = "Non-severe vs Severe",
                             tag_levels = "A")


ggsave("figure/04-fig07-severe_compare-pvalue_histogram.png",
       height = 3,width = 6,dpi = "retina")
```

##### diff reactive

###### list

```{r}
sev_d <- td %>% 
  filter(#adj.P.Val<0.05 &
           P.Value<0.05 #& 
           #logFC>1 & 
           #no_severo>1
           ) %>%
  #top_n(10,t) %>% 
  top_n(-10,P.Value) %>% 
  arrange(P.Value) %>% 
  #arrange(desc(no_severo)) %>% 
  select(ID,Gene.ID,Description,logFC,severo,no_severo,t,P.Value,adj.P.Val,p.order) %>% #slice(1:10)
  inner_join(all, by="Gene.ID") %>% inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")

sev_d
```


##### top reactive

###### list

```{r}
sev_t <- td %>% 
  arrange(desc(AveExpr)) %>% 
  select(ID,Gene.ID,Description,AveExpr,p.order) %>% slice(1:10) %>% 
  inner_join(all, by="Gene.ID") %>% inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")

sev_t
```

##### both

###### heatmap

```{r}
eset <- eset.VIVAX.SEV.FILTER

s <- td %>% 
  #filter(adj.P.Val<0.05 & logFC>1 & no_severo>1) %>% 
  filter(#adj.P.Val<0.05 &
           P.Value<0.05 #& 
           #logFC>1 & 
           #no_severo>1
           ) %>%
  top_n(-10,P.Value) %>% 
  arrange(P.Value) %>% 
  #arrange(desc(no_severo)) %>% 
  select(ID) %>% as.matrix() 

x <- td %>% 
  arrange(desc(AveExpr)) %>% 
  top_n(10,AveExpr) %>% 
  arrange(desc(no_severo)) %>% # If diff order at boxplot is achieved, then desactivate this line.
  select(ID) %>% as.matrix() 

pData(eset) <- biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
  group_by(sample) %>% dplyr::summarise(sample_mean=mean(value, na.rm=TRUE)) %>% 
  full_join(pData(eset) %>% 
              rownames_to_column() %>% 
              dplyr::rename(sample=rowname),
            by="sample") %>% 
  select(-Study,-Group,-episodio_previo,-edad_CAT, -expo_CAT, -episodio_previo_num) %>% 
  arrange(sev_WHO,sample_mean) %>% #dplyr::count(sample)
  as.data.frame() %>% #class()
  column_to_rownames(var="sample")

y <- pData(eset) %>% rownames_to_column() %>% 
  select(rowname) %>% as.matrix() %>% as.character()

eset <- eset[c(x,s),y]

aheatmap(exprs(eset), 
         Rowv = NA, Colv = NA, 
         annCol = pData(eset)#, 
         #layout = "_"
         )

aheatmap(exprs(eset), 
         Rowv = TRUE, Colv = NA, 
         annCol = pData(eset)#, 
         #layout = "_"
         )
```

###### boxplot

```{r, fig.width=5.1,fig.height=4, echo=TRUE}
eset <- eset.VIVAX.SEV.FILTER

biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% #count(gene)
  dplyr::rename(ID="gene") %>% 
  full_join(td, by = "ID") %>%
  full_join(x %>% as.data.frame() %>% 
              dplyr::mutate(group="Top 10") %>% 
              #dplyr::rename(gene="ID") %>% 
              dplyr::mutate(ID=as.character(ID)),
            by="ID") %>% 
  full_join(s %>% as.data.frame() %>% 
              dplyr::mutate(group="p < 0.05") %>% 
              #dplyr::rename(gene="ID") %>% 
              dplyr::mutate(ID=as.character(ID)),
            by="ID") %>% 
  unite(group, group.x, group.y) %>% 
  mutate(group=stringr::str_replace(group,"NA_(.)", "\\1")) %>% 
  mutate(group=stringr::str_replace(group,"(.)_NA", "\\1")) %>% 
  mutate(group=forcats::fct_relevel(group,"Top 10")) %>% 
  filter(group!="NA") %>% 
  inner_join(all, by="Gene.ID") %>% 
  inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID") %>% 
  #unite(id.name, Product.Description,Gene.Name, sep = " / ", remove = FALSE) %>% 
  unite(id.name, Gene.Name, Gene.ID, sep = " / ", remove = FALSE) %>% 
  mutate(id.name=str_replace(id.name,"NA / ","")) %>% 
  mutate(id.name=str_replace(id.name,"/","-")) %>% 
  mutate(sev_WHO=fct_rev(sev_WHO)) %>% 
  # test zone
  # mutate(id.name=fct_reorder(.f = id.name,
  #                            .x = value,
  #                            .fun = mean)) %>%
  # #                            .fun = median)) %>%
  # mutate(id.name=fct_reorder2(.f = id.name,
  #                             .x = value,
  #                             .y = no_severo,
  #                             .fun = median)) %>%
  # mutate(id.name = tidytext::reorder_within(x = id.name,
  #                                           by = value,
  #                                           within = group,
  #                                           fun = median)) %>%
  # filter(group=="Top 10") %>% 
  # arrange(id.name) %>% 
  # glimpse()
  # select(id.name,no_severo,value,sev_WHO) #%>%
  # dplyr::count(sev_WHO)
  # ggplot(aes(reorder(id.name,-p.order),value,fill=sev_WHO))+
  
  # All sorted w.r.t. Case mean reactivity
  # ggplot(aes(x = id.name,
  ggplot(aes(x = reorder(id.name,no_severo),
             y = value,
             fill = sev_WHO))+
  geom_boxplot()+
  #geom_point() +
  facet_grid(group~.,scales = "free", space = "free") +
  colorspace::scale_fill_discrete_qualitative(
    name="P. vivax\nmalaria",
    palette="Set 3",
    #breaks=c("with", "without"),
    limits=c("no_severo","severo"),
    labels=c("Non-severe", "Severe"),
    rev=TRUE
  ) +
  # scale_fill_discrete(name="WHO\ncriteria",
  #                     #breaks=c("with", "without"),
  #                     limits=c("no_severo","severo"),
  #                     labels=c("non-severe", "severe")
  #                     ) +
  theme_bw() +
  # guides(fill = guide_legend(title.position = "left")) +
  # theme(legend.position="bottom",
  #       legend.justification = "left",
  #       legend.direction = "vertical"
  #       # legend.position=c(-.25, -.13),#"bottom"
  #       # legend.margin = margin(0,0,0,0)#,
  #       #legend.text = element_text(size = 8),
  #       #legend.title = element_text(size = 8)
  #       ) +
  ylab( expression(log[2]~transformed~normalized~MFI) ) +
  # ylab("log2-transformed normalized MFI") +
  xlab("") +
  # labs(#title="Highly and Differentially reactive antigens",
  #      # subtitle="All sorted w.r.t. Case mean reactivity",
  #      # caption="* unadjust p<0.05"
  #      )+
  coord_flip() #+
  # theme(strip.background = element_rect(colour = "black"
  #                                       #, fill = "white"
  #                                       ),
  #       strip.text.y = element_text(colour = "black",
  #                                   size = 10#, face = "bold"
  #                                   ))
ggsave("figure/04-fig08-severe_compare-boxplot_highrve.png",
       height = 3,width = 6,dpi = "retina")
```

#### 5.2 previous episode

```{r}
eset <- eset.VIVAX.SEV.FILTER
pData(eset) %>% glimpse()
```

##### Differential expression

###### Define the control group

```{r}
#eset$episodio_previo <- factor(eset$episodio_previo)

table(eset$episodio_previo) # debe ser: control (fct referencia) vs caso
eset$episodio_previo <- relevel(eset$episodio_previo, ref = "sin")
table(eset$episodio_previo) # debe ser: control (fct referencia) vs caso
```

###### Matrix + Linear model + eBayes

```{r}
# related with manual imputation in 'pheno' object
eset <- eset[,(rowSums(is.na(pData(eset)["episodio_previo"]))==0)] # new
design <- model.matrix(~ eset$episodio_previo)
fit <- lmFit(eset, design)
eb <- eBayes(fit)
topTable(eb) %>% rownames_to_column() #%>% filter(adj.P.Val<0.05)#number=Inf
```

```{r}
# tidy toptable
td <- topTable(eb, coef = 2,               # default: slope
         sort.by =NULL, resort.by =NULL, 
         #genelist = NULL,                 # no fit$gene
         number = Inf,                     # show all the hipothesis
         confint=0.95) %>% rownames_to_column() %>% as.tbl() %>% 
  dplyr::rename(ID=rowname) %>% 
  arrange(P.Value) %>% 
  dplyr::mutate(p.order = seq(n())) %>% 
  mutate(significance=if_else(adj.P.Val<0.05,
                              "FDR<0.05","not sig.")) %>% #dplyr::count(significance)
  inner_join(biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% #dplyr::count(episodio_previo)
               group_by(gene,episodio_previo) %>% dplyr::summarise(mean_group=mean(value)) %>% 
               ungroup() %>% 
               spread(episodio_previo,mean_group) %>% 
               #arrange(desc(con)) %>% 
               dplyr::rename(ID=gene),
             by="ID") 
```

##### Plots

###### p-value histogram

```{r, fig.height=3, fig.width=6}
ax <- td %>% ggplot(aes(P.Value)) +
  geom_histogram(breaks=seq(0, 1, by = .05)) +
  scale_x_continuous(breaks=seq(0, 1, by = .5)) +
  geom_vline(aes(xintercept=0.05), lty=3, col="red") +
  labs(title="P-value histogram",
       subtitle="Without multiple testing correction") + xlab("p.value") + theme_bw()
bx <-  td %>% ggplot(aes(adj.P.Val)) +
  geom_histogram(breaks=seq(0, 1, by = .05)) +
  scale_x_continuous(breaks=seq(0, 1, by = .5)) +
  geom_vline(aes(xintercept=0.05), lty=3, col="red") +
  coord_cartesian(xlim = c(0,1)) +
  labs(title="P-value histogram",
       subtitle="After multiple testing correction") + xlab("adjust p.value") + theme_bw()
Rmisc::multiplot(ax,bx,cols = 2)
```

###### volcano plot

```{r, fig.height=6, fig.width=3.4}
td %>% 
  ggplot(aes(logFC,-log10(P.Value))) +
  geom_point(aes(colour=significance)) + scale_color_manual(values=c("red","black")) +
  #geom_point(aes(colour=con)) + viridis::scale_color_viridis() +
  ggrepel::geom_text_repel(data = td %>% 
                             filter((adj.P.Val<0.05 & logFC>1 & con>1)# | 
                                      #(adj.P.Val<0.001 & #logFC>0.9 & 
                                      #   con>2.7) 
                                    ) %>% 
                             inner_join(all, by="Gene.ID") %>% 
                             inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID") #%>% 
                             #arrange(desc(con)) %>% 
                             #top_n(10,con)
                           ,
                           aes(label=Gene.Name), #force = 15,
                           direction = "x",
                           box.padding = unit(4, "lines"),
                           point.padding = unit(.5, "lines")
                           ) + theme_bw() +
  #theme(legend.position="bottom") +
  theme(#legend.position=c(.18, .44),
        legend.position=c(.17, .92),
        legend.margin = margin(2,2,2,2)) +
    labs(title="With vs Without previous episode",
       #title="Con vs Sin episodio previo",
       subtitle="Effect size against statistical significance"#,
       #subtitle="Tamaño del efecto y significancia estadística",
       #caption="annotated features filtered by\n(adj.P.Val<0.05 & con>1 & logFC>1.5)"
       )+
  # or (adj.P.Val<0.001 & con>AveExpr(top1))
  geom_hline(aes(yintercept=-log10(0.05)), lty=3)+
  geom_vline(aes(xintercept=1), lty=3) +
  #geom_vline(aes(xintercept=-1), lty=3) +
  coord_cartesian(xlim = c(-2, 2), ylim = c(0,15)) 
```

```{r, fig.height=6, fig.width=3.4}
td %>% 
  ggplot(aes(logFC,-log10(P.Value))) +
  #geom_point(aes(colour=significance)) + scale_color_manual(values=c("red","black")) +
  geom_point(aes(colour=con)) + viridis::scale_color_viridis(name="Case\nmean\nintensity") +
  ggrepel::geom_text_repel(data = td %>% 
                             filter(#(adj.P.Val<0.05 & logFC>1.5 & con>1) | 
                                      (adj.P.Val<0.001 & #logFC>0.9 & 
                                         con>2.7) ) %>% 
                             inner_join(all, by="Gene.ID") %>% 
                             inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")
                             #arrange(desc(con)) %>% 
                             #top_n(10,con)
                           ,
                           aes(label=Product.Description), #force = 15,
                           direction = "x",
                           box.padding = unit(7, "lines"),
                           point.padding = unit(.4, "lines")
                           ) + theme_bw() +
  #theme(legend.position="bottom") +
  theme(legend.position=c(.11, .84),
        legend.margin = margin(2,2,2,2),
        legend.title = element_text(size=10)) +
  guides(colour = guide_colorbar(barwidth = 1, barheight = 5)) +
  #guides(colour = guide_legend(title.theme = element_text(size = 15))) +
  labs(title="With vs Without previous episode",
       #title="Con vs Sin episodio previo",
       subtitle="Effect size against statistical significance"#,
       #subtitle="Tamaño del efecto y significancia estadística",
       #caption="annotated features filtered by\nlogFC>1.5 or Case>AveExpr(top1)"
       )+
  # or (adj.P.Val<0.001 & con>AveExpr(top1))
  geom_hline(aes(yintercept=-log10(0.05)), lty=3)+
  geom_vline(aes(xintercept=1), lty=3) +
  coord_cartesian(xlim = c(-2, 2), ylim = c(0,15)) 
```

##### diff reactive

###### Cutt-off points

```{r, fig.height=6, fig.width=3}
sp <- td %>% filter(adj.P.Val<0.05, logFC>0) %>% dplyr::count()
sc <- td %>% filter(adj.P.Val<0.05, con>1) %>% dplyr::count()
sf <- td %>% filter(adj.P.Val<0.05, con>1, logFC>1) %>% dplyr::count()
top <- td %>% top_n(1,AveExpr) %>% select(AveExpr)

l <- td %>% 
  filter(adj.P.Val<0.05) %>% 
  select(p.order, logFC) %>% 
  arrange(logFC) %>% 
  ggplot(aes(logFC)) + geom_histogram() +
  geom_vline(aes(xintercept=1), lty=3, col="red") +
  labs(title="Distribution of logFC",
       subtitle="filtered by adj.P.Val<0.05") + theme_bw()

t <- td %>% 
  filter(adj.P.Val<0.05, logFC>0) %>% 
  select(p.order, logFC,con) %>% 
  arrange(logFC) %>% 
  ggplot(aes(logFC,con)) + geom_line() + #geom_point(size=.5) +
  geom_vline(aes(xintercept=1), lty=3, col="red") +
  #geom_hline(aes(yintercept=1), lty=3, col="red") +
  geom_hline(aes(yintercept=top$AveExpr), lty=2, col="grey") +
  ylab("case mean intensity") +
  labs(title="Case mean intensity vs logFC",
       subtitle="filtered by adj.P.Val<0.05") + theme_bw()

Rmisc::multiplot(l,t,cols = 1)
```

###### Application

- __PROBLEMA:__
    - Regla de decisión y gráficas.
    - Tradicionalmente,
        + Primero, generar lista de antígenos con reactividad diferenciada 
        (p.value o adj.p.val<0.05).
        + Segundo, ordenar lista por __intensidad del grupo Caso__.
        + Tercero, graficar con doble eje: (i) lista en diagramas de barra y (ii)
        p.value en diagrama barra opuesto o puntos+línea sobrelapante.
    - Bajo este protocolo, 
        + luego de generada la lista, ni la __significancia estadística__ 
        ni el __tamaño del efecto__ (del caso sobre el control) 
        son tomados en cuenta. 
        + Se pierde resolución sobre la distribución real de los datos por antígeno.
    - __Parámetros relevantes__:
        + significancia estadística,
        + intensidad del grupo Caso,
        + tamaño del efecto.

- __SOLUTION:__
    
    1. __Statistical significance:__
        - __previously exposed__ patients differentially recognized __`r sp$n`__ P. vivax protein segments
    2. __Case mean intensity:__
        - __previously exposed__ patients differentially recognized __`r sc$n`__ P. vivax protein segments
        with a __case mean intensity higher than 1__ (i.e., 2-fold higher than __noDNA control__).
    3. __Size Effect:__
        - __previously exposed__ patients differentially recognized __`r sf$n`__ P. vivax protein segments
        with a __fold change higher than 1__ (i.e., 2-fold reactivity in Case vs Control).
    4. __Exception:__
        - __previously exposed__ patients differentially recognized __1__ P. vivax protein segment
        with a __fold change higher than 0.97__ and with __the highest case mean intensity__ among 
        this subset.
        (moreover, higher than the average intensity of the antigen with the highest reactivity).
    4. __Report of discoveries:__
        - Final sorting w.r.t Case mean intensity.
        - __Boxplot__ declaring the parameters of the executed sorting.
            + _issue:_ observed variance, not posterior variance (after eBayes).

###### list

```{r}
exp_da <- td %>% filter(adj.P.Val<0.05, logFC>0) %>% #dplyr::count(Gene.ID) %>% filter(n==2)
  arrange(desc(con)) %>% 
  select(ID,Gene.ID,Description,logFC,con,sin,t,P.Value,adj.P.Val,p.order) %>% #slice(1:10)
  inner_join(all, by="Gene.ID") %>% inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")

exp_dc <- td %>% filter(adj.P.Val<0.05, con>1) %>% #dplyr::count(Gene.ID) %>% filter(n==2)
  arrange(desc(con)) %>% 
  select(ID,Gene.ID,Description,logFC,con,sin,t,P.Value,adj.P.Val,p.order) %>% #slice(1:10)
  inner_join(all, by="Gene.ID") %>% inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")

exp_df <- td %>% 
  filter(adj.P.Val<0.05 & (logFC>1 & con>1) | (logFC>0.9 & con>2)) %>%
  #top_n(-16, p.order) %>% 
  #arrange(desc(logFC)) %>% 
  arrange(desc(con)) %>% 
  select(ID,Gene.ID,Description,logFC,con,sin,t,P.Value,adj.P.Val,p.order) %>% #slice(1:10)
  inner_join(all, by="Gene.ID") %>% inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")

exp_df
```

##### top reactive

###### list

```{r}
exp_t <- td %>% 
  arrange(desc(AveExpr)) %>% 
  select(ID,Gene.ID,Description,AveExpr,p.order,logFC) %>% slice(1:10) %>% 
  inner_join(all, by="Gene.ID") %>% inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")

exp_t 
```

##### both

###### heatmap

```{r}
eset <- eset.VIVAX.SEV.FILTER

s <- td %>% 
  filter(adj.P.Val<0.05 & (logFC>1 & con>1) | (logFC>0.9 & con>2)) %>% 
  arrange(desc(con)) %>% 
  select(ID) %>% as.matrix() 

x <- td %>% 
  arrange(desc(AveExpr)) %>% 
  top_n(10,AveExpr) %>% 
  arrange(desc(con)) %>% # If diff order at boxplot is achieved, then desactivate this line.
  select(ID) %>% as.matrix() 

pData(eset) <- biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
  group_by(sample) %>% dplyr::summarise(sample_mean=mean(value, na.rm=TRUE)) %>% 
  full_join(pData(eset) %>% 
              rownames_to_column() %>% 
              dplyr::rename(sample=rowname),
            by="sample") %>% 
  select(-Study,-Group,-sev_WHO,-edad_CAT, -expo_CAT, -sev_WHO_num) %>% 
  arrange(episodio_previo,sample_mean) %>% #dplyr::count(sample)
  as.data.frame() %>% #class()
  column_to_rownames(var="sample")

y <- pData(eset) %>% rownames_to_column() %>% 
  select(rowname) %>% as.matrix() %>% as.character()

eset <- eset[c(x,s),y]

aheatmap(exprs(eset), 
         Rowv = NA, Colv = NA, 
         annCol = pData(eset)#, 
         #layout = "_"
         )

aheatmap(exprs(eset), 
         Rowv = TRUE, Colv = NA, 
         annCol = pData(eset)#, 
         #layout = "_"
         )
```

###### boxplot

```{r, fig.width=5.15,fig.height=4.5, echo=TRUE,message=FALSE}
eset <- eset.VIVAX.SEV.FILTER

exp_bp <- biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% #count(gene)
  dplyr::rename(ID="gene") %>% 
  full_join(td, by = "ID") %>%
  full_join(x %>% as.data.frame() %>% 
              dplyr::mutate(group="Top 10") %>% 
              #dplyr::rename(gene="ID") %>% 
              dplyr::mutate(ID=as.character(ID)),
            by="ID") %>% 
  full_join(s %>% as.data.frame() %>% 
              dplyr::mutate(group="Differentially*") %>% 
              #dplyr::rename(gene="ID") %>% 
              dplyr::mutate(ID=as.character(ID)),
            by="ID") %>% 
  unite(group, group.x, group.y) %>% 
  mutate(group=stringr::str_replace(group,"NA_(.)", "\\1")) %>% 
  mutate(group=stringr::str_replace(group,"(.)_NA", "\\1")) %>% 
  mutate(group=forcats::fct_relevel(group,"Top 10")) %>% 
  mutate(episodio_previo=forcats::fct_relevel(episodio_previo,"con")) %>% 
  filter(group!="NA") %>% 
  inner_join(all, by="Gene.ID") %>% 
  inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID") %>% 
  #unite(id.name, Product.Description,Gene.Name, sep = " / ", remove = FALSE) %>% 
  unite(id.name, Gene.Name, Gene.ID, sep = " / ", remove = FALSE) %>%
  #unite(id.name, id.name, p.order, sep = " / ", remove = FALSE)  %>% 
  dplyr::mutate(id.name= ifelse(ID == "PVX_003775_2o2.150", "MSP4. / PVX_003775", id.name))

#exp_ls <- exp_bp %>% 
#  dplyr::group_by(ID,id.name,group) %>% 
#  dplyr::summarise(avg=mean(con)) %>% 
#  ungroup() %>% 
#  dplyr::arrange(group,desc(avg)) %>% 
#  dplyr::mutate(id.name= ifelse(ID == "PVX_003775_2o2.150", "MSP4 / PVX_003775", id.name))

#exp_bp$ID.lab <- plyr::mapvalues(exp_bp$ID,
#                                 from = exp_bp$ID, 
#                                 to = exp_bp$id.name)

exp_bp  %>% 
  filter(!is.na(episodio_previo)) %>% 
  #ggplot(aes(reorder(id.name,con),value,fill=episodio_previo))+
  ggplot(aes(reorder(id.name,con),value,fill=episodio_previo))+
  geom_boxplot()+
  #geom_point() +
  facet_grid(group~.,scales = "free", space = "free") +
  #scale_x_discrete(labels = rev(exp_ls$id.name)) +
  scale_fill_discrete(name="Previous\nEpisodes",
                      #breaks=c("with", "without"),
                      labels=c("with", "without")) + theme_bw() +
  guides(fill = guide_legend(title.position = "left")) +
  theme(legend.position=c(-.25, -.1),#"bottom"
        legend.margin = margin(0,0,0,0)#,
        #legend.text = element_text(size = 8),
        #legend.title = element_text(size = 8)
        ) +  
  ylab("log2-transformed normalized MFI")+
  xlab("antigens")+
  labs(title="Highly and Differentially reactive antigens",
       subtitle="All sorted w.r.t. Case mean reactivity",
       caption="*filtered by adj.P.Val<0.05 & logFC>1")+
  # and \n (logFC>1 & con>1) | (logFC>0.9 & con>2)
  coord_flip() +
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.y = element_text(colour = "black",size = 10#, face = "bold"
                                    ))
```

### 6. Breath and Intensity of response

- Método:
    + __Primero,__ Prueba de Hipótesis para comparar varianzas:
        + Rpta: Bajo un n.s. 0.05, F cae en Región de no-Rechazo de Hipótesis Nula (__RnoRHo__).
            + Conclusión: Supuesto de igualdad de varianzas poblacionales SÍ es válido.
        + Rpta: Bajo un n.s. 0.05, F cae en Región de Rechazo de Hipótesis Nula (__RRHo__).
            + Conclusión: Supuesto de igualdad de varianzas poblacionales NO es válido.
    + __Segundo,__ Prueba de Hipótesis para comparar medias:
        + Si es que SÍ se asume que las varianzas son iguales: __Student t-test__
        + Si es que NO se asume que las varianzas son iguales: __Welch t-test__

- Definiciones:
    + __Amplitud:__ Número de Ag reactivos por paciente
    + __Intensidad:__ Media de intensidad de Ag por paciente

```{r}
eset <- eset.VIVAX.SEV.FILTER

fin <- biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
  group_by(sample) %>% 
  # dplyr::summarise(sample_mean=mean(value, na.rm=TRUE), 
  # cambio por exploracion de distribuciones!
  dplyr::summarise(sample_mean=median(value, na.rm=TRUE),
                   sample_freq=(sum(value>1, na.rm=TRUE)#*100/n()
                                )) %>% 
  full_join(pData(eset) %>% 
              rownames_to_column() %>% 
              dplyr::rename(sample=rowname),
            by="sample") %>% 
  select(-Study,-Group) %>% 
  mutate(episodio_previo=forcats::fct_relevel(episodio_previo,"con")) %>% 
  mutate(sev_WHO=forcats::fct_relevel(sev_WHO,"no-severo"))

#biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
#filter(sample=="LIM2071") %>% select(value) %>% filter(value > 1)
```

__evaluate distribution per subject__

```{r,fig.height=8,fig.width=8}
# explorar distribuciones individuales
biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
  ggplot(aes(x = value)) +
  geom_histogram() +
  facet_wrap(~sample)
```

```{r,fig.width=3,fig.height=3}
biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
  select(value) %>% 
  # group_by(sample) %>% 
  skimr::skim()

biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
  select(sev_WHO,value) %>% 
  group_by(sev_WHO) %>%
  skimr::skim()

biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
  select(episodio_previo,value) %>% 
  filter(!is.na(episodio_previo)) %>% 
  group_by(episodio_previo) %>%
  skimr::skim()

biobroom::tidy.ExpressionSet(eset,addPheno = TRUE) %>% 
  select(sample,value) %>% 
  group_by(sample) %>% 
  skimr::skim() %>% 
  as_tibble() %>% 
  mutate(diff_median_mean=numeric.p50-numeric.mean) %>% 
  ggplot(aes(x = diff_median_mean)) +
  geom_histogram()
```


#### 6.1 severe

```{r}
## Primero, Prueba de Hipótesis para comparar varianzas:
#fin %>% var.test(sample_freq ~ sev_WHO, data = .) %>% broom::tidy()
## Rpta: Bajo n.s. 0.05, F cae en Región de no-Rechazo de Hipótesis Nula (RnoRHo)
## Conclusión: Supuesto de igualdad de varianzas poblacionales SÍ es válido
## Segundo, Prueba de Hipótesis para comparar medias:
j <- fin %>% 
  t.test(sample_freq ~ sev_WHO, data = ., var.equal=TRUE) %>% 
  broom::tidy() %>% #%>% format(digits=2)
  mutate(test="breadth") %>% 
  select(test,everything())

## Primero, Prueba de Hipótesis para comparar varianzas
#fin %>% var.test(sample_mean ~ sev_WHO, data = .) %>% broom::tidy()
## Rpta: Bajo n.s. 0.05, F cae en Región de no-Rechazo de Hipótesis Nula (RnoRHo)
## Conclusión: Supuesto de igualdad de varianzas poblacionales SÍ es válido
## Segundo, Prueba de Hipótesis para comparar medias:
k <- fin %>% 
  t.test(sample_mean ~ sev_WHO, data = ., var.equal=TRUE) %>% 
  broom::tidy() %>% #%>% format(digits=2)
  mutate(test="intensity") %>% 
  select(test,everything()) #%>% format(digits=2)
```

```{r}

fin %>% 
  compareGroups(formula = sev_WHO ~ sample_freq + sample_mean,data = .) %>% 
  createTable(sd.type = 2,digits = 2) %>% 
  export2xls("table/04-tab05-severe_compare-subject_response.xls")

j %>% 
  union_all(k) %>% 
  writexl::write_xlsx("table/04-tab04-severe_compare-subject_response.xlsx")
```


```{r, fig.height=6, fig.width=3}

# sources:
# https://gist.github.com/benmarwick/5cfeda1d20ff9ab3231a5de8fac36213
# https://stackoverflow.com/a/60153154
# https://stackoverflow.com/a/5294214

r_df <- j$parameter %>% format(digits=2)
r_tstat <- j$statistic %>% format(digits=2)
r_pvalue <- if_else(condition = j$p.value<0.001,
                    true = "<0.001",
                    false = paste0("=",j$p.value %>% 
                                     format(digits=3)))
# r_one <- bquote(expression(t[.(r_df)]==.(r_tstat)* ", p" * .(r_pvalue) ) )
r_one <- paste0("p",r_pvalue)


r <- fin %>% 
  ggplot(aes(sev_WHO,sample_freq)) +
  geom_violin(alpha=0,lwd=0.4,draw_quantiles = c(0.25,0.5,0.75)) +
  ggbeeswarm::geom_quasirandom(method = "tukeyDense") +
  # geom_boxplot(#position=position_dodge(0.8)
  #              ) +
  # geom_dotplot(#aes(fill=sev_WHO), 
  #              binaxis='y', stackdir='center', 
  #              dotsize=1#, position=position_dodge(0.8)
  #              ) +
  scale_x_discrete(labels = c("Non-severe","Severe")) +
  # xlab("WHO criteria") + 
  xlab("P. vivax malaria") + 
  ylab("Reactive Antigens") +
  labs(#title="Breadth of Ab response", 
    # title= eval(r_one)
    title = r_one
    # subtitle= paste0("t=",j$statistic %>% format(digits=2),
    #                  ", df=",j$parameter %>% format(digits=2),
    #                  ", p",ifelse(j$p.value<0.001,"<0.001",
    #                               paste0("=",j$p.value %>% 
    #                                        format(digits=3))))#,
    # #caption=paste0("method:",j$method)
  ) + theme_bw()

s_df <- k$parameter %>% format(digits=2)
s_tstat <- k$statistic %>% format(digits=3)
s_pvalue <- if_else(condition = k$p.value<0.001,
                    true = "<0.001",
                    false = paste0("=",k$p.value %>% 
                                     format(digits=3)))
# s_one <- bquote(expression(t[.(s_df)]==.(s_tstat)* ", p" * .(s_pvalue) ) )
s_one <- paste0("p",s_pvalue)

s <- fin %>% 
  ggplot(aes(sev_WHO,sample_mean)) +
  geom_violin(alpha=0,lwd=0.4,draw_quantiles = c(0.25,0.5,0.75)) +
  ggbeeswarm::geom_quasirandom(method = "tukeyDense") +
  # geom_boxplot(#position=position_dodge(0.8)
  #              ) +
  # geom_dotplot(#aes(fill=sev_WHO), 
  #              binaxis='y', stackdir='center', 
  #              dotsize=1#, position=position_dodge(0.8)
  #              ) +
  scale_x_discrete(labels = c("Non-severe","Severe")) +
  xlab("P. vivax malaria") + 
  ylab("Median Intensity") +
  labs(#title="Intensity of Ab response", 
    # title= eval(s_one)
    title = s_one
    # "t=",k$statistic %>% format(digits=2),
    # ", df=",k$parameter %>% format(digits=2),
    # ", p",ifelse(k$p.value<0.001,"<0.001",
    #              paste0("=",k$p.value %>% 
    #                       format(digits=3))))#,
    #caption=paste0("method:",k$method)
  ) + theme_bw()

# s
# Rmisc::multiplot(r,s,cols = 1)
r + s +
  patchwork::plot_annotation(tag_levels = "A")
ggsave("figure/04-fig06-severe_compare-subject_response.png",
       height = 3,width = 6,dpi = "retina")
```

#### 6.2 previous episode

```{r}
## Primero, Prueba de Hipótesis para comparar varianzas:
#fin %>% var.test(sample_freq ~ episodio_previo, data = .) %>% broom::tidy() %>% 
#  mutate(decide= p.value<0.05) %>% select(decide)#TRUE
## Rpta: Bajo n.s. 0.05, F cae en Región de Rechazo de Hipótesis Nula (RRHo)
## Conclusión: Supuesto de igualdad de varianzas poblacionales NO es válido
## Segundo, Prueba de Hipótesis para comparar medias:
j <- fin %>% 
  t.test(sample_freq ~ episodio_previo, data = ., var.equal=FALSE) %>% 
  broom::tidy() #%>% format(digits=3)

## Primero, Prueba de Hipótesis para comparar varianzas:
#fin %>% var.test(sample_mean ~ episodio_previo, data = .) %>% broom::tidy()
## Rpta: Bajo n.s. 0.05, F cae en Región de no-Rechazo de Hipótesis Nula (RnoRHo)
## Conclusión: Supuesto de igualdad de varianzas poblacionales SÍ es válido
## Segundo, Prueba de Hipótesis para comparar medias:
k <- fin %>% 
  t.test(sample_mean ~ episodio_previo, data = ., var.equal=TRUE) %>% 
  broom::tidy() #%>% format(digits=3)
```

```{r, fig.height=6, fig.width=3}
r <- fin %>% 
  filter(!is.na(episodio_previo)) %>% 
  ggplot(aes(episodio_previo,sample_freq)) +
  geom_boxplot(#position=position_dodge(0.8)
               ) +
  geom_dotplot(#aes(fill=sev_WHO), 
               binaxis='y', stackdir='center', 
               dotsize=1#, position=position_dodge(0.8)
               ) +
  scale_x_discrete(labels = c("with","without")) +
  xlab("Previous Episode") + ylab("Number Reactive") +
  labs(title="Breadth of Ab response", 
       subtitle= paste0("t=",j$statistic %>% format(digits=3),
                        ", df=",j$parameter %>% format(digits=3),
                        ", P",ifelse(j$p.value<0.001,"<0.001",
                                               paste0("=",j$p.value)))#,
       #caption=paste0("method: ",j$method)
       ) + theme_bw()

s <- fin %>% 
  filter(!is.na(episodio_previo)) %>% 
  ggplot(aes(episodio_previo,sample_mean)) +
  geom_boxplot(#position=position_dodge(0.8)
               ) +
  geom_dotplot(#aes(fill=sev_WHO), 
               binaxis='y', stackdir='center', 
               dotsize=1#, position=position_dodge(0.8)
               ) +
  scale_x_discrete(labels = c("with","without")) +
  xlab("Previous Episode") + ylab("Mean Intensity") +
  labs(title="Intensity of Ab response", 
       subtitle= paste0("t=",k$statistic %>% format(digits=3),
                        ", df=",k$parameter,
                        ", P",ifelse(k$p.value<0.001,"<0.001",
                                               paste0("=",k$p.value)))#,
       #caption=paste0("method:",k$method)
       ) + theme_bw()

Rmisc::multiplot(r,s,cols = 1)
```

#### 6.3 Exploratory

##### Per age

```{r}
fin %>% 
  dplyr::count(edad_CAT)
```

```{r, fig.height=3, fig.width=6}
r <- fin %>%
  ggplot(aes(edad_CAT,sample_freq#, fill=sev_WHO
             )) +
  geom_boxplot(position=position_dodge(0.8)) +
  geom_dotplot(#aes(fill=sev_WHO), 
               binaxis='y', stackdir='center', 
               dotsize=1, position=position_dodge(0.8)) +
  xlab("Categorized Age") + ylab("Number Reactive") +
  labs(title="Breadth of Ab response", subtitle="per age category")  + theme_bw()

s <- fin %>% 
  ggplot(aes(edad_CAT,sample_mean#, fill=sev_WHO
             )) +
  geom_boxplot(position=position_dodge(0.8)) +
  geom_dotplot(#aes(fill=sev_WHO), 
               binaxis='y', stackdir='center', 
               dotsize=1, position=position_dodge(0.8)) +
  xlab("Categorized Age") + ylab("Mean Intensity") +
  labs(title="Intensity of Ab response", subtitle="per age category")  + theme_bw()

#t <- fin %>% filter(expo_CAT != "NA") %>% 
#  ggplot(aes(expo_CAT,sample_freq#, fill=episodio_previo
#             )) +
#  geom_boxplot(position=position_dodge(0.8)) +
#  geom_dotplot(#aes(fill=episodio_previo), 
#               binaxis='y', stackdir='center', 
#               dotsize=1, position=position_dodge(0.8)) +
#  labs(title="Breadth of Ab response", 
#       subtitle="per previous episode category")  + theme_bw()
#
#u <- fin %>% filter(expo_CAT != "NA") %>%  
#  ggplot(aes(expo_CAT,sample_mean#, fill=episodio_previo
#             )) +
#  geom_boxplot(position=position_dodge(0.8)) +
#  geom_dotplot(#aes(fill=episodio_previo), 
#               binaxis='y', stackdir='center', 
#               dotsize=1, position=position_dodge(0.8)) +
#  labs(title="Intensity of Ab response", 
#       subtitle="per previous episode category")  + theme_bw()

Rmisc::multiplot(r,s,cols = 2)
```


##### severe and episodes per age

```{r}
fin %>% 
  group_by(sev_WHO) %>% 
  dplyr::count(edad_CAT)

fin %>% 
  group_by(episodio_previo) %>% 
  dplyr::count(edad_CAT)
```


```{r, fig.height=6, fig.width=8.5}
r <- fin %>% 
  ggplot(aes(edad_CAT,sample_freq, fill=sev_WHO)) +
  geom_boxplot(position=position_dodge(0.8)) +
  geom_dotplot(aes(fill=sev_WHO), 
               binaxis='y', stackdir='center', 
               dotsize=1, position=position_dodge(0.8)) +
  scale_fill_discrete(name="WHO criteria",
                      #breaks=c("with", "without"),
                      labels=c("non-severe", "severe")) +
  xlab("Categorized Age") + ylab("Number Reactive") +
  labs(title="Breadth of Ab response", subtitle="per Age and Severity")  + theme_bw()

s <- fin %>% 
  ggplot(aes(edad_CAT,sample_mean, fill=sev_WHO)) +
  geom_boxplot(position=position_dodge(0.8)) +
  geom_dotplot(aes(fill=sev_WHO), 
               binaxis='y', stackdir='center', 
               dotsize=1, position=position_dodge(0.8)) +
  scale_fill_discrete(name="WHO criteria",
                      #breaks=c("with", "without"),
                      labels=c("non-severe", "severe")) +
  xlab("Categorized Age") + ylab("Mean Intensity") +
  labs(title="Intensity of Ab response", subtitle="per Age and Severity")  + theme_bw()

t <- fin %>% 
  filter(!is.na(episodio_previo)) %>% 
  ggplot(aes(edad_CAT,sample_freq, fill=episodio_previo)) +
  geom_boxplot(position=position_dodge(0.8)) +
  geom_dotplot(aes(fill=episodio_previo), 
               binaxis='y', stackdir='center', 
               dotsize=1, position=position_dodge(0.8)) +
  scale_fill_discrete(name="Previous\nEpisodes",
                      #breaks=c("with", "without"),
                      labels=c("with", "without")) +
  xlab("Categorized Age") + ylab("Number Reactive") +
  labs(title="Breadth of Ab response", subtitle="per Age and Previous Episode")  + theme_bw()

u <- fin %>% 
  filter(!is.na(episodio_previo)) %>% 
  ggplot(aes(edad_CAT,sample_mean, fill=episodio_previo)) +
  geom_boxplot(position=position_dodge(0.8)) +
  geom_dotplot(aes(fill=episodio_previo), 
               binaxis='y', stackdir='center', 
               dotsize=1, position=position_dodge(0.8)) +
  scale_fill_discrete(name="Previous\nEpisodes",
                      #breaks=c("with", "without"),
                      labels=c("with", "without")) +
  xlab("Categorized Age") + ylab("Mean Intensity") +
  labs(title="Intensity of Ab response", subtitle="per Age and Previous Episode")  + theme_bw()

Rmisc::multiplot(r,t,s,u,cols = 2)
```

### 0. Extra-Analysis

#### Shared proteins

- __MSP1__ was shared with Previous Exposure and Highly reactive antigens:
    + __S1__, N-terminal (PvMSP1~NT~), was differentially reactive.
    + __S2__, C-terminal (PvMSP1~19~), was highly reactive.

```{r, fig.align='center', fig.width=4.7, fig.height=5}
g1_id <- exp_t %>% dplyr::select(Gene.ID)      # 960
g2_id <- exp_df %>% dplyr::select(Gene.ID)     # 963
g3_id <- sev_d %>% dplyr::select(Gene.ID)     # 963

g1_idl= unlist(as.list(g1_id))
g2_idl= unlist(as.list(g2_id))
g3_idl= unlist(as.list(g3_id))

tmp <- gplots::venn(list("severe\nmalaria"=g3_idl, "with\nepisode"=g2_idl, "top reactive"=g1_idl))
```

```{r}
sg <- exp_t %>% inner_join(exp_df, by="Gene.ID") %>% select(Gene.ID)
td %>% filter(Gene.ID==sg$Gene.ID) %>% 
  inner_join(all, by="Gene.ID") %>% inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID") %>% 
  select(ID,Gene.ID,Product.Description,Gene.Name,adj.P.Val,p.order,logFC,AveExpr,con,GO.Components,GO.Processes)
```

#### Shared proteins all

- against all with previous episode
- Intensidad de reactividad por Episodios Previos __interviene__ en la intensidad por Malaria Severa:
    + __5__ antigens identified in Severe Malaria  were differentially reactive (adj.P.Val<0.05) 
    in samples with Previous Episode.

```{r, fig.align='center', fig.width=4.7, fig.height=5}
g1_id <- exp_t %>% dplyr::select(Gene.ID)      # 960
g2_id <- exp_da %>% dplyr::select(Gene.ID)     # 963
g3_id <- sev_d %>% dplyr::select(Gene.ID)     # 963

g1_idl= unlist(as.list(g1_id))
g2_idl= unlist(as.list(g2_id))
g3_idl= unlist(as.list(g3_id))

tmp <- gplots::venn(list("severe\nmalaria"=g3_idl, "with\nepisode"=g2_idl, "top reactive"=g1_idl))
```

```{r}
exp_da %>% 
  inner_join(sev_d, by="Gene.ID") %>% 
  select(Gene.ID,Product.Description.x,Gene.Name.x,
         con,sin,adj.P.Val.x,p.order.x,
         severo,no_severo,P.Value.y,adj.P.Val.y,p.order.y)
#sg <- exp_da %>% inner_join(sev_d, by="Gene.ID") %>% select(Gene.ID)
#sev_d %>% 
#  inner_join(exp_da %>% inner_join(sev_d, by="Gene.ID"),
#             by="Gene.ID")
#all %>% filter(Gene.ID==sg$Gene.ID)
#td %>% filter(Gene.ID==sg$Gene.ID) %>% 
#  inner_join(all, by="Gene.ID") %>% inner_join(lg %>% dplyr::rename(seg.num=n), by="Gene.ID")
```

#### Subset severe Ag

- __Update__ `sev_d` with `anti_join()`

```{r}
sev_d <- sev_d %>%
  anti_join(exp_da, by="Gene.ID") 

sev_d %>%
  dplyr::arrange(p.order) %>% 
  select(Gene.ID,Product.Description,Gene.Name,p.order, 
         everything(), -Description
         #severo,no_severo,P.Value,adj.P.Val,p.order
         )
```


### 7. PlasmoDB summaries

##### TMD + SP proportion

```{r, fig.height=3, fig.width=7}
a <- exp_t %>% #dplyr::count(TM.Domains)
  dplyr::mutate(TM.Domains.p= ifelse(TM.Domains>0,"present","absent"),
                SignalP= ifelse(SignalP=="SP","present","absent"),
                summary.db="Top 10") %>%  #dplyr::count(TM.Domains.p)
  select(ID,summary.db,TM.Domains.p,SignalP) %>% 
  gather(key,value,-summary.db,-ID) %>% 
  dplyr::mutate(key= fct_recode(key, "Signal Peptide (SP)"="SignalP", 
                                "Transmembral Domain (TMD)"="TM.Domains.p"))
b <- sev_d %>%  #dplyr::count(SignalP)
  dplyr::mutate(TM.Domains.p= ifelse(TM.Domains>0,"present","absent"),
                SignalP= ifelse(SignalP=="SP","present","absent"),
                summary.db="Severe (p<0.05)") %>%  #dplyr::count(TM.Domains.p)
  select(ID,summary.db,TM.Domains.p,SignalP) %>% 
  gather(key,value,-summary.db,-ID) %>% 
  dplyr::mutate(key= fct_recode(key, "Signal Peptide (SP)"="SignalP", 
                                "Transmembral Domain (TMD)"="TM.Domains.p"))
c <- exp_da %>%  #dplyr::count(SignalP)
  dplyr::mutate(TM.Domains.p= ifelse(TM.Domains>0,"present","absent"),
                SignalP= ifelse(SignalP=="SP","present","absent"),
                summary.db="w/ Episode (FDR<0.05)") %>%  #dplyr::count(TM.Domains.p)
  select(ID,summary.db,TM.Domains.p,SignalP) %>% 
  gather(key,value,-summary.db,-ID) %>% 
  dplyr::mutate(key= fct_recode(key, "Signal Peptide (SP)"="SignalP", 
                                "Transmembral Domain (TMD)"="TM.Domains.p"))
d <- exp_df %>%  #dplyr::count(SignalP)
  dplyr::mutate(TM.Domains.p= ifelse(TM.Domains>0,"present","absent"),
                SignalP= ifelse(SignalP=="SP","present","absent"),
                summary.db="w/ Episode (FDR<0.05 & logFC>1)") %>%  #dplyr::count(TM.Domains.p)
  select(ID,summary.db,TM.Domains.p,SignalP) %>% 
  gather(key,value,-summary.db,-ID) %>% 
  dplyr::mutate(key= forcats::fct_recode(key, 
                                         "Signal Peptide (SP)"="SignalP", 
                                         "Transmembral Domain (TMD)"="TM.Domains.p"))

rbind(a,b,c,d) %>% 
  replace_na(list(value = "absent")) %>% 
  ggplot(aes(x=summary.db, fill=value)) +
  geom_bar(position = "fill") +
  geom_text(data = rbind(a,b,c,d) %>% replace_na(list(value = "absent")) %>% 
              group_by(summary.db,key,value) %>% 
              dplyr::count() %>% ungroup() %>% 
              group_by(summary.db,key) %>% 
              dplyr::arrange(summary.db, key, desc(value)) %>% 
              dplyr::mutate(prop=(n/sum(n)),lab=cumsum(prop),lab= ifelse(lab==1,(2-prop)/2,lab/2)
                            ), 
            aes(x = summary.db, y= lab, label = n), size=3) +
  facet_grid(.~key,scales = "free", space = "free") +
  scale_fill_discrete(name="Value") + theme_bw() +
  #guides(fill = guide_legend(title.position = "left")) +
  theme(legend.position=c(-.33, .18),#"bottom"
        legend.margin = margin(0,0,0,0),
        axis.title.y=element_blank()
        #legend.text = element_text(size = 8),
        #legend.title = element_text(size = 8)
        ) +  
  ylab("proportion")+
  #xlab(label = "")+
  labs(title="Protein targeting and localization") +
  #labs(title="Proportion of Secretory or Membrane proteins per subset") +
  #     subtitle="") +
  #     caption="*filtered by adj.P.Val<0.05 & logFC>1")+
  # and \n (logFC>1 & con>1) | (logFC>0.9 & con>2)
  coord_flip() +
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black",size = 10#, face = "bold"
                                    ))
```

```{r, fig.height=3, fig.width=6, eval=FALSE}
##### TMD proportion
a <- exp_t %>% #dplyr::count(TM.Domains)
  dplyr::mutate(TM.Domains.p= ifelse(TM.Domains>0,"1+",0),
                summary.db="Top 10") %>%  #dplyr::count(TM.Domains.p)
  select(summary.db,TM.Domains.p)
b <- sev_d %>% #dplyr::count(TM.Domains)
  dplyr::mutate(TM.Domains.p= ifelse(TM.Domains>0,"1+",0),
                summary.db="severe p<0.05") %>%  #dplyr::count(TM.Domains.p)
  select(summary.db,TM.Domains.p)
c <- exp_da %>% #dplyr::count(TM.Domains) %>% dplyr::arrange(desc(n))
  dplyr::mutate(TM.Domains.p= ifelse(TM.Domains>0,"1+",0),
                summary.db="with episode FDR<0.05") %>%  #dplyr::count(TM.Domains.p)
  select(summary.db,TM.Domains.p)
d <- exp_df %>% #dplyr::count(TM.Domains)
  dplyr::mutate(TM.Domains.p= ifelse(TM.Domains>0,"1+",0),
                summary.db="with episode FDR<0.05 & logFC>1") %>%  #dplyr::count(TM.Domains.p)
  select(summary.db,TM.Domains.p)

rbind(a,b,c,d) %>% 
  ggplot(aes(x=summary.db, fill=TM.Domains.p)) +
  geom_bar(position = "fill") +
  scale_fill_discrete(name="TM\nDomains") +
  coord_flip() + theme_bw()
```

```{r, fig.height=3, fig.width=6, eval=FALSE}
##### SP proportion
a <- exp_t %>%  #dplyr::count(SignalP)
  dplyr::mutate(summary.db="Top 10") %>%
  select(summary.db,SignalP)
b <- sev_d %>%  #dplyr::count(SignalP)
  dplyr::mutate(summary.db="severe p<0.05") %>%
  select(summary.db,SignalP)
c <- exp_da %>%  #dplyr::count(SignalP)
  dplyr::mutate(summary.db="with episode FDR<0.05") %>%
  select(summary.db,SignalP)
d <- exp_df %>%  #dplyr::count(SignalP)
  dplyr::mutate(summary.db="with episode FDR<0.05 & logFC>1") %>%
  select(summary.db,SignalP)

rbind(a,b,c,d) %>% 
  replace_na(list(SignalP = "na")) %>% 
  ggplot(aes(x=summary.db, fill=SignalP)) +
  geom_bar(position = "fill") +
  scale_fill_discrete(name="Signal\nPeptide") +
  coord_flip() + theme_bw()
```

##### GO component proportion

```{r}
a <- exp_t %>%  #dplyr::count(GO.Components) %>% dplyr::arrange(desc(n))
  dplyr::mutate(summary.db="Top 10",
                protype= stringr::str_count(Product.Description, "hypothetical"),
                protype=as.factor(protype),
                protype= forcats::fct_recode(protype, "hypothetical"="1", "known"="0")) %>%
  select(Gene.ID,ID,Product.Description,protype,summary.db,GO.Components) #%>% dplyr::count(protype)

b <- sev_d %>%  #dplyr::count(GO.Components) %>% dplyr::arrange(desc(n))
  dplyr::mutate(summary.db="Severe Malaria (p<0.05)",
                protype= stringr::str_count(Product.Description, "hypothetical"),
                protype=as.factor(protype),
                protype= forcats::fct_recode(protype, "hypothetical"="1", "known"="0")) %>%
  select(Gene.ID,ID,Product.Description,protype,summary.db,GO.Components) #%>% dplyr::count(protype)
#exp_da %>% filter(GO.Components!="NA") %>%  dplyr::count(GO.Components) %>% dplyr::arrange(desc(n)) %>% filter(n>1)

c <- exp_da %>% select(Gene.ID,ID,Product.Description,GO.Components) %>% 
  separate(GO.Components, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,GO.Components,-Gene.ID,-ID,-Product.Description) %>% 
  filter(GO.Components!="NA") %>% #dplyr::count(GO.Components) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
  dplyr::mutate(summary.db="w/ Episode (FDR<0.05)",
                protype= stringr::str_count(Product.Description, "hypothetical"),
                protype=as.factor(protype),
                protype= forcats::fct_recode(protype, "hypothetical"="1", "known"="0")) %>%
  select(Gene.ID,ID,Product.Description,protype,summary.db,GO.Components) #%>% dplyr::count(protype)
#exp_df %>%  dplyr::count(GO.Components) %>% dplyr::arrange(desc(n))

d <- exp_df %>% select(Gene.ID,ID,Product.Description,GO.Components) %>% 
  separate(GO.Components, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,GO.Components,-Gene.ID,-ID,-Product.Description) %>% 
  filter(GO.Components!="NA") %>% #dplyr::count(GO.Components) %>% dplyr::arrange(desc(n)) #%>% filter(n>1)
  dplyr::mutate(summary.db="w/ Episode (FDR<0.05 & logFC>1)",
                protype= stringr::str_count(Product.Description, "hypothetical"),
                protype=as.factor(protype),
                protype= forcats::fct_recode(protype, "hypothetical"="1", "known"="0")) %>%
  select(Gene.ID,ID,Product.Description,protype,summary.db,GO.Components) #%>% dplyr::count(protype)
```

```{r,fig.height=5,fig.width=7}
go_ord <- rbind(a,b,c,d) %>% #class()
  group_by(GO.Components,summary.db) %>% 
  dplyr::summarise(count=n()) %>% ungroup() %>% 
  dplyr::arrange(desc(count)) %>% 
  spread(summary.db,count) %>% 
  dplyr::rename(sev="Severe Malaria (p<0.05)",
                top="Top 10",
                epida="w/ Episode (FDR<0.05)",
                epidf="w/ Episode (FDR<0.05 & logFC>1)") %>% 
  dplyr::arrange(desc(epida)) %>% 
  replace_na(list(epida = 0)) %>% 
  filter(epida==0 | epida>1)

rbind(a,b,c,d) %>% 
  inner_join(go_ord,by = "GO.Components") %>% 
  dplyr::arrange(desc(epida)) %>% 
  #replace_na(list(SignalP = "na")) %>% 
  ggplot(aes(#x=summary.db#, fill=GO.Components
             x=reorder(GO.Components,epida,order = TRUE), fill=protype
             )) +
  geom_bar(#position = "fill"
           #position = "identity"
           ) +
  facet_wrap(~summary.db, scales = "free_x" # CON O SIN scala
             #space = "free"
             ) +#
  coord_flip() + theme_bw() +
  scale_fill_discrete(name="Protein",
                      labels=c("with annotation", "hypothetical")) +
  theme(legend.position=c(.86, .09),#"bottom"
        legend.margin = margin(0,0,0,0),
        axis.title.y=element_blank()
        #legend.text = element_text(size = 8),
        #legend.title = element_text(size = 8)
        ) +
  labs(title="Gene Ontology: Predicted Cellular Components") +
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black"#,size = 10#, face = "bold"
                                    ))
```

```{r,fig.height=5,fig.width=6.5}
go_ord <- rbind(full_go,all_go,
      #a %>% select(-ID),
      c %>% select(-ID),
      d %>% select(-ID)
      ) %>% #class()
  group_by(GO.Components,summary.db) %>% 
  dplyr::summarise(count=n()) %>% ungroup() %>% 
  dplyr::arrange(desc(count)) %>% 
  spread(summary.db,count) %>% 
  dplyr::rename(all="Pf/Pv500 microarray",
                #top="Top 10",
                epida="w/ Episode (FDR<0.05)",
                epidf="w/ Episode (FDR<0.05 & logFC>1)") %>% 
  dplyr::arrange(desc(all)) %>% 
  #replace_na(list(epida = 0)) %>% 
  filter(all>3)
  
rbind(full_go,all_go,
      #a %>% select(-ID),
      c %>% select(-ID),
      d %>% select(-ID)
      ) %>% 
  inner_join(go_ord,by = "GO.Components") %>% 
  dplyr::arrange(desc(all)) %>% 
  #replace_na(list(SignalP = "na")) %>% 
  ggplot(aes(#x=summary.db#, fill=GO.Components
             x=reorder(GO.Components,all,order = TRUE), fill=protype
             )) +
  geom_bar(#position = "fill"
           #position = "identity"
           ) +
  facet_wrap(~summary.db, scales = "free_x" # CON O SIN scala
             #space = "free"
             ) +#
  coord_flip() + theme_bw() +
  scale_fill_discrete(name="Protein",
                      labels=c("with annotation", "hypothetical")) +
  theme(legend.position=c(.87, .09),#"bottom"
        legend.margin = margin(0,0,0,0),
        axis.title.y=element_blank()
        #legend.text = element_text(size = 8),
        #legend.title = element_text(size = 8)
        ) +
  labs(title="Gene Ontology: Predicted Cellular Components") +
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black"#,size = 10#, face = "bold"
                                    ))
```


##### GO proportion others

```{r}
all %>% select(GO.Components,GO.Functions,GO.Processes) %>% 
  replace_na(list(GO.Components="na",
                  GO.Functions="na",
                  GO.Processes="na")
             ) %>% 
  dplyr::mutate(GO.Components=ifelse(GO.Components!="na",1,0),
                GO.Functions=ifelse(GO.Functions!="na",1,0),
                GO.Processes=ifelse(GO.Processes!="na",1,0)
                ) %>% 
  summarise_all(sum,na.rm=T) #%>% t()
```

```{r, eval=FALSE}
exp_t %>%  dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n))
sev_d %>%  dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n))
sev_d %>% select(Gene.ID,GO.Functions) %>% 
  separate(GO.Functions, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,value,-Gene.ID) %>% 
  filter(value!="NA") %>% dplyr::count(value) %>% dplyr::arrange(desc(n)) #%>% filter(n>1)
exp_da %>% filter(GO.Functions!="NA") %>%  dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
exp_da %>% select(Gene.ID,GO.Functions) %>% 
  separate(GO.Functions, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,value,-Gene.ID) %>% 
  filter(value!="NA") %>% dplyr::count(value) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
exp_df %>%  dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n))
exp_df %>% select(Gene.ID,GO.Functions) %>% 
  separate(GO.Functions, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,value,-Gene.ID) %>% 
  filter(value!="NA") %>% dplyr::count(value) %>% dplyr::arrange(desc(n)) #%>% filter(n>1)
```

```{r,fig.height=5,fig.width=12, eval=FALSE, echo=FALSE}
a <- exp_t %>%  #dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n))
  dplyr::mutate(summary.db="Top 10") %>%
  select(summary.db,GO.Functions)
b <- sev_d %>%  #dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n))
  dplyr::mutate(summary.db="severe p<0.05") %>%
  select(summary.db,GO.Functions)
#exp_da %>% filter(GO.Functions!="NA") %>%  dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
c <- exp_da %>% select(Gene.ID,GO.Functions) %>% 
  separate(GO.Functions, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,GO.Functions,-Gene.ID) %>% 
  filter(GO.Functions!="NA") %>% #dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
  dplyr::mutate(summary.db="w/episode FDR<0.05") %>%
  select(summary.db,GO.Functions)
#exp_df %>%  dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n))
d <- exp_df %>% select(Gene.ID,GO.Functions) %>% 
  separate(GO.Functions, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,GO.Functions,-Gene.ID) %>% 
  filter(GO.Functions!="NA") %>% #dplyr::count(GO.Functions) %>% dplyr::arrange(desc(n)) #%>% filter(n>1)
  dplyr::mutate(summary.db="w/episode FDR<0.05 & logFC>1") %>%
  select(summary.db,GO.Functions)

rbind(a,b,c,
      d) %>% 
  #replace_na(list(SignalP = "na")) %>% 
  ggplot(aes(#x=summary.db#, fill=GO.Functions
    x=GO.Functions#, fill=
  )) +
  geom_bar(#position = "fill"
    position = "identity") +
  facet_grid(.~summary.db,
             scales = "free"
             #space = "free"
  ) +#
  coord_flip()
```

```{r, eval=FALSE}
exp_t %>%  dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n))
sev_d %>%  dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n))
exp_da %>% filter(GO.Processes!="NA") %>%  dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
exp_da %>% select(Gene.ID,GO.Processes) %>% 
  separate(GO.Processes, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,value,-Gene.ID) %>% 
  filter(value!="NA") %>% dplyr::count(value) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
exp_df %>%  dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n))
```

```{r,fig.height=5,fig.width=12, eval=FALSE, echo=FALSE}
a <- exp_t %>%  #dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n))
  dplyr::mutate(summary.db="Top 10") %>%
  select(summary.db,GO.Processes)
b <- sev_d %>%  #dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n))
  dplyr::mutate(summary.db="severe p<0.05") %>%
  select(summary.db,GO.Processes)
#exp_da %>% filter(GO.Processes!="NA") %>%  dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
c <- exp_da %>% select(Gene.ID,GO.Processes) %>% 
  separate(GO.Processes, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,GO.Processes,-Gene.ID) %>% 
  filter(GO.Processes!="NA") %>% #dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n)) %>% filter(n>1)
  dplyr::mutate(summary.db="w/episode FDR<0.05") %>%
  select(summary.db,GO.Processes)
#exp_df %>%  dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n))
d <- exp_df %>% select(Gene.ID,GO.Processes) %>% 
  separate(GO.Processes, c("c1","c2","c3"),sep = ",") %>% 
  mutate_at(vars(c1:c3), funs(trimws)) %>% 
  gather(component,GO.Processes,-Gene.ID) %>% 
  filter(GO.Processes!="NA") %>% #dplyr::count(GO.Processes) %>% dplyr::arrange(desc(n)) #%>% filter(n>1)
  dplyr::mutate(summary.db="w/episode FDR<0.05 & logFC>1") %>%
  select(summary.db,GO.Processes)

rbind(a,b,c,
      d) %>% 
  #replace_na(list(SignalP = "na")) %>% 
  ggplot(aes(#x=summary.db#, fill=GO.Processes
    x=GO.Processes#, fill=
  )) +
  geom_bar(#position = "fill"
    position = "identity") +
  facet_grid(.~summary.db,
             scales = "free"
             #space = "free"
  ) +#
  coord_flip()
```


##### HYPOTHETICAL proportion

```{r, eval=FALSE}
exp_t %>%  #dplyr::count(Product.Description) %>% dplyr::arrange(desc(n)) %>% 
  filter(stringr::str_detect(Product.Description, "hypothetical")) %>% 
  select(ID,Gene.ID,Product.Description,GO.Components)
sev_d %>%  #dplyr::count(Product.Description) %>% dplyr::arrange(desc(n)) %>% 
  filter(stringr::str_detect(Product.Description, "hypothetical")) %>% 
  select(ID,Gene.ID,Product.Description,P.Value,p.order,GO.Components)
```

- Among highly reactive antigens (n=10), __03__ were hypothetical
- Among differentially reactive antigens in:
    + severe malaria (n=10), __02__ were hypothetical.
    + previous exposure (n=140), __46__ were hypothetical. (table below)

```{r}
exp_da %>%  #dplyr::count(Product.Description) %>% dplyr::arrange(desc(n)) %>% 
  filter(stringr::str_detect(Product.Description, "hypothetical")) %>% 
  select(ID,Gene.ID,Product.Description,adj.P.Val,p.order,GO.Components)
```

##### KNOWN proportion

- Using Product Descriptions:
    + In previous exposure (adj.P.Val < 0.05):
        + __9__ antigens with known gene names were differentially reactive.
        + __5__ antigens belong to __MSP7__.
    + In previous exposure (adj.P.Val < 0.05, logFC > 1):
        + __14__ antigens were differentially reactive.
        + __2__ antigens belong to __MSP4__.

```{r}
exp_da %>%  dplyr::count(Product.Description) %>% dplyr::arrange(desc(n)) %>% 
  filter(stringr::str_detect(Product.Description, "\\(.{2,}\\)"))
#exp_df %>%  dplyr::count(Product.Description) %>% dplyr::arrange(desc(n))

#exp_da %>%  dplyr::count(Product.Description) %>% dplyr::arrange(desc(n)) %>% 
#  filter(stringr::str_detect(Product.Description, "MSP7"))
exp_da %>%  #dplyr::count(Product.Description) %>% dplyr::arrange(desc(n)) %>% 
  filter(stringr::str_detect(Product.Description, "MSP7")) %>% 
  select(ID,Gene.ID,Product.Description,adj.P.Val,p.order)
```

```{r, eval=FALSE}
exp_t %>%  dplyr::count(Gene.Name) %>% dplyr::arrange(desc(n))
sev_d %>%  dplyr::count(Gene.Name) %>% dplyr::arrange(desc(n))
exp_da %>%  dplyr::count(Gene.Name) %>% dplyr::arrange(desc(n))
exp_df %>%  dplyr::count(Gene.Name) %>% dplyr::arrange(desc(n))
```

##### ORTHOLOGS count

```{r, eval=FALSE}
exp_t %>% dplyr::mutate(Ortholog.count=as.numeric(Ortholog.count)) %>% 
  dplyr::count(Ortholog.count) %>% dplyr::arrange(Ortholog.count)
sev_d %>% dplyr::mutate(Ortholog.count=as.numeric(Ortholog.count)) %>% 
  dplyr::count(Ortholog.count) %>% dplyr::arrange(Ortholog.count)
exp_da %>% dplyr::mutate(Ortholog.count=as.numeric(Ortholog.count)) %>% 
  dplyr::count(Ortholog.count) %>% dplyr::arrange(Ortholog.count)
exp_df %>% dplyr::mutate(Ortholog.count=as.numeric(Ortholog.count)) %>% 
  dplyr::count(Ortholog.count) %>% dplyr::arrange(Ortholog.count)
```

- __PVX_082685 (MSP7)__ and __PVX_097730 (hypothetical protein, conserved)__ 
were differentially reactive in samples with previous exposure and have only __two orthologs__: 
_P. vivax_ and _P. cynomolgi_.
    + __MSP7__ had an adj.P.Val<0.05 and a Case mean intensity > 1,
    + __PVX_097730__ had an adj.P.Val<0.05, logFC>0.97, and Case mean intensity > 2.8

```{r}
exp_da %>% dplyr::mutate(Ortholog.count=as.numeric(Ortholog.count)) %>% 
  filter(Ortholog.count==2) %>% 
  select(ID,Gene.ID,Product.Description,adj.P.Val,p.order,logFC,con,Ortholog.count)
#exp_df %>% dplyr::mutate(Ortholog.count=as.numeric(Ortholog.count)) %>% 
#  filter(Ortholog.count==2)
```

##### SNP summary

```{r, fig.width=10, fig.height=5}
a <- exp_t %>% 
  dplyr::mutate(summary.db="Top 10") %>%  #dplyr::count(TM.Domains.p)
  select(ID, summary.db, Gene.ID,Gene.Name,
         NonSyn_Syn.SNP.Ratio.All.Strains,
         Total.SNPs.All.Strains,
         Transcript.Length,
         Ortholog.count,AveExpr) %>% 
  dplyr::rename(intensity=AveExpr)

b <- exp_df %>% 
  dplyr::mutate(summary.db="w/ Previous Episode (FDR<0.05 & logFC>1)") %>%
  select(ID, summary.db, Gene.ID,Gene.Name,
         NonSyn_Syn.SNP.Ratio.All.Strains,
         Total.SNPs.All.Strains,
         Transcript.Length,
         Ortholog.count,con) %>% 
  dplyr::rename(intensity=con)

rbind(a,b) %>% 
  ggplot(aes(log2(NonSyn_Syn.SNP.Ratio.All.Strains),
             Total.SNPs.All.Strains/Transcript.Length)) +
  geom_point(aes(colour=Transcript.Length, size=Ortholog.count)) + 
  #scale_colour_gradientn(colours = rainbow(10),
  #                       name="Transcript\nlength") +
  viridis::scale_color_viridis(name="Transcript\nlength(nt)",
                               trans="log10",
                               breaks = c(1000,2000,3000,4000,5000,10000)
                               #,option="plasma"#,discrete = TRUE
                               ) +
  scale_size(name="Ortholog\ncount"#,
             #trans="log10"#, breaks = c(1000,2000,3000,5000,10000)
             ) +
  #scale_colour_gradient(trans="log10") +
  ggrepel::geom_text_repel(data = rbind(a,b) %>% 
                             replace_na(list(Gene.Name="na")) %>% 
                             dplyr::mutate(Gene.Name=ifelse(Gene.Name=="na",Gene.ID,Gene.Name),
                                           Gene.Name=ifelse(ID=="PVX_099980_1o1_S2.504","MSP1_S2",Gene.Name),
                                           Gene.Name=ifelse(ID=="PVX_099980_1o1_S1.793","MSP1_S1",Gene.Name))
                             #mutate(Gene.Name=ifelse(NonSyn_Syn.SNP.Ratio.All.Strains < 1 | 
                                                       #NonSyn_Syn.SNP.Ratio.All.Strains > 2.5 | 
                              #                         Total.SNPs.All.Strains > 300 & 
                              #                         Total.SNPs.All.Strains < 400 | 
                              #                         Total.SNPs.All.Strains > 100 & 
                              #                         Total.SNPs.All.Strains < 200 |
                              #                         Total.SNPs.All.Strains < 10,
                              #                       Gene.ID,Gene.Name))
                           ,aes(label=Gene.Name,size=intensity*5
                                ),
                           #size=2.8,
                           box.padding = unit(.65, "lines"),segment.size =.2,
                           direction = "x",#nudge_x = 45,
                           point.padding = unit(.5, "lines")#,
                           #segment.color = "red"
                           ) +
  facet_grid(.~summary.db#, 
             #scales = "free", 
             #space = "free"
             ) + theme_bw() +
  theme(#legend.position=c(-.25, -.025),#"bottom"
        legend.margin = margin(0,0,0,0)#,
        #legend.text = element_text(size = 8),
        #legend.title = element_text(size = 8)
        ) +  
  coord_trans(y = "log10") +
  #geom_vline(aes(xintercept=0), lty=3, col="red") #+ 
  ylab("Total SNPs (length-normalized)")+
  xlab("NonSyn/Syn SNP ratio (log2-transformed)")+
  labs(title="Genetic variation among all strains per subset",#SNP space
       #subtitle="Among all strains",
       caption="*label size in d.p. to Case mean intensity")+
  # and \n (logFC>1 & con>1) | (logFC>0.9 & con>2)
  #coord_flip() # ORDENAR eje de proteínas!!!!
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black",size = 12#, face = "bold"
                                    ))
```

- PVX_097730, a pesar de no poseer un nivel de expresión tan alto como MSP1_S1, 
tiene una reactividad serológica mayor que MSP1_S1 y un tamaño de efecto igual a 0.9.
Además, es exclusivo de P. vivax.

```{r, fig.width=8, fig.height=3,eval=FALSE}
#exp_t
a <- exp_t %>% select(Gene.ID,NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains,Transcript.Length) %>% 
  ggplot(aes(NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains)) +
  geom_point(aes(colour=Transcript.Length)) + viridis::scale_color_viridis(name="Transcript\nlength") +
  ggrepel::geom_text_repel(data = exp_t %>% 
                             mutate(Gene.Name=ifelse(NonSyn_Syn.SNP.Ratio.All.Strains<1,
                                                     Gene.ID,Gene.Name))
                           ,aes(label=Gene.Name),
                           box.padding = unit(.11, "lines"),
                           #direction = "y",
                           point.padding = unit(.5, "lines"),
                           segment.color = "red"
                           ) +
  labs(title="SNP space",subtitle="Top 10") + theme_bw()
b <- sev_d %>% select(Gene.ID,NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains) %>% 
  ggplot(aes(NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains)) +
  geom_point() + ggrepel::geom_text_repel(data = sev_d,aes(label=Gene.Name)) +
  labs(title="SNP space",subtitle="Severe Malaria") + theme_bw()
#exp_da %>% select(Gene.ID,NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains) %>% 
#  ggplot(aes(NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains)) +
#  geom_point()
c <- exp_df %>% select(Gene.ID,NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains,Transcript.Length) %>%
  ggplot(aes(NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains)) +
  geom_point(aes(colour=Transcript.Length)) + viridis::scale_color_viridis(name="Transcript\nlength") +
  ggrepel::geom_text_repel(data = exp_df,
                           aes(label=Gene.Name),
                           box.padding = unit(.15, "lines"),
                           point.padding = unit(.5, "lines"),
                           segment.color = "red") +
  labs(title="SNP space",subtitle="with Previous Episode") + theme_bw()

Rmisc::multiplot(a,#b,
                 c,cols = 2)
```

- Subset of __differentially reactive antigens__ in samples __with previous episodes__ having:
    + __total SNPs__ lower than 100
    + __NS/S SNP ratio__ lower than 2.5

```{r}
exp_df %>% filter(Total.SNPs.All.Strains<100,NonSyn_Syn.SNP.Ratio.All.Strains<2.5) %>% 
  dplyr::arrange(NonSyn_Syn.SNP.Ratio.All.Strains) %>% 
  select(NonSyn_Syn.SNP.Ratio.All.Strains,Total.SNPs.All.Strains,everything())
```

```{r, eval=FALSE, echo=FALSE}
exp_t
#sev_t

sev_d

exp_da
#exp_dc
exp_df
```

##### (**) EXPRESSION pattern

- __Load RNA-seq profile__ from Zhu et al.
    - __Transcription profile of intraerythrocytic cycle (Zhu et al. 2016) per subset__

```{r, message=FALSE}
rna <- readr::read_tsv("data/04-listgit.tsv") %>% #colnames() %>% #class()
  #make.names(unique = TRUE) %>% 
  rename_all(
      funs(
        #stringr::str_to_lower(.) %>%
        stringr::str_replace_all(., '\\s', '\\.') %>% 
        stringr::str_replace_all(., '\\[|\\]', '') %>% 
        stringr::str_replace_all(., '\\#.', '') %>% 
        stringr::str_replace_all(., 'Computed.', '') %>% 
        stringr::str_replace_all(., '\\/', '\\_') %>% 
        stringr::str_replace_all(., '(smru\\d..-.smru)', 's') %>% 
        stringr::str_replace_all(., '.hours', 'h')
      )
  ) %>% 
  select(-source_id,-X33#-X18
         ) %>% 
  dplyr::rename(Gene.Name=Gene.Name.or.Symbol) %>% 
  dplyr::mutate_all(
    funs(
      stringr::str_replace_all(.,"N/A",replacement = NA_character_)
    )) %>% 
  dplyr::mutate(SignalP.Scores=stringr::str_replace(SignalP.Scores, "^NN.{0,}","SP")) %>% 
  dplyr::mutate(NonSyn_Syn.SNP.Ratio.All.Strains=as.numeric(NonSyn_Syn.SNP.Ratio.All.Strains),
                Total.SNPs.All.Strains=as.numeric(Total.SNPs.All.Strains),
                Transcript.Length=as.numeric(Transcript.Length),
                Ortholog.count=as.numeric(Ortholog.count)) %>% 
  #dplyr::mutate(SignalP=stringr::str_replace(SignalP.Scores, "^NN.{0,}","SP")) %>% 
  #group_by(SignalP.Scores,SignalP) %>% dplyr::count()
  dplyr::rename(SignalP=SignalP.Scores) %>% 
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_003775", "MSP4", Gene.Name)) %>%
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_003770", "MSP5", Gene.Name)) %>% 
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_097625", "MSP8", Gene.Name)) %>% 
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_114145", "MSP10", Gene.Name)) %>% 
  dplyr::mutate(Gene.Name= ifelse(Gene.ID == "PVX_116780", "SFT2", Gene.Name)) %>% 
  select(-(Ortholog.count:Transcripts))
#lg
tail(rna)
#raw %>% filter(Gene.ID=="PVX_003775")#only_2o2 --> IS THIS SUGGESTING A MISTAKE IN ID TIPYING? --> COMPARE!
#raw %>% filter(stringr::str_detect(Description, "MSP7"))
```

```{r,fig.height=5.3,fig.width=10}
rna_dt <- rbind(
  exp_df %>% 
    mutate(subset="diff") %>% 
    select(ID,Gene.ID,subset,con,logFC) %>% #
    dplyr::rename(intensity=con),
  exp_t %>% 
    mutate(subset="atop") %>% 
    select(ID,Gene.ID,subset,AveExpr,logFC) %>% #,logFC
    dplyr::rename(intensity=AveExpr)
  ) %>% 
  inner_join(rna,by = "Gene.ID") %>% 
  gather(rna,fpkm,-(ID:Gene.Name)) %>% 
  separate(rna,c("rna_s","rna_t")) %>% #dplyr::count(rna_t)
  dplyr::mutate(rna_t=stringr::str_replace(rna_t,"h",""),
                rna_t=as.numeric(rna_t),
                fpkm=as.numeric(fpkm)) %>% 
  dplyr::mutate(fpkm=log2(fpkm)) %>% filter(fpkm!="-Inf") %>% 
  filter(rna_s=="s1") %>% 
  unite(id.name, Gene.Name, Gene.ID, sep = " / ", remove = FALSE) %>%
  #unite(id.name, id.name, p.order, sep = " / ", remove = FALSE)  %>% 
  #dplyr::mutate(id.name= ifelse(ID == "PVX_003775_2o2.150", "MSP4. / PVX_003775", id.name))
  dplyr::mutate(subset= forcats::fct_recode(subset, 
                                            "w/ Previous Episode (FDR<0.05 & logFC>1)"="diff", 
                                            "Top 10"="atop"))

rna_dt %>% 
  ggplot(aes(rna_t,fpkm,colour=id.name)) +
  #geom_point() + 
  geom_point() + theme_bw() +
  geom_smooth(method = "lm", se = FALSE) +
  #viridis::scale_color_viridis(discrete = T,) +
  coord_cartesian(xlim = c(min(rna_dt$rna_t), max(rna_dt$rna_t) + 13)) +
  facet_grid(.~subset) +
  ggrepel::geom_text_repel(data=subset(rna_dt, rna_t == max(rna_t)),
                           aes(label=id.name),
                           size = 3,fontface = 'bold',
                           nudge_x = 45,#direction = "x",
                           box.padding = unit(.08, "lines"),
                           segment.color = NA
                           ) +
  scale_x_continuous(breaks=seq(6, 48, by = 6)) +
  theme(legend.position = "none") +
  xlab("Time (hours)") +
  ylab("log2 FPKM") +
  labs(title="Transcription profile of intraerythrocytic cycle (Zhu et al. 2016) per subset") +
  #coord_trans(y = "log10")
  theme(strip.background = element_rect(colour = "black"#, fill = "white"
                                        ),
        strip.text.x = element_text(colour = "black",size = 12#, face = "bold"
                                    ))
```

```{r}
rna_dtm.ann <- rna_dt %>% 
  dplyr::mutate(id.name= ifelse(ID == "PVX_003775_2o2.150", "MSP4. / PVX_003775", id.name),
                id.name=ifelse(ID=="PVX_099980_1o1_S2.504","MSP1_S2 / PVX_099980",id.name),
                id.name=ifelse(ID=="PVX_099980_1o1_S1.793","MSP1_S1 / PVX_099980",id.name)) %>% 
  separate(id.name,c("gene.name","gene.id"),sep = " / ") %>% 
  unite(id.name,gene.id,gene.name,sep = " / ") %>% 
  #mutate_at(vars(id.name), funs(trimws)) %>% 
  #filter(rna_s=="s1") %>% 
  #filter(subset=="diff") %>%
  select(ID,id.name,subset,rna_t,fpkm,intensity,logFC) #%>% 
  #dplyr::mutate(fpkm=log2(fpkm)) %>%
  #filter(fpkm!="-Inf") 
  
rna_dtm <- rna_dtm.ann  %>% 
  select(-subset) %>% 
  reshape2::acast(id.name ~ rna_t,
                  value.var = "fpkm") #%>% class()
```

```{r,fig.height=5,fig.width=7.7}
x.ann <- rna_dtm.ann %>%
  group_by(id.name) %>% 
  dplyr::mutate(fpkm.max=max(fpkm,na.rm = T),
                fpkm.min=min(fpkm,na.rm = T),
                fold.change=fpkm.max-fpkm.min) %>% 
  ungroup() %>% 
  filter(fpkm==fpkm.max) %>% 
  dplyr::arrange(subset,rna_t,fpkm.max) #fold.change

x <- x.ann %>% 
  select(id.name) %>% as.matrix() 

aheatmap(rna_dtm[x,], Rowv = NA, Colv = NA,
         annRow = x.ann %>% 
           select(subset,intensity,logFC,rna_t) %>% 
           dplyr::rename("time of maximum expression (FPKM)"=rna_t)
         )

aheatmap(rna_dtm[x,],Rowv = FALSE, Colv = NA,
         annRow = x.ann %>% 
           select(subset,intensity,logFC,rna_t) %>% 
           dplyr::rename("time of maximum expression (FPKM)"=rna_t)
         )
```

##### (**) GENERAL summary

- count/proportion of immune reactive Hypothetical proteins
- count/proportion of Differentially Reactive proteins
- count/proportion of R_square higher than 0.9
- count/proportion of Unique proteins in P vivax / Pv &PcyB / no Pf homolog
- count/proportion + list of Reactive proteins with more than gonw reactive polypeptide antigen

### 0. N6 subset

- __OBJETIVO:__
    + __Comparar__ lista de proteínas de interés para el Dept.Parasitología NAMRU-6

- __PROTOCOLO:__
    + __Leer__ data
        + Problema: Nombre de columna es observación
        + Limpieza: 
            - __Agregar__ nombre de columna como fila, y
            - __Renombrar__ columna a `Gene.ID`
    + __Unir__ dos bases de datos, conservando solo las observaciones (filas) en común.
        - Lista de Ag de __interés para N6__ con:
            - Lista de Ag reactivos __no presentes__ en el microarreglo,
            - Lista de Ag con mayor reactividad en toda la muestra __(Top10)__,
            - Lista de Ag dif. reactivos en __Malaria Severa (P.Value<0.05)__,
            - Lista de Ag dif. reactivos en pacientes __CON episodio previo (adj.P.Val<0.05)__, y
            - Lista de Ag dif. reactivos en pacientes __CON episodio previo (adj.P.Val<0.05 & logFC>1)__.

```{r}
a <- readxl::read_xlsx("data/00-protn6c.xlsx") %>% colnames()
prt_n6 <- readxl::read_xlsx("data/00-protn6c.xlsx") %>% 
  add_row(PVX_119355=a) %>% 
  dplyr::rename(Gene.ID=PVX_119355)
```

- __RESULTADOS:__
    + De un total de __`r dim(prt_n6)[1]`__ Ag en lista de __N6__:
        - __07__ son Ag reactivos __no presentes__ en el microarreglo,
        - __01__ es Ag con mayor reactividad en toda la muestra __(Top10)__,
        - __00__ son Ag dif. reactivos en __Malaria Severa (P.Value<0.05)__,
        - __05__ son Ag dif. reactivos en pacientes __CON episodio previo (adj.P.Val<0.05)__, y
        - __03__ son Ag dif. reactivos en pacientes __CON episodio previo (adj.P.Val<0.05 & logFC>1)__.

- __EXPLORAR:__
    + Explorar tablas con:
        + __Resultados__ de la comparación entre grupos, e
        + __Información__ molecular extraida de PlasmoDB

+ __07 AG REACTIVOS NO PRESENTES EN MICROARREGLO__
    - Relevancia:
        + 07 de 38 proteínas de interés presentan __evidencia de reactividad__ 
        ante la respuesta humoral contra _Plasmodium vivax_.

```{r}
np %>% inner_join(prt_n6,by="Gene.ID") #%>% #select(ID,Gene.ID,Gene.Name,Product.Description,Description)
```

+ __01 AG MÁS REACTIVOS__
    - NOTA: MSP1 __Segmento 2__

```{r}
exp_t %>% inner_join(prt_n6,by="Gene.ID") %>% select(-Description)
```

+ __00 AG EN MALARIA SEVERA__

```{r}
sev_d %>% inner_join(prt_n6,by="Gene.ID") %>% select(ID,Gene.ID,Product.Description,Gene.Name,everything())
```

+ __05 AG EN 140 DIF REACTIVOS CON EPISODIO PREVIO__
    - NOTA: 
        + MSP1 __Segmento 1__
        + Dos variantes de 6-cysteine protein
            - PVX_113775: __P12__
            - PVX_000995: __P41__ (antes llamada _transmission-blocking target antigen Pfs230, putative_)
    - INESPERADO: __StAR (PVX_081550)__ sí está en microarreglo.

```{r}
exp_da %>% inner_join(prt_n6,by="Gene.ID") %>% select(ID,Gene.ID,Product.Description,Gene.Name,everything())
```

+ __03 AG EN 15 DIF REACTIVOS CON EPISODIO PREVIO CON DUPLICACIÓN SOBRE CONTROL__

```{r}
exp_df %>% inner_join(prt_n6,by="Gene.ID") %>% select(ID,Gene.ID,Product.Description,Gene.Name,everything())
```

#### yapa

- __YAPA:__
    + Unir lista de microarreglos con Ag reportados como "reactivos no presentes" por ADi.

- __RESULTADOS__
    + De un total de 151, __54 Ag SÍ están presentes__ en el microarreglos
    + De un total de __`r dim(prt_n6)[1]`__ Ag en lista de __N6__:
        - __05__ son Ag reactivos que __SÍ están presentes__ en el microarreglo

```{r}
yp <- raw %>% inner_join(np,by="Gene.ID") %>% #filter(Gene.ID=="PVX_081550")
  select(ID,Gene.ID,Description) %>% 
  inner_join(all %>% select(Gene.ID,Product.Description,Gene.Name),by="Gene.ID") %>% 
  select(ID,Gene.ID,Gene.Name,everything())

yp
```

+ __05 AG REACTIVOS DECLARADOS COMO "NO PRESENTES EN MICROARREGLO", ¡SÍ LO ESTÁN!__
    - CelTOS, GEST, CSP, ICP.

```{r}
yp %>% inner_join(prt_n6,by="Gene.ID") %>% select(ID,Gene.ID,Gene.Name,Product.Description,Description)
```

### PAPER STRUCTURE

#### Introduction

#### Methods

##### Ethics statement

##### Study participants and sample collection

##### Protein microarray

##### Data analysis

#### Results

##### Study population characteristics

##### Characterization of P.vivax proteins on microarray

##### Antibody reactivity induced by Severe Vivax Malaria

##### Antibody profile associated with previous episode to P. vivax malaria

#### Discusion

#### Conclusions

#### Recommendations

#### Appendix

##### Supporting information

##### Acknowledgments

##### Author contributions

### Computer environment

```{r}
devtools::session_info()
```


### References