---
title: "Perfil de anticuerpos en respuesta a malaria vivax"
author: "Andree Valle Campos"
date: '`r Sys.Date()`'
output:
  html_document:
  #pdf_document:
  #html_notebook:
    toc: yes
    toc_depth: 4
    toc_float:
      collapsed: yes
    #theme: united
    code_folding: "hide"
    #fig_caption: TRUE
    #number_sections: TRUE
bibliography: malaria.bib
csl: american-medical-association.csl
---

<!-- last modification: 17nov2016 -->

# Setup 

This report is writen in **R** [@R] within a `Rmarkdown` [@rmarkdown] notebook using the `RStudio` [@rstudio] IDE software, and employing the `knitr` [@knitr] and `Hmisc` [@Hmisc] packages for the `html` setup.

This *dynamic document* integrates **text**, **code** and **results**. By "clicking" on the tab `Code` at the right of each result, the **code** will be shown. An option to download the whole *script* that generates this document can be activated upon request.

For more information about this dynamic reporting style, I recommend a recent 2016 review in spanish about **Reproducible science: what, why, how** [@CienciaReproducible2016].

```{r, results='hide', message=FALSE}
require(Hmisc)
require(plotly)
options(#grType='plotly',  # for certain graphics functions
        width = 110) # to expand the limits of CONSOLE output
mu <- markupSpecs$html   # markupSpecs is in Hmisc

# The following hidden command (<code>r mu$widescreen()</code>), causes the html notebook to use an entire wide screen.
#mu$widescreen()
```
`r mu$widescreen()`
 

***

# Introduction

# Hypothesis
*sección escrita el 17jun2016*

**Título de tesis**

Estudio del repertorio de anticuerpos en respuesta a la infección con malaria vivax en pacientes con perfil clínico severo y no complicado de la ciudad de Iquitos (Loreto-Perú), mediante un enfoque inmunómico y bioinformático.

**Hipótesis general**

Los pacientes diagnosticados con malaria vivax no complicada (no-severa) poseen un conjunto de anticuerpos que les brinda protección contra el perfil clínico severo de malaria vivax.

**Hipótesis principal**

* P1. Los pacientes con malaria vivax no-severa poseen anticuerpos diferencialmente reactivos contra antígenos de P. vivax en comparación a los pacientes severos
    + **Objetivo:** Comparar la respuesta (perfil) de anticuerpos entre pacientes severos y no-severos de acuerdo a su reactividad, amplitud/breadth y magnitud/intensidad.

* P3. Los antígenos con inmunoreactividad diferenciada pertenecen en su mayoría al estadio eritrocítico del parásito, localizados en la superficie integrados a membrana, e involucrados en la función de adhesión o rutas de invasión parasitaria.
    + **Objetivo:** Identificar distribución estadio-específica (ciclo de vida), componente celular y función molecular de los antígenos con mayor reactividad por grupo.

**Hipótesis secundarias**

* S1. La normalización por RLM permite reducir con mayor efectividad la variabilidad técnica en microarreglos de proteínas en comparación a los métodos adaptados de microarreglos de DNA.
    + **Objetivo:** Comparar tres protocolos de normalización: Fold-Over-Control (FOC), Variance-Stabilization-Method (VSN) y Robust-Linear-Method (RLM) mediante la visualización de gráficos de control de calidad.

* S2. Pacientes con mayor probabilidad de desarrollar malaria severa cumplen con las siguientes características: (i) mayor edad, (ii) baja parasitemia, (iii) menor número de infecciones pasadas (menor exposición), (iv) mayor tiempo desde el último día de infección.
    + **Objetivo:** Determinar la variación en amplitud y magnitud de anticuerpos reactivos como consecuencia de a la edad, exposición previa, parasitemia y anemia (severa) del paciente al momento del diagnóstico.

***

*Nota de Willy*

"(...) es importante conocer mejor **cual es la forma de la relación entre niveles de anticuerpos y severidad**. La mayoría de análisis asumen que hay una *relación linear*, pero eso debería verificarse previamente, tal vez dividiendo los niveles de anticuerpos en terciles o cuartiles para ver si al menos hay *incrementos continuos* en la chance de severidad. Con eso ya se puede justificar poner ese anticuerpo como variable continua en una regresión".

***

# Methods
The present report follows several online available workflows for microarray data analysis written by [Bioconductor](https://www.bioconductor.org/help/)/R developers [@Biobase]. The required packages are cited on each step of the analysis.

## SAP


***

# Dependencies

This document has the following dependencies:
```{r, warning=FALSE, message=FALSE}
library("Biobase")     #ExpressionSet
library("genefilter")  #DataCondensation
library("limma")       #DifferentialAnalysisOfGenes
library("gplots")      #Heatmaps!
library("NMF")         #AnnotatedHeatmaps!!!

library(PerformanceAnalytics) #GRAPHICAL CORRELATIONS

library("haven")

library(beeswarm)
library(vioplot)
library(beanplot)
#library(HSAUR3)

#library("stargazer")

library(dplyr)        #mutate
library(ggplot2)      #ggplot for VolcanoPlots
library(ggrepel)      #names2dots #devtools::install_github("slowkow/ggrepel")
library(Rmisc)        #multiploting ggplots

library("htmlTable")  #HTMLtables -minimalistic- without 'ResultsAsis'
library(knitr)        #better it seems

library(tidyr)        #Gather "extend" two point data
library(dplyr)

library(broom)        #Tidy Hipothesis testing and LinearRegression outputs
library(biobroom)
```


# Microarray Data Analysis

# 1. Data Tidying

## Data input

```{r}
###### [IMPORT DATA]
raw<-read.csv("RawData.csv")
```

```{r}
#
## MAKE A MATRIX of all the RAW DATA
#
all.raw_rnames <- raw[,6]                            # assign labels in column 1 to "rnames"
all.raw_data <- data.matrix(raw[,11:ncol(raw)])      # transform column into a matrix
rownames(all.raw_data) <- all.raw_rnames                 # assign row names 
```

```{r}
## organized RAW data.set
## subsetting per spot.type
pure.raw_rnames <- raw[1:24,6]                            # assign labels in column 1 to "rnames"
pure.raw_data <- data.matrix(raw[1:24,11:ncol(raw)])      # transform column into a matrix
rownames(pure.raw_data) <- pure.raw_rnames                 # assign row names 

## dim(pure.raw_data)

ctrl.raw_rnames <- raw[25:48,6]                            # assign labels in column 1 to "rnames"
ctrl.raw_data <- data.matrix(raw[25:48,11:ncol(raw)])      # transform column into a matrix
rownames(ctrl.raw_data) <- ctrl.raw_rnames                 # assign row names 

## dim(ctrl.raw_data)

median.ctrl.raw_data<-t(as.matrix(apply(ctrl.raw_data, 2, median)))
rownames(median.ctrl.raw_data) <- c("median.ctrl")                 # assign row names 

anti.raw_rnames <- raw[49:nrow(raw),6]                            # assign labels in column 1 to "rnames"
anti.raw_data <- data.matrix(raw[49:nrow(raw),11:ncol(raw)])      # transform column into a matrix
rownames(anti.raw_data) <- anti.raw_rnames                 # assign row names 

## dim(median.ctrl.raw_data)
## dim(anti.raw_data)

########################### MATRIX OF CTRL VALUES
median.ctrl.raw_data.1<-rep(median.ctrl.raw_data[1],nrow(anti.raw_data))
median.ctrl.raw_data.mat<-matrix(median.ctrl.raw_data.1,length(median.ctrl.raw_data.1),1)

for (i in 2:ncol(median.ctrl.raw_data)) {
  median.ctrl.raw_data.mat<-cbind(median.ctrl.raw_data.mat,rep(median.ctrl.raw_data[i],nrow(anti.raw_data)))
}

## dim(median.ctrl.raw_data.mat)

#head(anti.raw_data)[,1:5]
#head(ctrl.raw_data)[,1:5]
#median.ctrl.raw_data[,1:5]
#head(median.ctrl.raw_data.mat)[,1:5]
```


## Data Transformation 
**(a.k.a. FOC normalization)**[@King2015FOC]

```{r}
### logic removal of 0 and negative values
log2.zero = function(x) {log2(ifelse(x>0, x, 1))}
log2.NA = function(x) {log2(ifelse(x>0, x, NA))}    ## REQUIRED confirmation with ADI team --> modifies data dsitribution

# [06jul2016] [not present in IqMdD] 
# to choose which operation would be better, evaluate for -Inf, zero or negative values on the ratio
xxx<-anti.raw_data/median.ctrl.raw_data.mat<0 # [06jul2016] only 4
sum(xxx)
#xxx<-anti.raw_data/median.ctrl.raw_data.mat==0 # [06jul2016] non
#sum(xxx)
#sum(is.infinite(anti.raw_data/median.ctrl.raw_data.mat),na.rm = TRUE) # non
```

```{r}
## with log2.NA do not have NaN!!-----------------------------------------------------> THIS IS USED!
log2.NA.norm.ANTIGENS<-log2.NA(anti.raw_data/median.ctrl.raw_data.mat)
## head(log2.NA.norm.ANTIGENS)


```

```{r}
# [VERIFICATION] COUNT NUMBER OF Inf or NaN (09jun2016)
#sum(is.infinite(log2.NA.norm.ANTIGENS),na.rm = TRUE) #0
#sum(is.nan(log2.NA.norm.ANTIGENS),na.rm = TRUE) #0

```

## Data Normalization: Median centering 
**(I/O) ON**

```{r}

###### ***** Median Centering *****

# ¡¡¡ SWEEP !!! --> SIMPLEST NORMALIZATION: Align all log signal to have the same median (median= zero)
# reference: https://www.google.com.pe/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwiM3Y6lsuTMAhWK5yYKHR3GC8oQFggcMAA&url=http%3A%2F%2Fwww.public.iastate.edu%2F~dnett%2Fmicroarray%2F05normalization2color.ppt&usg=AFQjCNF6AMXXJl2GjsdX6uiTS7AFOsbrWg&bvm=bv.122129774,d.eWE

median.NA = function(x) {median(x,na.rm = TRUE)} # REMOVING NA FROM median() FUNCTION

# MEDIAN CENTERING!!
sample.medians <- apply(log2.NA.norm.ANTIGENS, 2, median.NA)
normalized.log2.NA.norm.ANTIGENS <- sweep(log2.NA.norm.ANTIGENS,2,sample.medians) # MEDIAN CENTERED log2.NA.norm.ANTIGENS
#
head(normalized.log2.NA.norm.ANTIGENS)[,1:5]

```

```{r}

# **** [21jul2016] [CRITICAL modification] ****
normalized.log2.NA.norm.ANTIGENS["PVX_111175.544","LIM2017"] <- 0      ## deliberate modification. 
# REASON: value equals 21 in raw[525,"LIM2017"] and -9.333741 in log2.NA.norm.ANTIGENS["PVX_111175.544","LIM2017"], which is an extreme lower value



#### STEP 1 RESULTS ----> [30jun2016 - UPDATE]
#log2.NA.norm.ANTIGENS            ## NO median centered
#normalized.log2.NA.norm.ANTIGENS ## MEDIAN CENTERING GOES AFTER FILTERING ? --> YES!!! ---> [30jun2016] THIS SHOULD NOT BE USED ---> INSTEAD, scale by heatmap!!!

## [URGENT UPDATE] 
## [30jun2016] THIS SHOULD NOT BE USED ---> INSTEAD, scale by heatmap!!!


# THEN MODIFIED: we are not going to use median centered normalization, instead, heatmap.2 Z-SCORE is going to be used

# **** [06jul2016] [CRITICAL EVALUATION]
#normalized.log2.NA.norm.ANTIGENS <- log2.NA.norm.ANTIGENS 

##########################################################

```

***

# 2. Data Integration w/ Metadata
**Sample and Feature covariates**

## Add to an Expression Set 
**`Biobase`[@Biobase] package**

For a complete overview of this procedure, click [here](kasperdanielhansen.github.io/genbioconductor/pdf/BiocIntro_ExpressionSet_Overview.pdf).

For a single capture of the data structure, take a look at this [image](https://i.ytimg.com/vi/5EAOwLDD6Wo/maxresdefault.jpg).
```{r}
#### ::ExprS:: START ::

#### ::ExprS::PHENOTYPIC DATA:: (rows: SAMPLES, columns: SAMPLE COVARIATES)

# BETTER than read.csv:
##################### read.table [OBVIOUSLY PREFERED instead of READ.CSV]
pheno.t <- read.table("samples.csv",row.names=2,header=TRUE, sep=","
                    #, 
                    #colClasses = c(rep("factor",10),"numeric","factor","numeric","factor","factor","numeric",rep("factor",2),
                    #"factor",rep("numeric",2),"factor","numeric","factor","numeric")
)

## [16sep2016] DELIVERATE MODIFICATION
#pheno.t["LIM2017","con_exp"] <- "Con exposicion" ##BECAUSE NA interrupts group comparison

######### [19sep2016]
#head(pheno.t[,c(1:8,18:ncol(pheno.t))])
#########

#head(pheno.t)
#rownames(pheno.t)
dim(pheno.t) # total: 208 samples= 200 N6 samples + 8 ADi controls

# GET RID OF: "Control" in "Sample.Type" covariate
sample.cov<-subset(pheno.t,Sample.Type=="Sample")
dim(sample.cov)
kable(head(subset(sample.cov, Study!="Primaquine"#, select = c(Study:Pad,casecontrol:malaria_pasado_num,edad)
                  )), align = "c")

# add META-DATA to sample.covariates = pheno.Data
#metadata <- data.frame(labelDescription=                              ## COMPLETE THE MISSING COVARIATES
#                         c("GenePix result (.gpr) files",
#                           "Probed Sample ID",
#                           "ADi controls/N-6 studies",
#                           "Control/Case groups A:Case-PQ, B:Control, C:Control, D:SEVERE-1?, E:SEVERE-2?", ## any diference on Severe groups?
#                           "26slides x 8pads = 208samples",
#                           "1pad x 4arrays x 17x17spots",
#                           "2 probing days",
#                           "ADi controls/N-6 samples",
#                           "AGE in years (1):0-4, (2):5-11, (3):12-14, (4):15-+",
#                           "weight in kg",
#                           "(0):what?",                               ## define the sex!!!
#                           "parasites/uL",
#                           "(1):asdfg, (2):qwerty, (3):zxcvb",        ## define SAMPLED cities: PadreCocha, Santa Clara, San Juan
#                           "(1):2006, (2):2007, (3):2008",            ## is this right?
#                           "days x [arm dosis]",                      ## explain more
#                           "scheme in days of treatment",
#                           "Anemia x ADD WHO manual CRITERIA",                ## missing!!!
#                           
#                           "Symptomatology",
#                           "Symptoms out of 7 criterias",
#                           "Symptoms out of 10 criterias",
#                           "Exposure to infection",
#                           "Previous malaria events",
#                           "Age categories",
#                           "Age years"),
#                       row.names=names(pheno.t)) 

## create annotatedDataFrame --------------------------------------------------> can be subsetted as a Data.Frame
phenoDataN6<- new("AnnotatedDataFrame", data=sample.cov#, varMetadata=metadata
                  ) ## match columns(phenotype) with row(metadata) 
#phenoDataN6


#### [MODIFICATION] ::ExprS::FEATURE DATA:: (rows: FEATURES, columns: FEATURE COVARIATES)
# ONLY featureData
feature.cov.IVTT.AG_rnames <- raw[49:nrow(raw),6]                            # assign labels in column 1 to "rnames"
feature.cov.IVTT.AG <- raw[49:nrow(raw),1:10]      # transform column into a matrix
rownames(feature.cov.IVTT.AG) <- feature.cov.IVTT.AG_rnames                 # assign row names 
#dim(feature.cov.IVTT.AG)
#head(feature.cov.IVTT.AG)

## LOOK UP and MERGE it with row(assayData)
# tutorial pg7 last paragraph ### DO IT!!! [PENDING]
featureDataN6.IVTT.AG<- new("AnnotatedDataFrame", data=feature.cov.IVTT.AG) ## STH THAT COULD BE DONE: match ROW(feature.cov) with COLUMN(featureMeta-Data: feature.cov DESCRIPTION)
#featureDataN6.IVTT.AG

##### [MODIFIED] ::ExprS:: MERGE all in ExpressionSet

## assign the ExpressionSet
# COPIED and adapted...........................................................................#
normal.log2.raw.eSet<-ExpressionSet(assayData = normalized.log2.NA.norm.ANTIGENS,
                                    phenoData = phenoDataN6,
                                    featureData = featureDataN6.IVTT.AG)                        ## still missing FEATURE DATA (SEPARATE Pfal vs Pviv)!!!!
```

## Subset eSet 
**per Pv/Pf chip and Pq/Sev experiment**
```{r}
normal.log2.raw.eSet

##### ::ExprS:: [SOLVED] SUBSET BOTH Samples and Features in ExpressionSet / "feature.cov" with no LabelDescription/
#### with normalized data //// DUAL SUBSET ExpressionSet per SAMPLE.COVARIATE and FEATURE.COVARIATE

##
normal.log2.raw.eSet.VIVAX.PQ<-normal.log2.raw.eSet[featureData(normal.log2.raw.eSet)$Species=="Pv",phenoData(normal.log2.raw.eSet)$Study=="Primaquine"]
normal.log2.raw.eSet.VIVAX.PQ # ExpressionSet subseted by both SAMPLES and FEATURE covariates!!!!

normal.log2.raw.eSet.FALCIP.PQ<-normal.log2.raw.eSet[featureData(normal.log2.raw.eSet)$Species=="Pf",phenoData(normal.log2.raw.eSet)$Study=="Primaquine"]
normal.log2.raw.eSet.FALCIP.PQ # ExpressionSet subseted by both SAMPLES and FEATURE covariates!!!!
##
normal.log2.raw.eSet.VIVAX.SEV<-normal.log2.raw.eSet[featureData(normal.log2.raw.eSet)$Species=="Pv",phenoData(normal.log2.raw.eSet)$Study!="Primaquine"]
normal.log2.raw.eSet.VIVAX.SEV # ExpressionSet subseted by both SAMPLES and FEATURE covariates!!!!

normal.log2.raw.eSet.FALCIP.SEV<-normal.log2.raw.eSet[featureData(normal.log2.raw.eSet)$Species=="Pf",phenoData(normal.log2.raw.eSet)$Study!="Primaquine"]
normal.log2.raw.eSet.FALCIP.SEV # ExpressionSet subseted by both SAMPLES and FEATURE covariates!!!!
##


```

```{r, eval=FALSE}

##
summary(pData(normal.log2.raw.eSet.VIVAX.PQ))
#summary(pData(normal.log2.raw.eSet.FALCIP.PQ)) # the same that VIVAX
varMetadata(normal.log2.raw.eSet.VIVAX.PQ)
##
summary(pData(normal.log2.raw.eSet.VIVAX.SEV))   #///////////// [MISSING SEVERE.COVARIATES]
#summary(pData(normal.log2.raw.eSet.FALCIP.SEV)) #///////////// [MISSING SEVERE.COVARIATES]  # the same that VIVAX
varMetadata(normal.log2.raw.eSet.VIVAX.SEV)      #///////////// [MISSING SEVERE.COVARIATES]

# [21jun2016] SINGLE SUBSET ExpressionSet per SAMPLE.COVARIATE (e.g. parasite density)
## visualize data.frame summary, already inside ExpressionSet (+ metadata or LabeLDescription)
#
normal.log2.raw.eSet.VIVAX.PQ    # 515 features X 140 samples
normal.log2.raw.eSet.FALCIP.PQ   # 499 features X 140 samples
normal.log2.raw.eSet.VIVAX.SEV   # 515 features X 60 samples
normal.log2.raw.eSet.FALCIP.SEV  # 499 features X 60 samples
#

## [PAST EXAMPLE - START] NOW try with EXPRESSION-SET
namru6$Study  # subset from EXPRESSION-SET --> output: factor
pData(normal.log2.raw.eSet.VIVAX.PQ) # obtain DATA-FRAME
phenoData(namru6) # obtain ANNOTATED-DATA-FRAME

pData(namru6)$Study # subset from DATA-FRAME --> output: factor
phenoData(namru6)$Study # subset from ANNOTATED-DATA-FRAME --> output: factor

summary(pData(namru6)$Study)
table(pData(namru6)$Study)

table(pData(namru6)$Group)
## [PAST EXAMPLE - END] NOW try with EXPRESSION-SET

#### ::ExprS:: END ::

```

# 3. Data Condensation

**Feature filtering per Pv/Pf chip and Pq/Sev experiment**

**Using `genefilter`[@genefilter] package**
```{r, echo=TRUE}
#
#normal.log2.raw.eSet             # 1014 features X 200 samples   (10%= 20)
#
#normal.log2.raw.eSet.VIVAX.PQ    # 515 features X 140 samples    (10%= 14)
#normal.log2.raw.eSet.FALCIP.PQ   # 499 features X 140 samples    (10%= 14)
#normal.log2.raw.eSet.VIVAX.SEV   # 515 features X 60 samples     (10%= 6)
#normal.log2.raw.eSet.FALCIP.SEV  # 499 features X 60 samples     (10%= 6)
#
############################## [START] GENEFILTER for each DATASET (n=4)
f1 <- kOverA(20,1)                           ## expression measure above 1 in at least 20 samples
ffun <- filterfun(f1)
wh1 <- genefilter(exprs(normal.log2.raw.eSet),ffun)  # WHOLE data set
sum(wh1) # 526 -> 818 ---------------------------------------------------->>>> (ALL of this are WITHOUT mean centering)
#
# GENERATE the NEW ExpressionSet
normal.log2.raw.eSet.FILTER<-normal.log2.raw.eSet[wh1,]
normal.log2.raw.eSet.FILTER

############################## GENEFILTER for VIVAX PRIMAQUINE
f2 <- kOverA(14,1)                           ## expression measure above 1 in at least 14 samples
ffun2 <- filterfun(f2)
wh2 <- genefilter(exprs(normal.log2.raw.eSet.VIVAX.PQ),ffun2)  # WHOLE data set
sum(wh2) # 252 -> 397
#
# GENERATE the NEW ExpressionSet
normal.log2.raw.eSet.VIVAX.PQ.FILTER<-normal.log2.raw.eSet.VIVAX.PQ[wh2,]
normal.log2.raw.eSet.VIVAX.PQ.FILTER

############################## GENEFILTER for FALCIPARUM PRIMAQUINE
f2 <- kOverA(14,1)                           ## expression measure above 1 in at least 14 samples
ffun2 <- filterfun(f2)
wh3 <- genefilter(exprs(normal.log2.raw.eSet.FALCIP.PQ),ffun2)  # WHOLE data set
sum(wh3) # 279 -> 430
#
# GENERATE the NEW ExpressionSet
normal.log2.raw.eSet.FALCIP.PQ.FILTER<-normal.log2.raw.eSet.FALCIP.PQ[wh3,]
normal.log2.raw.eSet.FALCIP.PQ.FILTER

############################## [DONE] GENEFILTER for VIVAX SEVERE
f3 <- kOverA(6,1)                           ## expression measure above 1 in at least 6 samples
ffun3 <- filterfun(f3)
wh4 <- genefilter(exprs(normal.log2.raw.eSet.VIVAX.SEV),ffun3)  # WHOLE data set
sum(wh4) # 255 -> 394
#
# GENERATE the NEW ExpressionSet
normal.log2.raw.eSet.VIVAX.SEV.FILTER<-normal.log2.raw.eSet.VIVAX.SEV[wh4,]
normal.log2.raw.eSet.VIVAX.SEV.FILTER

############################## [DONE] GENEFILTER for FALCIPARUM SEVERE
f3 <- kOverA(6,1)                           ## expression measure above 1 in at least 6 samples
ffun3 <- filterfun(f3)
wh5 <- genefilter(exprs(normal.log2.raw.eSet.FALCIP.SEV),ffun3)  # WHOLE data set
sum(wh5) # 280 -> 419
#
# GENERATE the NEW ExpressionSet
normal.log2.raw.eSet.FALCIP.SEV.FILTER<-normal.log2.raw.eSet.FALCIP.SEV[wh5,]
normal.log2.raw.eSet.FALCIP.SEV.FILTER

############################## [END] OF GENEFILTER 
#
# results
#
#normal.log2.raw.eSet.FILTER

#normal.log2.raw.eSet.VIVAX.PQ.FILTER      # 252 /515 features X 140 samples
#normal.log2.raw.eSet.FALCIP.PQ.FILTER     # 279 /499 features X 140 samples
#normal.log2.raw.eSet.VIVAX.SEV.FILTER     # 255 /515 features X 60 samples
#normal.log2.raw.eSet.FALCIP.SEV.FILTER    # 280 /499 features X 60 samples
#
##########################################################

```

# X. QC Graphics 
raw -> log2 -> median centering -> filtered

## whole dataset
```{r, eval=FALSE, echo=FALSE}
##### [cool merge pre and post normalization - start]     [EXTRA COOL!!!!!!!!]
png("raw_norm_med_FILT_DENSITIES.png",    # create PNG for the heat map        
    width = 6000,
    height = 5000,        # DIMENSIONS of final image: 1500x1500 pixels
    res = 300,            # 300 pixels per inch (RESOLUTION)
    pointsize = 8)        # smaller font size
##

#
dev.off()
##
```

```{r, fig.height=15, fig.width=20}

colours = hsv(seq(0,1,length=200),0.6,1)                 ## CLEARIFY THIS FUNCTION

par(mfrow=c(3,4))
#par(mfcol=c(4,3))
#
plotDensities(all.raw_data, legend = FALSE, main = "N6 Raw Intensity")
plotDensities(log2.NA.norm.ANTIGENS, legend = FALSE, main = "N6 Normalized Intensity (only)")
plotDensities(normalized.log2.NA.norm.ANTIGENS, legend = FALSE, main = "N6 Normalized + Median Centering")
plotDensities(exprs(normal.log2.raw.eSet.FILTER), legend = FALSE, main = "AFTER FILTERING")
#
hist(all.raw_data,breaks=600, col = "black", main = NULL, ylim = c(0,2500))
hist(log2.NA.norm.ANTIGENS,breaks=600, col = "black", main = NULL, ylim = c(0,2500))
hist(normalized.log2.NA.norm.ANTIGENS,breaks=600, col = "black", main = NULL, ylim = c(0,2500))
hist(exprs(normal.log2.raw.eSet.FILTER),breaks=600, col = "black", main = NULL, ylim = c(0,2500))           # THIS IS IT!!
#
boxplot(all.raw_data,col=colours,pch=20, las=2, 
        ylab="Untransformed intensity")
boxplot(log2.NA.norm.ANTIGENS,col=colours,                 #JUST log2 NORMALIZATION
        pch=20, ylim=c(-3,6.3),
        las=2, 
        ylab="LOG2 intensity")
boxplot(normalized.log2.NA.norm.ANTIGENS,col=colours,      #sweep --> MEDIAN CENTERING
        pch=20, ylim=c(-3,6.3),
        las=2, 
        ylab="LOG2 intensity")                # ¿check this? SINCE output of negative values turns to NA, median gets NULL
boxplot(exprs(normal.log2.raw.eSet.FILTER),col=colours,
        pch=20, ylim=c(-3,6.3),
        las=2, 
        ylab="LOG2 intensity")                             # REQUIRED!!

##### [cool merge pre and post normalization - end]

```

## Pq & Sev
```{r, eval=FALSE, echo=FALSE}
##### [cool merge pre and post normalization - start]     [EXTRA COOL!!!!!!!!]
png("PQ_SV_FILT.png",    # create PNG for the heat map        
    width = 6000,
    height = 3000,        # DIMENSIONS of final image: 1500x1500 pixels
    res = 300,            # 300 pixels per inch (RESOLUTION)
    pointsize = 8)        # smaller font size
##

#
dev.off()
##
```

```{r, fig.height=10, fig.width=20}

# [ADDED 28jul2016]
esetSEV<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
esetSEV$Group <- factor(esetSEV$Group)
table(esetSEV$Group)
# [end 28jul2016]
esetPQ<-normal.log2.raw.eSet.VIVAX.PQ.FILTER
esetPQ$Group <- factor(esetPQ$Group)
table(esetPQ$Group)
#

color.map.SEV <- function(Group) { if (Group=="D") "#ff0000" else "#00ffff" }   # GROUP D is RED!!!!
patientcolors.SEV <- unlist(lapply(esetSEV$Group, color.map.SEV))
table(patientcolors.SEV)
color.map.PQ <- function(Group) { if (Group=="B") "#0000ff" else "#ffff00" }   # GROUP B is ¿¿what?? !!!!
patientcolors.PQ <- unlist(lapply(esetPQ$Group, color.map.PQ))
table(patientcolors.PQ)


##### [cool merge pre and post normalization - start]     [EXTRA COOL!!!!!!!!]  [PER PROJECT]

par(mfrow=c(2,4))
# PRIMAQUINE
hist(exprs(normal.log2.raw.eSet.VIVAX.PQ.FILTER),breaks=600,
     xlim=c(-2,6.3))
hist(exprs(normal.log2.raw.eSet.FALCIP.PQ.FILTER),breaks=600,
     xlim=c(-2,6.3))
# SEVERE
hist(exprs(normal.log2.raw.eSet.VIVAX.SEV.FILTER),breaks=600,
     xlim=c(-2,6.3))
hist(exprs(normal.log2.raw.eSet.FALCIP.SEV.FILTER),breaks=600,
     xlim=c(-2,6.3))
# PRIMAQUINE
boxplot(exprs(normal.log2.raw.eSet.VIVAX.PQ.FILTER),col=patientcolors.PQ,
        pch=20,ylim=c(-2,6.3),
        las=2, 
        ylab="VIVAX-PQ FILTERING")   
boxplot(exprs(normal.log2.raw.eSet.FALCIP.PQ.FILTER),col=patientcolors.PQ,
        pch=20,ylim=c(-2,6.3),
        las=2, 
        ylab="FALCIPARUM-PQ FILTERING")   
# SEVERE
boxplot(exprs(normal.log2.raw.eSet.VIVAX.SEV.FILTER),col=patientcolors.SEV,
        pch=20,ylim=c(-2,6.3),
        las=2, 
        ylab="VIVAX-SEV FILTERING")   
boxplot(exprs(normal.log2.raw.eSet.FALCIP.SEV.FILTER),col=patientcolors.SEV,
        pch=20,ylim=c(-2,6.3),
        las=2, 
        ylab="FALCIPARUM-SEV FILTERING")   

##### [cool merge pre and post normalization - end]

```

***

# 4. Descriptive Statistics
*Of sample covariates, feature covariates and microarray whole dataset*

## Sample covariates
**Using `Hmisc`[@Hmisc] package**

```{r, echo=TRUE}
#library("haven")

#PQ_main<-read_stata("Primaquine base 05apr2014-Meddly_Julio.dta") #----> CONSERVE THE LABELS / AFTER write.csv THEY ARE GONE!
SV_main<-read_stata("malaria severa 28Nov2014_Ed.dta")             #----> WITHOUT LABELS
#
SV_main$codigo <- gsub('LIM0','LIM',SV_main$codigo) #---> http://stackoverflow.com/questions/7641666/changing-text-in-a-data-frame
#
#write.csv(PQ_main, file="PQ_05apr2014-Meddly_Julio.csv")
write.csv(SV_main, file="SEVERE_28nov2014.csv")
#
tabAll <- read.table("SEVERE_28nov2014.csv", row.names=2, header=TRUE, sep=",")
##

#sevData<-read.table("SEVERA_covariates_224samples_EXTENDED_18ago2016.csv",row.names=1,header=TRUE, sep=",", 
#                        colClasses = c(rep("factor",5),"numeric","factor","numeric","numeric",rep("factor",5),"numeric","numeric",
#                                       rep("factor",5),rep("numeric",9),"factor","numeric",rep("factor",2),rep("numeric",2)))
# ABOVE replaced by BELOW "SELECTEDcovarariates"
SELECTEDcovarariates <- c("criticos","casecontrol","cerebral2","criticalnopros",
                          "edad","sexo","talla","peso",
                          "admin_uci",
                          "postracion","confusion","coma","convulsion",
                          "temperatura",
                          "glasgow","sdra","shock","anemia_severa","hto_bajo","hemog_bajo","hto","hemoglobina","creatinina","glucosa",
                          "bilitot","bilind","bildirect","parasitemia","hematies",
                          "sin_enf_cronica",
                          "malaria_pasado_num","tx_pasado",
                          "trombocitopenia","cont_plaquetas","eritro_parasit") # NOW able to ADD whatever you want!

tabSELECT <- tabAll[,SELECTEDcovarariates]
#tabSELECT <- tabAll[,colnames(sevData)]
#tabSELECT <- SV_main[,colnames(sevData)] #--------> On dplyr format
#sapply(tabSELECT, class)

###################
write.csv(tabSELECT, file="SEVERE_28nov2014_SELECTEDcovariates.csv")

sevData<-read.table("SEVERE_28nov2014_SELECTEDcovariates.csv",row.names=1,header=TRUE, sep=",", 
                        colClasses = c(rep("factor",5),"numeric","factor","numeric","numeric",rep("factor",5),"numeric","numeric",
                                       rep("factor",5),rep("numeric",9),"factor","numeric",rep("factor",2),rep("numeric",2)))
#sapply(sevData, class)
###################

#subset to SEVERE Sample Names
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
sampleSEV <- sevData[sampleNames(eset),]
#
#sapply(sampleSEV, class)
```

```{r, echo=TRUE}
# ONE WAY
#sevFunction <- function(casecontrol) {if (casecontrol == "1") "Severe" else "Non-severe"}
#sampleSEV$casecontrol <- as.factor(unlist(lapply(sampleSEV$casecontrol, sevFunction)))
# SIMPLIFIED
levels(sampleSEV$casecontrol) <- c("Non-severe","Severe") # EVIDENCE: all pregnats were 2
levels(sampleSEV$sexo) <- c("hombre","mujer") # EVIDENCE: all pregnats were 2
#levels(sampleSEV$sin_enf_cronica) <- c("1","0") # BECAUSE: here 0 means presence of cronic disease!!! ¿CON O SIN?
#

# NEW CREATED variables --> WHO CRITERIA
sampleSEV$jaundice = factor(sampleSEV$bilitot > 2.5, labels= c("0", "1")) # 3 in Quispe, 2014 (paper), 2.5 in Smith(poster)
sampleSEV$renal = factor(sampleSEV$creatinina > 3, labels= c("0", "1"))
sampleSEV$sevAnemia = factor(sampleSEV$hemoglobina < 7, labels= c("0", "1")) # 5 if Severe Anemia
sampleSEV$hipoglicemia = factor(sampleSEV$glucosa < 40, labels= c("0"))
sampleSEV$cerebralX = factor(sampleSEV$glasgow < 11, labels= c("0", "1"))
#



#
# to avoid 98/99 maintaining "NA"
PREfunction <- function(covariate) {
  if (is.na(covariate) == TRUE) {
    NA
  } else if (covariate == "1") {
      "1"
  } else {
      "0"
    }}
#
for (i in c(1,3:5,7:ncol(sampleSEV))) {
  if (class(sampleSEV[,i]) == "factor") {
    sampleSEV[,i] <- as.factor(unlist(lapply(sampleSEV[,i], PREfunction)))
    #sampleSEV[,i] <- factor(unlist(lapply(sampleSEV[,i], PREfunction)), exclude = NULL)
    }
}
#

#
levels(sampleSEV$sdra) <- c("1", "0")
  sampleSEV$sdra <- relevel(sampleSEV$sdra, ref = "0")
levels(sampleSEV$postracion) <- c("1", "0")
  sampleSEV$postracion <- relevel(sampleSEV$postracion, ref = "0")
levels(sampleSEV$trombocitopenia) <- c("1", "0")
  sampleSEV$trombocitopenia <- relevel(sampleSEV$trombocitopenia, ref = "0")
levels(sampleSEV$hipoglicemia) <- c("0", "1")
#

# WHY THIS METHOD DOES NOT WORK?
#dfr7df <- subset(sampleSEV, select = c("jaundice", "renal", "sevAnemia", "hipoglicemia", "sdra", "shock", "cerebralX"))
#sampleSEV$CumSumSevere7 = as.numeric(apply(dfr7, 1, sum, na.rm=TRUE))
dfr7 <- cbind(sampleSEV$jaundice, sampleSEV$renal, sampleSEV$sevAnemia, sampleSEV$hipoglicemia, 
              sampleSEV$sdra, sampleSEV$shock, sampleSEV$cerebralX) -1
colnames(dfr7) <- c("jaundice", "renal", "sevAnemia", "hipoglicemia", "sdra", "shock", "cerebralX")
#dfr7 <- as.data.frame(dfr7)
#dfr7df == dfr7 #TRUE
sampleSEV$CumSumSevere7 = as.numeric(apply(dfr7, 1, sum, na.rm=TRUE))
#subset(sampleSEV, select = c("jaundice", "renal", "sevAnemia", "hipoglicemia", "sdra", "shock", "cerebralX", "CumSumSevere7"))

#
dfr10 <- cbind(sampleSEV$jaundice, sampleSEV$renal, sampleSEV$sevAnemia, sampleSEV$hipoglicemia, 
               sampleSEV$sdra, sampleSEV$shock, sampleSEV$cerebralX, 
               sampleSEV$postracion, sampleSEV$coma, sampleSEV$convulsion) -1 #
colnames(dfr10) <- c("jaundice", "renal", "sevAnemia", "hipoglicemia", "sdra", "shock", "cerebralX", 
                     "postracion", "coma", "convulsion") #

sampleSEV$CumSumSevere10 = as.numeric(apply(dfr10, 1, sum, na.rm=TRUE))
#

sampleSEVx <- sampleSEV
sampleSEVx$sample <- rownames(sampleSEVx)


#
# to CHANGE 1/0 to presence/absence
ONEfunction <- function(covariate) {
  if (is.na(covariate) == TRUE) {
    NA
  } else if (covariate == "1") {
      "present"
  } else {
      "absent"
    }}

for (i in c(1,3:5,7:ncol(sampleSEV))) {
  if (class(sampleSEV[,i]) == "factor") {
    sampleSEV[,i] <- as.factor(unlist(lapply(sampleSEV[,i], ONEfunction)))
    }
}
#

levels(sampleSEV$sdra) <- c("present", "absent")
  sampleSEV$sdra <- relevel(sampleSEV$sdra, ref = "absent")
levels(sampleSEV$postracion) <- c("present", "absent")
  sampleSEV$postracion <- relevel(sampleSEV$postracion, ref = "absent")
levels(sampleSEV$trombocitopenia) <- c("present", "absent")
  sampleSEV$trombocitopenia <- relevel(sampleSEV$trombocitopenia, ref = "absent")
levels(sampleSEV$hipoglicemia) <- c("absent", "present")

#
# TO EVALUATE
#table(sampleSEV$sdra, useNA = "ifany") #like trombocitopenia
#table(sampleSEV$postracion, useNA = "ifany")
#table(sampleSEV$convulsion, useNA = "ifany") #like coma
#table(sampleSEV$hipoglicemia, useNA = "ifany")
#eset$Group <- relevel(eset$Group, ref = "0")
#
# rm(list = ls())


####### On NUMBER of previous malaria events
#sampleSEV["LIM2017",]
#sampleSEV["LIM2017","malaria_pasado_num"] # 98? -> NA
sampleSEV["LIM2017","malaria_pasado_num"] <- NA      ## deliberate modification. [coincidently, also in "PVX_111175.544", because extremely low value]
#table(sampleSEV$malaria_pasado_num)
#summary(sampleSEV$malaria_pasado_num)
sampleSEV$con_exp = factor(sampleSEV$malaria_pasado_num >= 1, labels= c("Sin exposicion", "Con exposicion"))
#summary(sampleSEV$con_exp)


#summary(sampleSEV)
```

```{r, echo=TRUE, eval=TRUE}

symptom10 <- subset(sampleSEVx, 
                  select = c("sdra", "shock", "cerebralX", "renal", "sevAnemia", 
                             "jaundice", "hipoglicemia", 
                             "postracion", "coma", "convulsion"))

#colnames(symptom10) <- c("Lesion pulmonar", "Shock circulatorio","Malaria cerebral*", "Insuficiencia renal*", "Anemia Severa*", "Ictericia*",  "Hipoglicemia*", "Postracion", "Coma", "Convulsion")

colnames(symptom10) <- c("Lesion pulmonar (SDRA)", "Shock circulatorio (PA sistolica < 80mmHg)","Malaria cerebral* (Glasgow < 11/15)", "Insuficiencia renal* (creatinina > 3mg/dL)", "Anemia Severa* (hemoglobina < 7g/dL)", "Ictericia* (bilirrubina > 2.5mg/dL)",  "Hipoglicemia* (glucosa < 40mg/dL)", "Postracion", "Coma", "Convulsion")


for (i in 1:10) {
  levels(symptom10[,i]) <- c("",colnames(symptom10)[i])
  symptom10[,i] <- as.character(symptom10[,i])
  symptom10[,i][is.na(symptom10[,i])] <- ""
}


#
#CriticalSymp5 <- mChoice(symptom10$`Lesion pulmonar`, symptom10$`Shock circulatorio`, symptom10$`Malaria cerebral*`, symptom10$`Insuficiencia renal*`, symptom10$`Anemia Severa*`, label='Critical illness criteria')
CriticalSymp5 <- mChoice(symptom10$`Lesion pulmonar (SDRA)`, symptom10$`Shock circulatorio (PA sistolica < 80mmHg)`, symptom10$`Malaria cerebral* (Glasgow < 11/15)`, symptom10$`Insuficiencia renal* (creatinina > 3mg/dL)`, symptom10$`Anemia Severa* (hemoglobina < 7g/dL)`, label='Critical illness criteria')
#summary(CriticalSymp5)

#CriticalSymp2 <- mChoice(symptom10$`Ictericia*`, symptom10$`Hipoglicemia*`, label='Non-critical illness criteria')
CriticalSymp2 <- mChoice(symptom10$`Ictericia* (bilirrubina > 2.5mg/dL)`, symptom10$`Hipoglicemia* (glucosa < 40mg/dL)`, label='Non-critical illness criteria')
#summary(CriticalSymp2)

CriticalSymp3 <- mChoice(symptom10$Postracion, #symptom10$Coma, 
                         symptom10$Convulsion, label='Central Nervous System')
#summary(CriticalSymp3)

#

#CriticalSymp7 <- mChoice(symptom10$`Lesion pulmonar`, symptom10$`Shock circulatorio`, symptom10$`Malaria cerebral*`, symptom10$`Insuficiencia renal*`, symptom10$`Anemia Severa*`, symptom10$`Ictericia*`, symptom10$`Hipoglicemia*`, label='WHO Severe criteria 7')

#CriticalSymp10 <- mChoice(symptom10$`Lesion pulmonar`, symptom10$`Shock circulatorio`, symptom10$`Malaria cerebral*`, symptom10$`Insuficiencia renal*`, symptom10$`Anemia Severa*`, symptom10$`Ictericia*`, symptom10$`Hipoglicemia*`, symptom10$Postracion, symptom10$Coma, symptom10$Convulsion, label='WHO Severe criteria 10')


```

```{r, results='asis', echo=TRUE}
#
#AGEfunction <- function(x) {
#  if (x < 18) {
#    "< 18"
#  } else if (x >= 65) {
#      ">= 65"
#  } else {
#      "18-64"
#  }}#
#AGEfunction <- function(x) {cut(x, c(0,15,25,35,45,60,85))}
AGEfunction <- function(x) {cut(x, c(0,18,40,65,Inf))}
EXPOSUREfunction <- function(x) {cut(x, c(-Inf,0,1,4,Inf))}
#
# Have upData move units from labels to separate attribute
pbc <- upData(sampleSEV,
              cont_plaquetas = cont_plaquetas / 1000,
              edad_CAT = AGEfunction(edad),
              expo_CAT = EXPOSUREfunction(malaria_pasado_num),
              
              labels = c(edad = "Edad", edad_CAT = "Edad*", expo_CAT="Exposicion*", sexo = "Sexo", sin_enf_cronica = "Sin enfermedad cronica", 
                         malaria_pasado_num = "Eventos previos de malaria", 
                         con_exp = "Exposicion previa a malaria", casecontrol = "Sintomatologia",
                         
                         hemoglobina = "Hemoglobina", creatinina= "Creatinina", glucosa= "Glucosa",
                         bilitot= "Bilirrubina total", bilind= "Bilirrubuna indirecta", 
                         bildirect= "Bilirrubina directa",
                         parasitemia= "Parasitemia circulante", eritro_parasit = "eritrocitos parasitados",
                         glasgow = "Glasgow", cont_plaquetas = "Conteo de plaquetas", 
                         
                         postracion = "-postracion", coma= "-coma", convulsion = "-convulsion",
                         
                         admin_uci = "Admision a UCI", jaundice = "Ictericia*", 
                         shock= "Shock circulatorio",
                         cerebralX = "Malaria cerebral?*", renal = "Insuficiencia renal*", 
                         sevAnemia = "Anemia Severa*",
                         sdra= "Lesion pulmonar", hipoglicemia = "Hipoglicemia*",
                         
                         tx_pasado = "¿tx_pasado?", criticalnopros = "[¿criticalnopros?]",
                         criticos = "[criticos]", cerebral2 = "[cerebral2]",
                         anemia_severa = "¿anemia severa?", trombocitopenia = "¿trombocitopenia?",
                         
                         CumSumSevere7 = "Pacientes con #/7 criterios", 
                         CumSumSevere10 = "Pacientes con #/10 criterios"),
              
              units = c(edad = "años", edad_CAT = "años", expo_CAT="numero",
                        hemoglobina = "g/dL", creatinina= "mg/dL", glucosa= "mg/dL", 
                        bilitot= "mg/dL", bilind= "mg/dL", bildirect= "mg/dL",
                        parasitemia= "par/uL", cont_plaquetas = "x1000/uL", 
                        eritro_parasit = "numero",
                        
                        shock = "PA sistolica < 80mmHg", sdra= "SDRA", cerebralX = "Glasgow < 11/15", 
                        jaundice = "bilirrubina > 2.5mg/dL", renal = "creatinina > 3mg/dL",
                        sevAnemia = "hemoglobina < 7g/dL", hipoglicemia = "glucosa < 40mg/dL",
                        
                        criticalnopros = "[¿?]", criticos = "[¿?]", cerebral2 = "[*]",
                        anemia_severa = "[¿?]", trombocitopenia = "[¿?]"),
              #moveUnits=TRUE, 
              html=TRUE)
#
#html(contents(pbc), maxlevels=10, levelType='table')
```

### 0. Without Stratification
The html method is used for the `describe` function, and the output is put in a scrollable box.  But the graphical display of the descriptives statistics that follows this is probably better.

#### Hmisc
**useful for categorical data**
```{r, fig.height=8}
d <- describe(pbc)
html(d, size=80, scroll=TRUE)
p <- plot(d, which="categorical")
p
#prList(p)
```

#### ggplots
**useful for continuous data**
```{r, message=FALSE, fig.align="center", fig.width=8, fig.height=4}
a1 <- ggplot(sampleSEV, aes(x=parasitemia)) + 
  geom_histogram(alpha=.5) + 
  labs(title="Histograma de Parasitemia") #+
  #geom_density() #+
  #scale_x_continuous(breaks=seq(0, 85, by = 5))
  #labs(x="Age", y="Count") + 

a2 <- ggplot(sampleSEV, aes(x=edad)) + 
  geom_histogram(
    #aes(y=..density..), 
                 breaks=seq(0, 85, by = 5), 
                 alpha=.5) + 
  labs(title="Histograma de Edad") +
  #geom_density() #+
  scale_x_continuous(breaks=seq(0, 85, by = 5)) #+
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  #ylim(c(0,30))
#scale_y_continuous(limits = c(0,0.05))
multiplot(a2,a1,cols = 2)
```

```{r, eval=FALSE}
ggplot(sampleSEV, aes(x=talla)) + geom_histogram(alpha=.5)
ggplot(sampleSEV, aes(x=peso)) + geom_histogram(alpha=.5)
```

#### ggplot stratification

```{r, message=FALSE, fig.align="center", fig.width=12, fig.height=8}
a <- ggplot(sampleSEV, aes(x=edad, fill=casecontrol)) + 
  geom_histogram(
    #aes(y=..density..), 
                 breaks=seq(0, 85, by = 5), 
                 alpha=.5 ,position = "identity") + 
  labs(title="Histograma de Edad") +
  #geom_density() #+
  scale_x_continuous(breaks=seq(0, 85, by = 5)) #+
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  #ylim(c(0,30))
#scale_y_continuous(limits = c(0,0.05))

b <- ggplot(sampleSEV, aes(x=edad, fill=con_exp)) + 
  geom_histogram(
    #aes(y=..density..), 
                 breaks=seq(0, 85, by = 5), 
                 alpha=.5 ,position = "identity") + 
  labs(title="Histograma de Edad") +
  #geom_density() #+
  scale_x_continuous(breaks=seq(0, 85, by = 5)) #+
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  #ylim(c(0,30))
  #scale_y_continuous(limits = c(0,0.05))


c <- ggplot(sampleSEV, aes(x=edad, fill=casecontrol)) + 
  #geom_histogram(
   # aes(y=..density..), 
    #             breaks=seq(0, 85, by = 5), position = "dodge") + 
  #labs(title="Histograma de Edad") +
  geom_density(alpha=.3) #+
  #scale_x_continuous(breaks=seq(0, 85, by = 5))
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  #ylim(c(0,30))

d <- ggplot(sampleSEV, aes(x=edad, fill=con_exp)) + 
  #geom_histogram(
    #aes(y=..density..), 
                 #breaks=seq(0, 85, by = 5), position = "dodge") + 
  #labs(title="Histograma de Edad") +
  geom_density(alpha=.3) +
  #scale_x_continuous(breaks=seq(0, 85, by = 5)) +
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  scale_y_continuous(limits = c(0,0.025))


multiplot(a,b,c,d,cols=2)
```


```{r, message=FALSE, fig.align="center", fig.width=12, fig.height=8}
a <- ggplot(sampleSEV, aes(x=parasitemia, fill=casecontrol)) + 
  geom_histogram(
    #aes(y=..density..), 
                 #breaks=seq(0, 85, by = 5), 
                 alpha=.5 ,position = "identity") + 
  labs(title="Histograma de Parasitemia") #+
  #geom_density() #+
  #scale_x_continuous(breaks=seq(0, 85, by = 5))
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  #ylim(c(0,30))

b <- ggplot(sampleSEV, aes(x=parasitemia, fill=con_exp)) + 
  geom_histogram(
    #aes(y=..density..), 
                 #breaks=seq(0, 85, by = 5), 
                 alpha=.5 ,position = "identity") + 
  labs(title="Histograma de Parasitemia") #+
  #geom_density() #+
  #scale_x_continuous(breaks=seq(0, 85, by = 5))
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  #ylim(c(0,30))


c <- ggplot(sampleSEV, aes(x=parasitemia, fill=casecontrol)) + 
  #geom_histogram(
    #aes(y=..density..), 
                 #breaks=seq(0, 85, by = 5), position = "dodge") + 
  #labs(title="Histograma de Edad") +
  geom_density(alpha=.3) #+
  #scale_x_continuous(breaks=seq(0, 85, by = 5))
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  #ylim(c(0,30))

d <- ggplot(sampleSEV, aes(x=parasitemia, fill=con_exp)) + 
  #geom_histogram(
    #aes(y=..density..), 
                 #breaks=seq(0, 85, by = 5), position = "dodge") + 
  #labs(title="Histograma de Edad") +
  geom_density(alpha=.3) +
  #scale_x_continuous(breaks=seq(0, 85, by = 5))
  #labs(x="Age", y="Count") + 
  #xlim(c(18,52)) + 
  scale_y_continuous(limits = c(0,0.00025))


multiplot(a,b,c,d,cols=2)
###
```


**Group inclusion criteria**

### I. Symptomatology: 

**Severe Malaria**

#### Clinical data
```{r, warning=FALSE, fig.height=5}
s2 <- summaryM(edad + edad_CAT + expo_CAT + sexo + sin_enf_cronica + con_exp + malaria_pasado_num #+ tx_pasado + # general
                #admin_uci + postracion + coma + convulsion + 
                 #shock + sdra + glasgow + cerebralX # clinical
              ~ casecontrol,
              data=pbc,
              #pbc$drug!="not randomized",
							overall=FALSE, test=TRUE)
plot(s2, which='categorical')
Key(0,1)
```

```{r, warning=FALSE,  fig.width=8, fig.height=4}
par(mfrow=c(1,2))
plot(s2, which='continuous')
par(mfrow=c(1,1))
```

```{r, eval=TRUE}
latex(s2, caption='Clinical data',
     exclude1=TRUE, npct='both', 
     digits=3,
     prmsd=TRUE, brmsd=TRUE, msdsize=mu$smaller2, #NOT-EVALUATE if PDF
     middle.bold=TRUE, long = TRUE,
     #legend.bottom = TRUE, #insert.bottom = TRUE, 
     what="%", 
     html = TRUE) #change here for LaTeX PDF
```

#### Laboratory data
```{r}
s3 <- summaryM(hemoglobina + creatinina + # lab critical
                glucosa + bilitot + bilind + bildirect + # lab non-critical
                parasitemia + eritro_parasit + cont_plaquetas  # covariates
              ~ casecontrol,
              data=pbc,
							overall=FALSE, test=TRUE)
#plot(s3, which='categorical')
```

```{r, warning=FALSE,  fig.width=10, fig.height=10}
par(mfrow=c(3,3))
plot(s3, which='continuous')
par(mfrow=c(1,1))
```

```{r, eval=TRUE}
latex(s3, caption='Laboratory data',
     exclude1=TRUE, npct='both', 
     digits=3,
     prmsd=TRUE, brmsd=TRUE, msdsize=mu$smaller2,
     middle.bold=TRUE, long = TRUE,
     #legend.bottom = TRUE, insert.bottom = TRUE, 
     what="%", 
     html = TRUE) #change here for LaTeX PDF
```

#### WHO criteria

According to the most recent WHO report[@WHOsevere2014], for epidemiological and research purposes, **Severe Malaria** is defined as one or more of the following, occurring in the absence of an identified alternative cause, and in the presence of asexual parasitaemia:

```{r, warning=FALSE, fig.height=9, fig.width=10}
s1xx <- summaryM(admin_uci + con_exp + # CRITICAL CARE
                 CriticalSymp3 + CriticalSymp5 + CriticalSymp2 + parasitemia +
                 CumSumSevere7 + CumSumSevere10
              ~ casecontrol,
              data=pbc,
							overall=FALSE, test=TRUE)
plot(s1xx, which='categorical')
Key(0,0)
#legend("bottomleft", legend = c("Severe", "Non-severe"), col = c("white", "black"), pch = 19) #WHY NOT WORK?
#plot(s1, which='continuous')
```

```{r, warning=FALSE}
latex(s1xx, caption='WHO criteria for severe malaria: Clinical and Laboratory data',
     exclude1=TRUE, npct='both', 
     digits=3,
     prmsd=TRUE, brmsd=TRUE, msdsize=mu$smaller2,
     middle.bold=TRUE, long = TRUE,
     #legend.bottom = TRUE, insert.bottom = TRUE, 
     what="%", 
     html = TRUE) #change here for LaTeX PDF
```

***
#### NA considered*
```{r, warning=FALSE, fig.height=9, fig.width=10}
s1 <- summaryM(cerebral2 + criticalnopros + criticos + # clasifications
                 admin_uci + con_exp +# CRITICAL CARE
                 postracion + #coma + 
                 convulsion + 
                 sdra + shock + cerebralX + renal + sevAnemia + 
                 jaundice + hipoglicemia + 
                 parasitemia + # NOT-PRACTICAL
                 CumSumSevere7 + CumSumSevere10
              ~ casecontrol,
              data=pbc,
							overall=FALSE, test=TRUE)
#plot(s1, which='categorical')
#plot(s1, which='continuous')
```

```{r, warning=FALSE}
latex(s1, caption='WHO criteria for severe malaria: Clinical and Laboratory data',
     exclude1=TRUE, npct='both', 
     digits=3,
     prmsd=TRUE, brmsd=TRUE, msdsize=mu$smaller2,
     middle.bold=TRUE, long = TRUE,
     #legend.bottom = TRUE, insert.bottom = TRUE, 
     what="%", 
     html = TRUE) #change here for LaTeX PDF
```
***


### II. Exposure: 

**Previous infections with Malaria**

#### Clinical data
```{r, eval=TRUE, warning=FALSE, fig.height=5}

table(sampleSEV$malaria_pasado_num)
summary(sampleSEV$con_exp)

s3expo <- summaryM(edad + edad_CAT + expo_CAT + sexo + sin_enf_cronica +  casecontrol + malaria_pasado_num 
                   ~ con_exp,
              data=pbc,
							overall=FALSE, test=TRUE)
plot(s3expo, which='categorical')
Key(0,0)
#par(mfrow=c(2,3))
#plot(s3age1, which='continuous')
#par(mfrow=c(1,1))
#
```

```{r, warning=FALSE,  fig.width=8, fig.height=4}
par(mfrow=c(1,2))
plot(s3expo, which='continuous')
par(mfrow=c(1,1))
```

```{r, eval=TRUE}
latex(s3expo, caption='Clinical data (exposure)',
     exclude1=TRUE, npct='both', 
     digits=3,
     prmsd=TRUE, brmsd=TRUE, msdsize=mu$smaller2, #NOT-EVALUATE if PDF
     middle.bold=TRUE, long = TRUE,
     #legend.bottom = TRUE, #insert.bottom = TRUE, 
     what="%", 
     html = TRUE) #change here for LaTeX PDF
```

#### Laboratory data
```{r}
s3expoLAB <- summaryM(hemoglobina + creatinina + # lab critical
                glucosa + bilitot + bilind + bildirect + # lab non-critical
                parasitemia + eritro_parasit + cont_plaquetas  # covariates
              ~ con_exp,
              data=pbc,
							overall=FALSE, test=TRUE)
#plot(s3, which='categorical')
```

```{r, warning=FALSE,  fig.width=10, fig.height=10}
par(mfrow=c(3,3))
plot(s3expoLAB, which='continuous')
par(mfrow=c(1,1))
```

```{r, eval=TRUE}
latex(s3expoLAB, caption='Laboratory data (exposure)',
     exclude1=TRUE, npct='both', 
     digits=3,
     prmsd=TRUE, brmsd=TRUE, msdsize=mu$smaller2,
     middle.bold=TRUE, long = TRUE,
     #legend.bottom = TRUE, insert.bottom = TRUE, 
     what="%", 
     html = TRUE) #change here for LaTeX PDF
```

#### WHO criteria

According to the most recent WHO report [@WHOsevere2014], for epidemiological and research purposes, **Severe Malaria** is defined as one or more of the following, occurring in the absence of an identified alternative cause, and in the presence of asexual parasitaemia:


```{r, warning=FALSE, fig.height=9, fig.width=10}
s1expo <- summaryM(admin_uci + casecontrol + # CRITICAL CARE
                 CriticalSymp3 + CriticalSymp5 + CriticalSymp2 + parasitemia +
                 CumSumSevere7 + CumSumSevere10
              ~ con_exp,
              data=pbc,
							overall=FALSE, test=TRUE)
plot(s1expo, which='categorical')
Key(0,0)
#plot(s1, which='continuous')
```

```{r, warning=FALSE}
latex(s1expo, caption='WHO criteria for severe malaria: Clinical and Laboratory data (exposure)',
     exclude1=TRUE, npct='both', 
     digits=3,
     prmsd=TRUE, brmsd=TRUE, msdsize=mu$smaller2,
     middle.bold=TRUE, long = TRUE,
     #legend.bottom = TRUE, insert.bottom = TRUE, 
     what="%", 
     html = TRUE) #change here for LaTeX PDF
```

***
#### NA considered*
```{r, warning=FALSE, fig.height=9, fig.width=10}
s1 <- summaryM(cerebral2 + criticalnopros + criticos + # clasifications
                 admin_uci + casecontrol +# CRITICAL CARE
                 postracion + #coma + 
                 convulsion + 
                 sdra + shock + cerebralX + renal + sevAnemia + 
                 jaundice + hipoglicemia + 
                 parasitemia + # NOT-PRACTICAL
                 CumSumSevere7 + CumSumSevere10
              ~ con_exp,
              data=pbc,
							overall=FALSE, test=TRUE)
#plot(s1, which='categorical')
#plot(s1, which='continuous')
```

```{r, warning=FALSE}
latex(s1, caption='WHO criteria for severe malaria: Clinical and Laboratory data',
     exclude1=TRUE, npct='both', 
     digits=3,
     prmsd=TRUE, brmsd=TRUE, msdsize=mu$smaller2,
     middle.bold=TRUE, long = TRUE,
     #legend.bottom = TRUE, insert.bottom = TRUE, 
     what="%", 
     html = TRUE) #change here for LaTeX PDF
```
***


### III. Useful covariates

**Symptomatology, Previous experience and Age**

An important missing variable related to **previous experience** is *time since the last infection*[@Helb2015exposure] or *travel history* (time outside of the endemic area). But, since Iquitos is a *low transmission setting* (Iquitos EIR?, PR?), lost of exposure to the infection could be more frequent even without living the endemic area (?).

According to a last review on the topic [@Davies2015Large], a colective message from *malaria proteome microarray studies* is that **naturally acquired immunity** is associated with elevated antibodies against hundreds of *blood stage antigens*. For the african context, a vaccine that boost antibody responses against a collection of these blood stage antigens may allow *children* to reach protection from symptomatic malaria at earlier age. Also  *waning immune responses* of **adults** residing in areas *undergoing elimination* could be boosted with vaccines of this kind.

In both Brazilian and Peruvian Amazonia, transmission is considered to be *hypoendemic* and of *low intensity*, and asymptomatic cases of parasitemia imply that clinical immunity may develop rapidly in low-transmission setting, in contrast to high-transmission regions where **acquired immunity** is manifested by asymptomatic parasitemia that takes years of intense seasonal or continuous transmission [@Torres2014asymptomatic].

```{r}
#pbc$sample <- rownames(pbc)
sampleSEVcovariates <- pbc[,c("casecontrol", "CumSumSevere7", "CumSumSevere10", "con_exp","expo_CAT", "malaria_pasado_num", "edad_CAT", "edad")]
#levels(sampleSEVcovariates$edad_CAT) <- c("menor a 18", "18-64", "mayor o igual a 65")
#sampleSEVcovariates
kable(sampleSEVcovariates[order(sampleSEVcovariates$CumSumSevere10, decreasing = TRUE),], align = "c")

#write.csv(sampleSEVcovariates, file = "sampleSEVcovariates.csv")

## [16sep2016] DELIVERATE MODIFICATION
sampleSEVcovariates["LIM2017","con_exp"] <- "Con exposicion" ##BECAUSE NA interrupts group comparison
sampleSEVcovariates["LIM2017","malaria_pasado_num"] <- 1 ##BECAUSE NA interrupts group comparison
sampleSEVcovariates["LIM2017","expo_CAT"] <- "(0,1]" ##BECAUSE NA interrupts group comparison

#rownames(pData(eset))==rownames(sampleSEVcovariates)
pData(normal.log2.raw.eSet.VIVAX.SEV) <- cbind(pData(normal.log2.raw.eSet.VIVAX.SEV),sampleSEVcovariates)#ncol(pData(eset))
pData(normal.log2.raw.eSet.VIVAX.SEV.FILTER) <- cbind(pData(normal.log2.raw.eSet.VIVAX.SEV.FILTER),sampleSEVcovariates)#ncol(pData(eset))
#rownames(sampleSEVcovariates)
#cat_2 <- data.frame(as.factor(matrix(sampleSEVcovariates$edad_CAT)))
#rownames(cat_2) <- rownames(sampleSEVcovariates)
#colnames(cat_2) <- "edad_CAT_2"
#pData(eset)[23] <- sampleSEVcovariates$edad_CAT
#pData(eset) <- cbind(pData(eset),cat_2)#ncol(pData(eset))

levels(normal.log2.raw.eSet.VIVAX.SEV.FILTER$con_exp) #<- c("CON","SIN") # CHANGE: D: CON exposicion, E: SIN exposicion
normal.log2.raw.eSet.VIVAX.SEV.FILTER$con_exp <- relevel(normal.log2.raw.eSet.VIVAX.SEV.FILTER$con_exp, ref = "Con exposicion") # CONTROL: E (NOT-SEVERE) -> CASE: D (severe)
levels(normal.log2.raw.eSet.VIVAX.SEV.FILTER$con_exp)
```

#### Correlations

```{r, fig.align="center", fig.height=6, fig.width=10, warning=FALSE}
#str(sampleSEVcovariates)
example <- sampleSEVcovariates
example$casecontrol <- as.numeric(example$casecontrol)
example$CumSumSevere7 <- as.numeric(example$CumSumSevere7)
example$CumSumSevere10 <- as.numeric(example$CumSumSevere10)
example$con_exp <- as.numeric(example$con_exp)
example$expo_CAT <- as.numeric(example$expo_CAT)
example$malaria_pasado_num <- as.numeric(example$malaria_pasado_num)
example$edad_CAT <- as.numeric(example$edad_CAT)
example$edad <- as.numeric(example$edad)
#str(example)

chart.Correlation(example, pch=16)
```

```{r, fig.align="center", fig.height=8, fig.width=10, warning=FALSE}
#str(sampleSEV)
example <- sampleSEV[,c(1:9,14,22:28,31,41:43)]
#str(example)
example$criticos <- as.numeric(example$criticos)
example$casecontrol <- as.numeric(example$casecontrol)
example$cerebral2 <- as.numeric(example$cerebral2)
example$criticalnopros <- as.numeric(example$criticalnopros)
example$edad <- as.numeric(example$edad)
example$sexo <- as.numeric(example$sexo)
example$talla <- as.numeric(example$talla)
example$peso <- as.numeric(example$peso)
example$admin_uci <- as.numeric(example$admin_uci)
example$temperatura <- as.numeric(example$temperatura)
example$hemoglobina <- as.numeric(example$hemoglobina)
example$creatinina <- as.numeric(example$creatinina)
example$glucosa <- as.numeric(example$glucosa)
example$bilitot <- as.numeric(example$bilitot)
example$bilind <- as.numeric(example$bilind)
example$bildirect <- as.numeric(example$bildirect)
example$parasitemia <- as.numeric(example$parasitemia)
example$malaria_pasado_num <- as.numeric(example$malaria_pasado_num)
example$CumSumSevere7 <- as.numeric(example$CumSumSevere7)
example$CumSumSevere10 <- as.numeric(example$CumSumSevere10)
example$con_exp <- as.numeric(example$con_exp)
#str(example)

chart.Correlation(example, pch=16)
```

```{r, fig.height=3, fig.width=9, fig.align='center', eval=FALSE}
par(mfrow=c(1,3))
plot(sampleSEVcovariates$edad, sampleSEVcovariates$malaria_pasado_num, xlab="Edad", ylab="Eventos previos de malaria") # pch=19,
plot(sampleSEVcovariates$malaria_pasado_num, sampleSEVcovariates$CumSumSevere10, xlab="Eventos previos de malaria", ylab="#/7 síntomas")
plot(sampleSEVcovariates$malaria_pasado_num, sampleSEVcovariates$CumSumSevere7, xlab="Eventos previos de malaria", ylab="#/10 síntomas")
```

```{r, fig.height=6, fig.width=6, fig.align='center', warning=FALSE, message=FALSE}
#library(ggplot2)
#library(Rmisc)
edexp1<- ggplot(sampleSEVcovariates, aes(x=edad,y=malaria_pasado_num, colour=casecontrol)) + geom_point() #+ ggtitle("xx") + geom_line() + # group=con_exp
edexp1x<- edexp1 + theme(legend.position=c(.8,.8))
#edexp2<- ggplot(sampleSEVcovariates, aes(x=edad,y=malaria_pasado_num, colour=casecontrol)) +geom_point() + geom_smooth(alpha=.2,size=.1)
#edexp2x <- edexp2 + theme(legend.position=c(.8,.8))

edexp3<- ggplot(sampleSEVcovariates, aes(x=edad,y=malaria_pasado_num)) +geom_point() + geom_smooth()

expSymp71<- ggplot(sampleSEVcovariates, aes(x=malaria_pasado_num,y=CumSumSevere7, colour=casecontrol)) + geom_point()
expSymp71x <- expSymp71 + theme(legend.position=c(.8,.8))
expSymp101<- ggplot(sampleSEVcovariates, aes(x=malaria_pasado_num,y=CumSumSevere10, colour=casecontrol)) + geom_point()
expSymp101x <- expSymp101 + theme(legend.position=c(.8,.8))

#expSymp72<- ggplot(sampleSEVcovariates, aes(x=malaria_pasado_num,y=CumSumSevere7)) +  geom_point() + geom_smooth(alpha=.2,size=.1)
#expSymp102<- ggplot(sampleSEVcovariates, aes(x=malaria_pasado_num,y=CumSumSevere10)) +geom_point() + geom_smooth(alpha=.2,size=.1)


multiplot(edexp1x,expSymp71x,edexp3,expSymp101x,cols=2)
#multiplot(edexp1,expSymp71,expSymp101,cols=3)
#multiplot(edexp1,edexp2,edexp3,cols=3)

```

```{r, fig.height=6, fig.width=6, fig.align='center', warning=FALSE, message=FALSE}
#library(ggplot2)
#library(Rmisc)
edexp1<- ggplot(sampleSEVcovariates, aes(x=edad_CAT,y=malaria_pasado_num, fill=casecontrol)) + geom_boxplot() #+ geom_point() #+ ggtitle("xx") + geom_line() + # group=con_exp
edexp1x<- edexp1 + theme(legend.position=c(.8,.8))
#edexp2<- ggplot(sampleSEVcovariates, aes(x=edad,y=malaria_pasado_num, colour=casecontrol)) +geom_point() + geom_smooth(alpha=.2,size=.1)
#edexp2x <- edexp2 + theme(legend.position=c(.8,.8))

edexp3<- ggplot(sampleSEVcovariates, aes(x=edad_CAT,y=malaria_pasado_num)) +geom_boxplot() #+geom_point() + geom_smooth()

expSymp71<- ggplot(sampleSEVcovariates, aes(x=expo_CAT,y=CumSumSevere7, colour=casecontrol)) + geom_boxplot() #+ geom_point()
expSymp71x <- expSymp71 + theme(legend.position=c(.8,.8))
expSymp101<- ggplot(sampleSEVcovariates, aes(x=expo_CAT,y=CumSumSevere10, colour=casecontrol)) + geom_boxplot() # + geom_point()
expSymp101x <- expSymp101 + theme(legend.position=c(.8,.8))

#expSymp72<- ggplot(sampleSEVcovariates, aes(x=malaria_pasado_num,y=CumSumSevere7)) +  geom_point() + geom_smooth(alpha=.2,size=.1)
#expSymp102<- ggplot(sampleSEVcovariates, aes(x=malaria_pasado_num,y=CumSumSevere10)) +geom_point() + geom_smooth(alpha=.2,size=.1)


multiplot(edexp1x,expSymp71x,
          edexp3,expSymp101x,cols=2)
#multiplot(edexp1,expSymp71,expSymp101,cols=3)
#multiplot(edexp1,edexp2,edexp3,cols=3)

```


## Feature covariates
```{r, eval=FALSE}
normal.log2.raw.eSet
head(fData(normal.log2.raw.eSet))
##POSSIBLE ADDING --> INFO ABOUT Exon_Segment per antigen ----> BIOSIGNATURES https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL18316
```


## Microarray data
### RAW heatmap

*Heatmaps w/o clustering*

**WHOLE MICROARRAY**

```{r}
########################### SAMPLE ORDER per group

# [START FROM THE BEGINNNING]
eset<-normal.log2.raw.eSet
eset$Group <- factor(eset$Group)
#eset$Group

#pData(eset) <- cbind(pData(eset),cat_2,t(col.mean.m_dataX))#ncol(pData(eset))
#rownames(sampleSEVcovariates)
#cat_2 <- data.frame(as.factor(matrix(sampleSEVcovariates$edad_CAT)))
#rownames(cat_2) <- rownames(sampleSEVcovariates)
#colnames(cat_2) <- "edad_CAT_2"
#pData(eset)[23] <- sampleSEVcovariates$edad_CAT
#pData(eset) <- cbind(pData(eset),cat_2)#ncol(pData(eset))

# starts
#class(exprs(eset)) #matrix
m_data <- exprs(eset)
m_data<-as.data.frame(m_data)

# [MODIFICATION 28jul2016]
col.mean.m_dataXX<-t(as.matrix(apply(m_data, 2, mean, na.rm=TRUE)))
rownames(col.mean.m_dataXX) <- c("col.mean.m")                 # assign row names 
#col.mean.m_dataX

row.mean.m_dataXX<-as.matrix(apply(m_data, 1, mean, na.rm=TRUE))
colnames(row.mean.m_dataXX) <- c("row.mean.m")                 # assign row names 
#row.mean.m_data


m_data_row.mean<-cbind(m_data,row.mean.m_dataXX)
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean<-m_data_row.mean[order(-m_data_row.mean$row.mean.m),]
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean$row.mean.m<-NULL
#dim(m_data_row.mean)
#m_data_row.mean
m_data_row.mean<-as.matrix(m_data_row.mean)
#

m_data_col.row.mean<-rbind(m_data_row.mean,col.mean.m_dataXX)

m_data_col.row.mean<-t(m_data_col.row.mean)
m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)

m_data_col.row.mean<-m_data_col.row.mean[order(m_data_col.row.mean$col.mean.m),]
m_data_col.row.mean$col.mean.m<-NULL
m_data_col.row.mean<-t(m_data_col.row.mean)

m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)
#dim(m_data_col.row.mean)
#m_data_col.row.mean
m_data_col.row.mean<-as.matrix(m_data_col.row.mean)
# ends


##### TOP10 OF HIGHER REACTIVITY in BOTH GROUPS -> the same as sort.by AveExp
#head(rownames(m_data_col.row.mean), 10)

# [ADDED 28jul2016]
feature_sort_meanXX<-rownames(m_data_col.row.mean)
sample_sort_mean_FIRSTXX<-colnames(m_data_col.row.mean)
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataXX))#ncol(pData(eset))
#
eset <- eset[feature_sort_meanXX,sample_sort_mean_FIRSTXX] ## REORDER samples and features inside the ExpressionSet

#eset$Group
```

```{r, fig.height=8, fig.width=8, fig.align="center", eval=FALSE, echo=FALSE}
#**`gplots` package** [@gplots]
#
color.map <- function(Group) { if (Group=="D") "#ff0000" else "#00ffff" }   # GROUP D is RED!!!!
patientcolors.SEV <- unlist(lapply(eset$Group, color.map))
#table(patientcolors.SEV)
#
##### THIS IS THE RIGHT HEATMAP (with unordered groups)
heatmap.2(exprs(eset), 
          
          Rowv = FALSE,
          Colv = FALSE,
          dendrogram = "none",
          
          col=bluered(10), 
          scale = "none", 
          ColSideColors = patientcolors.SEV, 
          key = TRUE, symkey = FALSE,
          density.info = "histogram", denscol = "black", 
          trace = "none", cexRow = 0.5)

par(lend = 1)           # square line ends for the color legend
legend("topright",      # location of the legend on the heatmap plot
    legend = c("Severe", "Non-severe"), # category labels
    col = c("#ff0000", "#00ffff"),  # color key
    lty= 1,             # line style
    lwd = 10            # line width
)
#
```

**using `NMF` package** [@Gaujoux2010NMF]

#### By Mean Sample Intensity
Sample controls (group F) are absent. They include **Day1** and **Day2** raw intensities of:

- No1ry.1 (?)
- No1ry.2 (?)
- AntiEcoli
- PosCtrl.PNG.Pool

Required to **test Reproducibility** and microarray **Quality Control**.

```{r, eval=TRUE, fig.height=7, fig.width=10, fig.align="center"}
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,3:ncol(pData(eset))]
#library(NMF)
aheatmap(exprs(eset), Rowv = NA, Colv = NA, annCol = pData(eset))
```

### LOG2 heatmap

**SEVERE specific**

```{r}
########################### SAMPLE ORDER per group

# [START FROM THE BEGINNNING]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
#eset$Group

#pData(eset) <- cbind(pData(eset),cat_2,t(col.mean.m_dataX))#ncol(pData(eset))
#rownames(sampleSEVcovariates)
#cat_2 <- data.frame(as.factor(matrix(sampleSEVcovariates$edad_CAT)))
#rownames(cat_2) <- rownames(sampleSEVcovariates)
#colnames(cat_2) <- "edad_CAT_2"
#pData(eset)[23] <- sampleSEVcovariates$edad_CAT
#pData(eset) <- cbind(pData(eset),cat_2)#ncol(pData(eset))

# starts
#class(exprs(eset)) #matrix
m_data <- exprs(eset)
m_data<-as.data.frame(m_data)

# [MODIFICATION 28jul2016]
col.mean.m_dataX<-t(as.matrix(apply(m_data, 2, mean, na.rm=TRUE)))
rownames(col.mean.m_dataX) <- c("col.mean.m")                 # assign row names 
#col.mean.m_dataX

row.mean.m_data<-as.matrix(apply(m_data, 1, mean, na.rm=TRUE))
colnames(row.mean.m_data) <- c("row.mean.m")                 # assign row names 
#row.mean.m_data


m_data_row.mean<-cbind(m_data,row.mean.m_data)
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean<-m_data_row.mean[order(-m_data_row.mean$row.mean.m),]
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean$row.mean.m<-NULL
#dim(m_data_row.mean)
#m_data_row.mean
m_data_row.mean<-as.matrix(m_data_row.mean)
#

m_data_col.row.mean<-rbind(m_data_row.mean,col.mean.m_dataX)

m_data_col.row.mean<-t(m_data_col.row.mean)
m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)

m_data_col.row.mean<-m_data_col.row.mean[order(m_data_col.row.mean$col.mean.m),]
m_data_col.row.mean$col.mean.m<-NULL
m_data_col.row.mean<-t(m_data_col.row.mean)

m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)
#dim(m_data_col.row.mean)
#m_data_col.row.mean
m_data_col.row.mean<-as.matrix(m_data_col.row.mean)
# ends


##### TOP10 OF HIGHER REACTIVITY in BOTH GROUPS -> the same as sort.by AveExp
#head(rownames(m_data_col.row.mean), 10)

# [ADDED 28jul2016]
feature_sort_mean<-rownames(m_data_col.row.mean)
sample_sort_mean_FIRST<-colnames(m_data_col.row.mean)
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#
eset <- eset[feature_sort_mean,sample_sort_mean_FIRST] ## REORDER samples and features inside the ExpressionSet

#eset$Group
```


#### By Age category
*potentially GENERAL*
```{r, eval=TRUE, echo=TRUE, fig.height=8, fig.width=8, fig.align="center"}
#
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
#
#pData(eset)[23] <- sampleSEVcovariates$edad_CAT
#eset$Group
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#
eset <- eset[feature_sort_mean,sample_sort_mean_FIRST] ## REORDER samples and features inside the ExpressionSet
#eset$Group
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]
#
sample_sort_AGE <- rownames(pData(eset)[order(pData(eset)$edad_CAT, decreasing = FALSE),])
eset <- eset[feature_sort_mean,sample_sort_AGE] ## REORDER samples and features inside the ExpressionSet
#library(NMF)
aheatmap(exprs(eset), Rowv = NA, Colv = NA, annCol = pData(eset))
#
```

```{r, fig.height=4, fig.width=9, fig.align='center', warning=FALSE, message=FALSE}
bindAGE <- t(as.matrix(tapply(exprs(eset)[1,], eset$edad_CAT, mean)))
for (i in 2:dim(exprs(eset))[1]) {
  bindAGE <- rbind(bindAGE,t(as.matrix(tapply(exprs(eset)[i,], eset$edad_CAT, mean, na.rm=TRUE))))
}

totalAGE <- as.matrix(apply(exprs(eset), 1, mean, na.rm=TRUE))
#colnames(totalAGE) <- c("tot.mean")
matAGE <- cbind(totalAGE, bindAGE)
colnames(matAGE) <- c("tot.mean", "age1", "age2", "age3","age4")
matAGE <- as.data.frame(matAGE)

dfAGE <- matAGE %>%
  gather(ageClass, sum.react, tot.mean:age4) %>%
  ggplot(aes(x=ageClass, y=sum.react)) + geom_boxplot()

age2 <- ggplot(matAGE) + 
  geom_point(aes(x=tot.mean, y=age1, colour="age1")) +
  geom_point(aes(x=tot.mean, y=age2, colour="age2")) +
  geom_point(aes(x=tot.mean, y=age3, colour="age3")) +
  geom_point(aes(x=tot.mean, y=age3, colour="age4"))

multiplot(dfAGE, age2, cols=2)

```

```{r, fig.height=4, fig.width=14, fig.align='center', warning=FALSE, message=FALSE}
#library(ggplot2)
#library(Rmisc)
ageBOX1<- ggplot(pData(eset), aes(x=edad_CAT,y=col.mean.m
                                 #, fill=casecontrol
                                 )) + geom_boxplot() #+ geom_point() #+ ggtitle("xx") + geom_line() + # group=con_exp
ageBOX2<- ggplot(pData(eset), aes(x=edad_CAT,y=col.mean.m
                                 , fill=casecontrol
                                 )) + geom_boxplot() #+ geom_point() #+ ggtitle("xx") + geom_line() + # group=con_exp
ageBOX2x<- ageBOX2 + theme(legend.position=c(.9,.9))
ageBOX3<- ggplot(pData(eset), aes(x=edad_CAT,y=col.mean.m
                                 , fill=con_exp
                                 )) + geom_boxplot() #+ geom_point() #+ ggtitle("xx") + geom_line() + # group=con_exp
ageBOX3x<- ageBOX3 + theme(legend.position=c(.9,.9))
#
multiplot(ageBOX1,ageBOX2x,ageBOX3x,cols=3)
```


#### By Exposure category
```{r, eval=TRUE, echo=TRUE, fig.height=8, fig.width=8, fig.align="center"}
########################### SAMPLE ORDER per group

# [START FROM THE BEGINNNING]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
#eset$Group
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#
eset <- eset[feature_sort_mean,sample_sort_mean_FIRST] ## REORDER samples and features inside the ExpressionSet
#eset$Group
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]

sample_sort_EXPOSURE <- rownames(pData(eset)[order(pData(eset)$expo_CAT, decreasing = FALSE),])
eset <- eset[feature_sort_mean,sample_sort_EXPOSURE] ## REORDER samples and features inside the ExpressionSet
#library(NMF)
aheatmap(exprs(eset), Rowv = NA, Colv = NA, annCol = pData(eset))
```


```{r, fig.height=4, fig.width=9, fig.align='center', warning=FALSE, message=FALSE}
bindEXPO <- t(as.matrix(tapply(exprs(eset)[1,], eset$expo_CAT, mean)))
for (i in 2:dim(exprs(eset))[1]) {
  bindEXPO <- rbind(bindEXPO,t(as.matrix(tapply(exprs(eset)[i,], eset$expo_CAT, mean, na.rm=TRUE))))
}

totalEXPO <- as.matrix(apply(exprs(eset), 1, mean, na.rm=TRUE))
#colnames(totalEXPO) <- c("tot.mean")
matEXPO <- cbind(totalEXPO, bindEXPO)
colnames(matEXPO) <- c("tot.mean", "expo1", "expo2", "expo3", "expo4")
matEXPO <- as.data.frame(matEXPO)

dfEXPO <- matEXPO %>%
  gather(expoClass, sum.react, tot.mean:expo4) %>%
  ggplot(aes(x=expoClass, y=sum.react)) + geom_boxplot()

expo2 <- ggplot(matEXPO) + 
  geom_point(aes(x=tot.mean, y=expo1, colour="expo1")) +
  geom_point(aes(x=tot.mean, y=expo2, colour="expo2")) +
  geom_point(aes(x=tot.mean, y=expo3, colour="expo3")) +
  geom_point(aes(x=tot.mean, y=expo4, colour="expo4"))


multiplot(dfEXPO, expo2, cols=2)

```

```{r, fig.height=3, fig.width=8, fig.align='center', warning=FALSE, message=FALSE}
#library(ggplot2)
#library(Rmisc)
expBOX1<- ggplot(pData(eset), aes(x=expo_CAT,y=col.mean.m
                                 #, fill=casecontrol
                                 )) + geom_boxplot() #+ geom_point() #+ ggtitle("xx") + geom_line() + # group=con_exp
expBOX2<- ggplot(pData(eset), aes(x=expo_CAT,y=col.mean.m
                                 , fill=casecontrol
                                 )) + geom_boxplot() #+ geom_point() #+ ggtitle("xx") + geom_line() + # group=con_exp
#expBOX2x<- expBOX2 + theme(legend.position=c(.8,.8))
#expBOX3<- ggplot(pData(eset), aes(x=expo_CAT,y=col.mean.m
#                                 , fill=con_exp
#                                 )) + geom_boxplot() #+ geom_point() #+ ggtitle("xx") + geom_line() + # group=con_exp
#expBOX3x<- expBOX3 + theme(legend.position=c(.8,.8))
#
multiplot(expBOX1,expBOX2,cols=2)
```


#### By Cumulative Symptoms
```{r, eval=TRUE, echo=TRUE, fig.height=8, fig.width=8, fig.align="center"}
#
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
#eset$Group
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#
eset <- eset[feature_sort_mean,sample_sort_mean_FIRST] ## REORDER samples and features inside the ExpressionSet
#eset$Group
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]
#
sample_sort_SYMPTOMS <- rownames(pData(eset)[order(pData(eset)$CumSumSevere10, decreasing = FALSE),])
eset <- eset[feature_sort_mean,sample_sort_SYMPTOMS] ## REORDER samples and features inside the ExpressionSet
#library(NMF)
aheatmap(exprs(eset), Rowv = NA, Colv = NA, annCol = pData(eset))
#
```


```{r, eval=TRUE}
#head(anti.raw_data)[,1:5]
#head(log2.NA.norm.ANTIGENS)[,1:5]
#head(exprs(normal.log2.raw.eSet.VIVAX.SEV.FILTER))[,1:5]

rawhm <- anti.raw_data[feature_sort_meanXX,sample_sort_mean_FIRSTXX] ## REORDER samples and features inside the ExpressionSet
#head(rawhm)[,1:5]

#log2hp <- exprs(normal.log2.raw.eSet.VIVAX.SEV.FILTER)[feature_sort_mean,sample_sort_mean_FIRST] ## REORDER samples and features inside the ExpressionSet
#sum(log2hp==exprs(eset))
#head(log2hp)[,1:5]
#head(exprs(eset))[,1:5]

ctrlhm <- ctrl.raw_data[,sample_sort_mean_FIRSTXX] ## REORDER samples and features inside the ExpressionSet
#head(ctrlhm)[,1:5]

#head(median.ctrl.raw_data)[,1:5]
#median.ctrl.raw_data
```

```{r, fig.height=8, fig.width=8, fig.align="center", eval=FALSE, echo=FALSE}
##### THIS IS THE RIGHT HEATMAP (with unordered groups)
heatmap.2(rbind(rawhm,ctrlhm), 
          
          Rowv = FALSE,
          Colv = FALSE,
          dendrogram = "none",
          
          col=bluered(100), 
          scale = "none", 
          ColSideColors = patientcolors.SEV, 
          key = TRUE, symkey = FALSE,
          density.info = "histogram", denscol = "black", 
          trace = "none", cexRow = 0.5)

par(lend = 1)           # square line ends for the color legend
legend("topright",      # location of the legend on the heatmap plot
    legend = c("Severe", "Non-severe"), # category labels
    col = c("#ff0000", "#00ffff"),  # color key
    lty= 1,             # line style
    lwd = 10            # line width
)
#
```

```{r, eval=FALSE, fig.align="center"}
aheatmap(rbind(rawhm,ctrlhm), Rowv = NA, Colv = NA)
```


## Technical plots
### Backgroud effect
```{r, fig.height=8, fig.width=8, fig.align='center', warning=FALSE, message=FALSE}
#rownames(row.mean.m_data)
RAW_data <- anti.raw_data[rownames(row.mean.m_dataXX),]
row.mean.RAW_data<-as.matrix(apply(RAW_data, 1, median, na.rm=TRUE))
colnames(row.mean.RAW_data) <- c("row.mean.RAW")                 # assign row names 
#
col.mean.RAW_data<-as.matrix(apply(RAW_data[,colnames(col.mean.m_dataXX)], 2, median, na.rm=TRUE)) ### ONLY FOR 60 samples
colnames(col.mean.RAW_data) <- c("col.mean.RAW")                 # assign row names 
#
#head(row.mean.RAW_data)
#head(row.mean.m_data)
totint <- cbind(row.mean.m_dataXX,row.mean.RAW_data)
totint <- as.data.frame(totint)
colnames(totint) <- c("LOG2.feature.mean.int", "RAW.feature.median.int")
t1 <- ggplot(totint, aes(x=LOG2.feature.mean.int,y=RAW.feature.median.int)) + geom_point() + geom_smooth()
#
sampleint <- cbind(t(col.mean.m_dataXX),col.mean.RAW_data)
sampleint <- as.data.frame(sampleint)
colnames(sampleint) <- c("LOG2.sample.mean.int", "RAW.sample.median.int")
t2 <- ggplot(sampleint, aes(x=LOG2.sample.mean.int,y=RAW.sample.median.int)) + geom_point() + geom_smooth()
#
#class(col.mean.m_dataX)
#class(median.ctrl.raw_data)
No.DNA.int <- subset(median.ctrl.raw_data, select= colnames(col.mean.m_dataXX))
#No.DNA.int
#col.mean.m_dataX
back <- rbind(col.mean.m_dataXX,No.DNA.int)
back <- as.data.frame(t(back))
colnames(back) <- c("LOG2.sample.mean.int", "RAW.No.DNA.median.int")
b1 <- ggplot(back, aes(x=LOG2.sample.mean.int,y=RAW.No.DNA.median.int)) + geom_point() + geom_smooth()
#
rawk <- cbind(col.mean.RAW_data,t(No.DNA.int))
rawk <- as.data.frame(rawk)
colnames(rawk) <- c("RAW.sample.median.int", "RAW.No.DNA.median.int")
b2 <- ggplot(rawk, aes(x=RAW.sample.median.int,y=RAW.No.DNA.median.int)) + geom_point() + geom_smooth()
#
multiplot(t1,b2,t2,b1, cols=2)
#
```

```{r, fig.width=14, fig.height=7, fig.align='center'}
par(mfrow=c(1,2))

# [START FROM THE BEGINNNING]
eset<-normal.log2.raw.eSet
#
eset <- eset[feature_sort_meanXX,sample_sort_mean_FIRSTXX] ## REORDER samples and features inside the ExpressionSet

aheatmap(exprs(eset), Rowv = NA, Colv = NA, main = "LOG2")
aheatmap(rbind(rawhm,ctrlhm), Rowv = NA, Colv = NA, main = "RAW")
```


### Heteroskedasticity 
*pre- & post- transformation/normalization*
```{r, fig.height=4, fig.width=8, fig.align='center', warning=FALSE, message=FALSE}
#
RAW_data <- anti.raw_data[rownames(row.mean.m_dataXX),]
row.sd.RAW_data<-as.matrix(apply(RAW_data, 1, sd, na.rm=TRUE))
colnames(row.sd.RAW_data) <- c("row.mean.RAW")                 # assign row names 
#
homoskRAW <- cbind(row.mean.RAW_data,row.sd.RAW_data)
homoskRAW <- as.data.frame(homoskRAW)
colnames(homoskRAW) <- c("RAW.feature.median.int","RAW.feature.sd.int")
h2 <- ggplot(homoskRAW, aes(x=RAW.feature.median.int,y=RAW.feature.sd.int)) + geom_point() + geom_smooth()
#
row.sd.m_data<-as.matrix(apply(m_data, 1, sd, na.rm=TRUE))
colnames(row.sd.m_data) <- c("row.sd.m")                 # assign row names 
#
homosk <- cbind(row.mean.m_data,row.sd.m_data)
homosk <- as.data.frame(homosk)
colnames(homosk) <- c("LOG2.feature.mean.int", "LOG2.feature.sd.int")
h1 <- ggplot(homosk, aes(x=LOG2.feature.mean.int,y=LOG2.feature.sd.int)) + geom_point() + geom_smooth()#method="lm",se=FALSE
#
multiplot(h2,h1,cols=2)
```

### Pad position PCA **[pending]**
```{r, eval=FALSE}
eset <- normal.log2.raw.eSet
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,1:8]
head(pData(eset))
```


### Test Validity

*Using Pf IVTT and purified proteins*

As a reference, check Crompton, 2010 [@crompton2010] supplementary info [here](http://www.pnas.org/content/suppl/2010/03/25/1001323107.DCSupplemental/pnas.201001323SI.pdf).

All comparison are in raw scale, since normalization can not be done with the purified protein spots.

***

```{r}
rownames(pure.raw_data)
```


**Pf EBA175**

- PF3D7_0731500_e1s2.361	falciparum	IVTT.AG	PF3D7_0731500	erythrocyte binding antigen-175 (EBA175)
- EBA175, 0.3mg/mL.23	falciparum	PURIFIED.PROTEIN	NA	EBA175, 0.3mg/mL
- EBA175, 0.1mg/mL.24	falciparum	PURIFIED.PROTEIN	NA	EBA175, 0.1mg/mL

**Pf MSP3a**

- PVX_097720.273	vivax	IVTT.AG	PVX_097720	merozoite surface protein 3 alpha (MSP3a)
- MSP3, 0.3mg/mL.25	falciparum	PURIFIED.PROTEIN	NA	MSP3, 0.3mg/mL
- MSP3, 0.1mg/mL.26	falciparum	PURIFIED.PROTEIN	NA	MSP3, 0.1mg/mL

**Pf Rh1 PEP8 PEP1**

- Rh1 PEP8, 0.3mg/mL	falciparum	PURIFIED.PROTEIN	NA	Rh1 PEP8, 0.3mg/mL
- Rh1 PEP8, 0.1mg/mL	falciparum	PURIFIED.PROTEIN	NA	Rh1 PEP8, 0.1mg/mL
- Rh1 PEP1, 0.3mg/mL	falciparum	PURIFIED.PROTEIN	NA	Rh1 PEP1, 0.3mg/mL
- Rh1 PEP1, 0.1mg/mL	falciparum	PURIFIED.PROTEIN	NA	Rh1 PEP1, 0.1mg/mL

**Pf EBA140**

- EBA140, 0.3mg/mL.312	falciparum	PURIFIED.PROTEIN	NA	EBA140, 0.3mg/mL
- EBA140, 0.1mg/mL.313	falciparum	PURIFIED.PROTEIN	NA	EBA140, 0.1mg/mL

**Pf RH2b**

- PF3D7_1335300_e1s2.38	falciparum	IVTT.AG	PF3D7_1335300	reticulocyte binding protein 2 homologue b (RH2b)
- PF3D7_1335300_e1s1.327	falciparum	IVTT.AG	PF3D7_1335300	reticulocyte binding protein 2 homologue b (RH2b)
- Rh2, 0.3mg/mL.314	falciparum	PURIFIED.PROTEIN	NA	Rh2, 0.3mg/mL
- Rh2, 0.1mg/mL.315	falciparum	PURIFIED.PROTEIN	NA	Rh2, 0.1mg/mL

**Pf MSP1**

- PF3D7_0930300_s1.67	falciparum	IVTT.AG	PF3D7_0930300	merozoite surface protein 1 (MSP1)
- PF3D7_0930300_s2.941	falciparum	IVTT.AG	PF3D7_0930300	merozoite surface protein 1 (MSP1)
- MSP1, 0.3mg/mL.601	falciparum	PURIFIED.PROTEIN	NA	MSP1, 0.3mg/mL
- MSP1, 0.1mg/mL.602	falciparum	PURIFIED.PROTEIN	NA	MSP1, 0.1mg/mL

**Pf MSP10 and Pv MSP10**

- PF3D7_0620400.974	falciparum	IVTT.AG	PF3D7_0620400	merozoite surface protein 10 (MSP10)
- PVX_114145_1o1.272	vivax	IVTT.AG	PVX_114145	Merozoite surface protein 10, MSP10

**Pf LSA1**

- PF3D7_1036400_e2s2.78	falciparum	IVTT.AG	PF3D7_1036400	liver stage antigen 1 (LSA1)
- PF3D7_1036400_e1.360	falciparum	IVTT.AG	PF3D7_1036400	liver stage antigen 1 (LSA1)
- PF3D7_1036400_e2s1.945	falciparum	IVTT.AG	PF3D7_1036400	liver stage antigen 1 (LSA1)
- PF3D7_1036400_e1.951	falciparum	IVTT.AG	PF3D7_1036400	liver stage antigen 1 (LSA1)
- Pf LSA1, 0.3mg/mL.605	falciparum	PURIFIED.PROTEIN	NA	Pf LSA1, 0.3mg/mL
- Pf LSA1, 0.1mg/mL.606	falciparum	PURIFIED.PROTEIN	NA	Pf LSA1, 0.1mg/mL

**Pf MSP2**

- PF3D7_0206800.65	falciparum	IVTT.AG	PF3D7_0206800	merozoite surface protein 2 (MSP2)
- MSP2, 0.3mg/mL.890	falciparum	PURIFIED.PROTEIN	NA	MSP2, 0.3mg/mL
- MSP2, 0.1mg/mL.891	falciparum	PURIFIED.PROTEIN	NA	MSP2, 0.1mg/mL

**Pf CSP**

- PF3D7_0304600.932	falciparum	IVTT.AG	PF3D7_0304600	circumsporozoite (CS) protein (CSP)
- Pf CSP, 0.3mg/mL.894	falciparum	PURIFIED.PROTEIN	NA	Pf CSP, 0.3mg/mL
- Pf CSP, 0.1mg/mL.895	falciparum	PURIFIED.PROTEIN	NA	Pf CSP, 0.1mg/mL

***
  
**Pv AMA1**

- Pvivax AMA1, 0.3mg/mL.603	vivax	PURIFIED.PROTEIN	NA	Pvivax AMA1, 0.3mg/mL
- Pvivax AMA1, 0.1mg/mL.604	vivax	PURIFIED.PROTEIN	NA	Pvivax AMA1, 0.1mg/mL
- Pvivax AMA1 Eoto monomer prep2, 0.3mg/mL.892	vivax	PURIFIED.PROTEIN	NA	Pvivax AMA1 Eoto monomer prep2, 0.3mg/mL
- Pvivax AMA1 Eoto monomer prep2, 0.1mg/mL.893	vivax	PURIFIED.PROTEIN	NA	Pvivax AMA1 Eoto monomer prep2, 0.1mg/mL

***

#### **Pf EBA175, MSP2, CSP, Pv AMA**
```{r}
EBA175 <- as.data.frame(cbind(as.matrix(anti.raw_data["PF3D7_0731500_e1s2.361",]), as.matrix(pure.raw_data["EBA175, 0.3mg/mL.23",]), as.matrix(pure.raw_data["EBA175, 0.1mg/mL.24",])))
colnames(EBA175) <- c("PF3D7_0731500_e1s2.361", "EBA175_0.3.23", "EBA175_0.1.24")

# PF3D7_0930300_s1.67	falciparum	IVTT.AG	PF3D7_0930300	merozoite surface protein 1 (MSP1)
MSP1s1 <- as.data.frame(cbind(as.matrix(anti.raw_data["PF3D7_0930300_s1.67",]), as.matrix(pure.raw_data["MSP1, 0.3mg/mL.601",]), as.matrix(pure.raw_data["MSP1, 0.1mg/mL.602",])))
colnames(MSP1s1) <- c("PF3D7_0930300_s1.67", "MSP1_0.3.601", "MSP1_0.1.602")

# PF3D7_0930300_s2.941	falciparum	IVTT.AG	PF3D7_0930300	merozoite surface protein 1 (MSP1)
MSP1s2 <- as.data.frame(cbind(as.matrix(anti.raw_data["PF3D7_0930300_s2.941",]), as.matrix(pure.raw_data["MSP1, 0.3mg/mL.601",]), as.matrix(pure.raw_data["MSP1, 0.1mg/mL.602",])))
colnames(MSP1s2) <- c("PF3D7_0930300_s2.941", "MSP1_0.3.601", "MSP1_0.1.602")

# PF3D7_0206800.65	falciparum	IVTT.AG	PF3D7_0206800	merozoite surface protein 2 (MSP2)
MSP2 <- as.data.frame(cbind(as.matrix(anti.raw_data["PF3D7_0206800.65",]), as.matrix(pure.raw_data["MSP2, 0.3mg/mL.890",]), as.matrix(pure.raw_data["MSP2, 0.1mg/mL.891",])))
colnames(MSP2) <- c("PF3D7_0206800.65", "MSP2_0.3.890", "MSP2_0.1.891")

# PF3D7_0304600.932	falciparum	IVTT.AG	PF3D7_0304600	circumsporozoite (CS) protein (CSP)
CSP <- as.data.frame(cbind(as.matrix(anti.raw_data["PF3D7_0304600.932",]), as.matrix(pure.raw_data["Pf CSP, 0.3mg/mL.894",]), as.matrix(pure.raw_data["Pf CSP, 0.1mg/mL.895",])))
colnames(CSP) <- c("PF3D7_0304600.932", "PfCSP_0.3.894", "PfCSP_0.1.895")

# VIVAX AMA1
PvAMA1 <- as.data.frame(cbind(as.matrix(pure.raw_data["Pvivax AMA1, 0.3mg/mL.603",]), as.matrix(pure.raw_data["Pvivax AMA1, 0.1mg/mL.604",]), as.matrix(pure.raw_data["Pvivax AMA1 Eoto monomer prep2, 0.3mg/mL.892",]), as.matrix(pure.raw_data["Pvivax AMA1 Eoto monomer prep2, 0.1mg/mL.893",])))
colnames(PvAMA1) <- c("PvAMA1_0.3.603", "PvAMA1_0.1.604", "PvAMA1_Eoto_prep2_0.3.892", "PvAMA1_Eoto_prep2_0.1.893")

```

```{r, fig.align="center", fig.height=12, fig.width=10, eval=FALSE}
v11 <- ggplot(EBA175, aes(x=EBA175_0.3.23,y=PF3D7_0731500_e1s2.361)) + geom_point() + geom_smooth(method = "lm")
v12 <- ggplot(EBA175, aes(x=EBA175_0.1.24,y=PF3D7_0731500_e1s2.361)) + geom_point() + geom_smooth(method = "lm")
v21 <- ggplot(MSP1s1, aes(x=MSP1_0.3.601,y=PF3D7_0930300_s1.67)) + geom_point() + geom_smooth(method = "lm")
v22 <- ggplot(MSP1s1, aes(x=MSP1_0.1.602,y=PF3D7_0930300_s1.67)) + geom_point() + geom_smooth(method = "lm")
v61 <- ggplot(MSP1s2, aes(x=MSP1_0.3.601,y=PF3D7_0930300_s2.941)) + geom_point() + geom_smooth(method = "lm")
v62 <- ggplot(MSP1s2, aes(x=MSP1_0.1.602,y=PF3D7_0930300_s2.941)) + geom_point() + geom_smooth(method = "lm")
v31 <- ggplot(MSP2, aes(x=MSP2_0.3.890,y=PF3D7_0206800.65)) + geom_point() + geom_smooth(method = "lm")
v32 <- ggplot(MSP2, aes(x=MSP2_0.1.891,y=PF3D7_0206800.65)) + geom_point() + geom_smooth(method = "lm")
v41 <- ggplot(CSP, aes(x=PfCSP_0.3.894,y=PF3D7_0304600.932)) + geom_point() + geom_smooth(method = "lm")
v42 <- ggplot(CSP, aes(x=PfCSP_0.1.895,y=PF3D7_0304600.932)) + geom_point() + geom_smooth(method = "lm")
v51 <- ggplot(PvAMA1, aes(x=PvAMA1_0.3.603,y=PvAMA1_Eoto_prep2_0.3.892)) + geom_point() + geom_smooth(method = "lm")
v52 <- ggplot(PvAMA1, aes(x=PvAMA1_0.1.604,y=PvAMA1_Eoto_prep2_0.1.893)) + geom_point() + geom_smooth(method = "lm")

multiplot(v11, v41, v31, #v21, v61, 
          v51,
          v12, v42, v32, #v22, v62, 
          v52,
          cols= 2)
```


```{r, warning=FALSE, fig.align="center", fig.height=6, fig.width=10}
#plot(PF3D7_0731500_e1s2.361 ~ EBA175_0.3.23, EBA175)
#plot(EBA175)
#hist(EBA175)

cor(EBA175, method = "pearson")
cor.test(~ PF3D7_0731500_e1s2.361 + EBA175_0.3.23, EBA175, method = "pearson")

#library(PerformanceAnalytics)
chart.Correlation(EBA175, method = "pearson")
chart.Correlation(MSP2, method = "pearson")
chart.Correlation(CSP, method = "pearson")
chart.Correlation(PvAMA1, method = "pearson")

#table.Correlation(EBA175[,2:3],EBA175[,1, drop=FALSE], conf.level=.95)
#table.Correlation(managers[,2:3],managers[,8, drop=FALSE], conf.level=.95)

#chart.Correlation(pbc, method = "spearman")
#chart.Correlation(sampleSEV, method = "spearman")

#summary(lm(PF3D7_0731500_e1s2.361 ~ EBA175_0.3.23, EBA175))$r.squared
```

#### **MSP1 s1/2**
```{r, fig.align="center", fig.height=6, fig.width=10, eval=FALSE}
multiplot(v21, v61, 
          v22, v62, 
          cols= 2)
```

```{r, warning=FALSE, fig.align="center", fig.height=6, fig.width=10}
chart.Correlation(MSP1s1, method = "pearson")
chart.Correlation(MSP1s2, method = "pearson")
```

#### **MSP10 distribution**

```{r, fig.height=12, fig.width=10, fig.align="center", message=FALSE}
# PF3D7_0620400.974	falciparum	IVTT.AG	PF3D7_0620400	merozoite surface protein 10 (MSP10)
# PVX_114145_1o1.272	vivax	IVTT.AG	PVX_114145	Merozoite surface protein 10, MSP10
MSP10 <- as.data.frame(cbind(as.matrix(anti.raw_data["PF3D7_0620400.974",]), as.matrix(anti.raw_data["PVX_114145_1o1.272",])))
MSP10 <- cbind(MSP10,rownames(MSP10),pData(normal.log2.raw.eSet)[,"Study"])
colnames(MSP10) <- c("PfMSP10", "PvMSP10", "sample.ID", "Study")

MSP10$Study <- factor(MSP10$Study)

#length(rownames(MSP10))
#phenoData(normal.log2.raw.eSet)
#pData(normal.log2.raw.eSet)
#varLabels(normal.log2.raw.eSet)

MSP10.x <- MSP10 %>%
  gather(specie, intensity, PfMSP10:PvMSP10) %>%
  mutate(specie=factor(specie)) %>%
  #filter(con_exp=="CON") %>%
  #mutate(feature=factor(feature)) %>%
  ggplot(aes(x=intensity,fill=specie)) + 
  geom_histogram(alpha=.5 ,position = "identity") + 
  #geom_density(alpha=.3) +
  labs(title="Raw Ab intensity against MSP10 per specie")
MSP10.xd <- MSP10 %>%
  gather(specie, intensity, PfMSP10:PvMSP10) %>%
  mutate(specie=factor(specie)) %>%
  #filter(con_exp=="CON") %>%
  #mutate(feature=factor(feature)) %>%
  ggplot(aes(x=intensity,fill=specie)) + 
  #geom_histogram(alpha=.5 ,position = "identity") + 
  geom_density(alpha=.3) +
  labs(title="Raw Ab intensity against MSP10 per specie")


#
MSP10 <- subset(MSP10, Study!="Primaquine")
MSP10 <- cbind(MSP10, pData(normal.log2.raw.eSet.VIVAX.SEV.FILTER)[,c("casecontrol","con_exp")])
#levels(MSP10$con_exp) <- c("CON","SIN")
#MSP10$Study <- factor(MSP10$Study)
#varLabels(normal.log2.raw.eSet.VIVAX.SEV.FILTER) #,
#phenoData(normal.log2.raw.eSet.VIVAX.SEV.FILTER)

#
MSP10.z <- MSP10 %>%
  gather(specie, intensity, PfMSP10:PvMSP10) %>%
  mutate(specie=factor(specie)) %>%
  #filter(Study!="Primaquine") %>%
  #mutate(feature=factor(feature)) %>%
  ggplot(aes(x=intensity,fill=specie)) + 
  geom_histogram(alpha=.5 ,position = "identity") + 
  #geom_density(alpha=.3) +
  labs(title="Raw Ab intensity against MSP10 per specie \n only Severe study")
MSP10.zd <- MSP10 %>%
  gather(specie, intensity, PfMSP10:PvMSP10) %>%
  mutate(specie=factor(specie)) %>%
  #filter(Study!="Primaquine") %>%
  #mutate(feature=factor(feature)) %>%
  ggplot(aes(x=intensity,fill=specie)) + 
  #geom_histogram(alpha=.5 ,position = "identity") + 
  geom_density(alpha=.3) +
  labs(title="Raw Ab intensity against MSP10 per specie \n only Severe study")
#
#str(MSP10)
#
MSP10.w <- MSP10 %>%
  gather(specie, intensity, PfMSP10:PvMSP10) %>%
  mutate(specie=factor(specie)) %>%
  #filter(Study!="Primaquine") %>%
  filter(specie=="PvMSP10") %>%
  #mutate(feature=factor(feature)) %>%
  ggplot(aes(x=intensity,fill=con_exp)) + 
  geom_histogram(alpha=.5 ,position = "identity") + 
  #geom_density(alpha=.3) +
  labs(title="Raw Ab intensity against MSP10 per specie \n only Severe study and PvMSP10")
MSP10.wd <- MSP10 %>%
  gather(specie, intensity, PfMSP10:PvMSP10) %>%
  mutate(specie=factor(specie)) %>%
  #filter(Study!="Primaquine") %>%
  filter(specie=="PvMSP10") %>%
  #mutate(feature=factor(feature)) %>%
  ggplot(aes(x=intensity,fill=con_exp)) + 
  #geom_histogram(alpha=.5 ,position = "identity") + 
  geom_density(alpha=.3) +
  labs(title="Raw Ab intensity against MSP10 per specie \n only Severe study and PvMSP10")
#
MSP10.y <- MSP10 %>%
  gather(specie, intensity, PfMSP10:PvMSP10) %>%
  mutate(specie=factor(specie)) %>%
  #filter(Study!="Primaquine") %>%
  filter(specie=="PvMSP10") %>%
  #mutate(feature=factor(feature)) %>%
  ggplot(aes(x=intensity,fill=casecontrol)) + 
  geom_histogram(alpha=.5 ,position = "identity") + 
  #geom_density(alpha=.3) +
  labs(title="Raw Ab intensity against MSP10 per specie \n only Severe study and PvMSP10")
MSP10.yd <- MSP10 %>%
  gather(specie, intensity, PfMSP10:PvMSP10) %>%
  mutate(specie=factor(specie)) %>%
  #filter(Study!="Primaquine") %>%
  filter(specie=="PvMSP10") %>%
  #mutate(feature=factor(feature)) %>%
  ggplot(aes(x=intensity,fill=casecontrol)) + 
  #geom_histogram(alpha=.5 ,position = "identity") + 
  geom_density(alpha=.3) +
  labs(title="Raw Ab intensity against MSP10 per specie \n only Severe study and PvMSP10")

multiplot(MSP10.x, MSP10.z, MSP10.w, #MSP10.y,
          MSP10.xd, MSP10.zd, MSP10.wd, #MSP10.yd,
          cols = 2)
```


### Test Reproducibility **[pending]**

*This will required sample **column** controls*. 

***

**REQUEST** 

- **REQUEST** Day1 and Day2 controls intensity data per feature!
- **REQUEST** also polypeptides per coded feature!

***

## Threshold statistics 
*TRUE/FALSE matrix*

### Magnitud and Frequency 

**per Feature**

*Seroprevalence of antigens **DO NOT APPLY**. Instead, report frequency*
```{r}
########## (1) [PROGRESS 5/6] SEROPOSITIVITY (of antigen per sample [column]) -> SEROPROVALENCE (of antigen across samples [row]) -> to identify # REACTIVE and # NON-REACTIVE antigens
#     [using the NoDNA control spots per sample]
######### LEARNED: twice "IVTT background" is not more conservative than "mean+2SD" seropositivity assessment

## - SEROPOSITIVITY (2xIVTTbkgd = 2xNoDNActrl = normalized_MFI>1) -> SEROPREVALENCE (10%)  -> # REACTIVE ANTIGENS      (ADi-N6 SAP 2015) [more conservative]
## - SEROPOSITIVITY (mean+1SD)                                    -> SEROPREVALENCE (10%)  -> # REACTIVE ANTIGENS      (Helb 2015)
## - SEROPOSITIVITY (mean+2SD)                                    -> SEROPREVALENCE (1%)   -> # REACTIVE ANTIGENS      (Skinner 2015)
# SUMMARY TABLE: 
# ID /Description / #Rve=A+B / % sero.prevalence / mean.rve.intensity / #A=Case / % A.sero.prevalence / A.mean.rve.intensity / #B=Control / % B.sero.prevalence / B.mean.rve.intensity /
# GRAPHIC: Venn-diagram



## (1.1) SEROPOSITIVITY (2xIVTTbkgd = 2xNoDNActrl = normalized_MFI>1) -> SEROPREVALENCE (10%)  -> # REACTIVE ANTIGENS      (ADi-N6 SAP 2015) [more conservative]

#eset <- normal.log2.raw.eSet.VIVAX.PQ    # 515 features X 140 samples
#eset <- normal.log2.raw.eSet.FALCIP.PQ   # 499 features X 140 samples
eset <- normal.log2.raw.eSet.VIVAX.SEV   # 515 features X 60 samples
#eset <- normal.log2.raw.eSet.FALCIP.SEV  # 499 features X 60 samples

#### BY 'genefilter' FUNCTION
#
f2 <- kOverA((dim(eset)[2])*.1,1)                           ## if (A= SEROPOSITIVE>= 1.00 and k= SEROPREVALENCE= 10%) then REACTIVE
ffun2 <- filterfun(f2)
wh2 <- genefilter(exprs(eset),ffun2)  # WHOLE data set
sum(wh2) # 255
#

#### BY HAND [this have validate the 'genefilter']
#
filter<-matrix(1,dim(eset)[1],dim(eset)[2])
filter.eset<-exprs(eset)>filter               # LOGICAL statement result  ------------> MAIN TRUE/FALSE MATRIX

#####-------------------------------------------------- SEROPREVALENCE (function)
#
sero.percent <- function(x) {
  (sum(x, na.rm = TRUE)/length(x))*100
}
#

sero.posit<-as.matrix(apply(filter.eset, 1, sero.percent))

# order [DONE]
sero.posit<-as.data.frame(sero.posit)
colnames(sero.posit) <- c("sero.prevalence") 

sero_mean<-as.matrix(apply(exprs(eset), 1, mean, na.rm = TRUE))
colnames(sero_mean) <- c("mean.intensity")

sero_sum<-as.matrix(apply(filter.eset, 1, sum, na.rm = TRUE))
colnames(sero_sum) <- c("rve.samples")


sero.posit<-cbind(sero_sum,sero.posit,sero_mean)
sero.posit<-sero.posit[order(-sero.posit$sero.prevalence),]
sero.posit<-as.data.frame(sero.posit)

#####-------------------------------------------------- SEROPREVALENCE (results)
sero.posit.over10<-subset(sero.posit, sero.prevalence>=10)
dim(sero.posit.over10) # ANSWER: ---------------------- REACTIVE ANTIGENS + SEROPREVALENCE + MEAN magnitude PER ANTIGEN

## normal.log2.raw.eSet.VIVAX.PQ    # 515 features X 140 samples	=> 515= 252 reactive + 263 non-reactive
## normal.log2.raw.eSet.FALCIP.PQ   # 499 features X 140 samples	
## normal.log2.raw.eSet.VIVAX.SEV   # 515 features X 60 samples   => 515= 255 reactive + 260 non-reactive
## normal.log2.raw.eSet.FALCIP.SEV  # 499 features X 60 samples	  

sero.posit.over10.LIST <- rownames(sero.posit.over10)
length(sero.posit.over10.LIST)
#sero.posit.over10.LIST         #---------------> FOR BREADTH
#

kable(head(sero.posit.over10), align = "c")

```

### Magnitud and Breadth 

**per Sample**
```{r}
########### (2) [PROGRESS 6/6] BREADTH* and INTENSITY of ANTIBODY RESPONSES (index per sample [column]) *(Helb 2015)
###########     Q: Is there any reason to use an alternative test, taking in account the data dispersion difference between groups?
###########     A: Yes. The F-test have been used to compare two variances

#head(filter.eset)               # LOGICAL statement result
length(sero.posit.over10.LIST)

breadth.filter.eset <- filter.eset[sero.posit.over10.LIST,]

## - BREADTH "SCORE" equals the SUM of REACTIVE ANTIGENS per sample                                        (Campo 2015)    -> POISSON REGRESSION for comparison
#                                                                                                          (Arama 2015)    -> BINOMIAL-GENERALIZED LINEAR MODEL for comparison
breadth.SCORE.filter.eset<-as.data.frame(apply(breadth.filter.eset, 2, sum, na.rm = TRUE))
colnames(breadth.SCORE.filter.eset) <- c("breadth.score")

## - PERCENTAGE of REACTIVE ANTIGENS per sample (X*100/# total RVE ANTIGENS, X= RVE ANTIGENS per sample)   (Skinner 2015)  -> LINEAR-MIXED MODEL for comparison
#                                                                                                          (Helb 2015)     -> MANN-WHITNEY (Bonferoni-corrected) for comparison
breadth.PERCENTAGE.filter.eset <- breadth.SCORE.filter.eset*100/length(sero.posit.over10.LIST)
colnames(breadth.PERCENTAGE.filter.eset) <- c("breadth.percentage")

## - MEAN INTENSITY of ANTIBODY RESPONSES to ALL RVE-Ag per sample (index per sample [column])             (Helb 2015)     -> MANN-WHITNEY (Bonferoni-corrected) for comparison
#
sample_mean<-as.data.frame(apply(exprs(eset)[sero.posit.over10.LIST,], 2, mean, na.rm = TRUE))
colnames(sample_mean) <- c("mean.intensity")
#


## - SUMMARY TABLE and add it to the pData(eset) DATA.FRAME
pData(eset) <- cbind(pData(eset),breadth.SCORE.filter.eset,breadth.PERCENTAGE.filter.eset,sample_mean)
#
#write.csv(pData(eset), "SEVERE.breadth.SUMMARY.csv")
#

#
# GRAPHIC: Bee-swarm scatter plot
#summary(subset(pData(eset),Group=="D", breadth.score:mean.intensity))
#summary(subset(pData(eset),Group=="E", breadth.score:mean.intensity))
# 03ago2016
eset$Study <- factor(eset$Study)
eset$Group <- factor(eset$Group)
eset$Sample.Type <- factor(eset$Sample.Type)
#summary(pData(eset))
kable(head(subset(pData(eset),select= c(Group,breadth.score:mean.intensity))), align = "c")
#kable(summary(subset(pData(eset),select= c(Group,breadth.score:mean.intensity))), align = "c")
# boxplot(y:numeric ~ grp:factor, data= data.frame)
esetEXTRA <- eset
############################## 
```

#### Correlations
```{r, fig.height=6, fig.width=14, fig.align='center', warning=FALSE, message=FALSE}
corrX <- subset(pData(esetEXTRA),select= c(casecontrol,con_exp,edad:mean.intensity))
kable(head(corrX[order(corrX$edad, decreasing = TRUE),]), align = "c")

#corrX
EdadBreadth1 <- ggplot(corrX, aes(x=edad,y=breadth.score, colour=casecontrol)) + geom_point() + geom_smooth(alpha=.2) #+ ggtitle("xx") + geom_line() + # group=con_exp
#ggplot(corrX, aes(x=edad,y=breadth.percentage, colour=casecontrol)) + geom_point() + geom_smooth(alpha=.2,size=.1) #+ ggtitle("xx") + geom_line() + # group=con_exp
EdadBreadth2 <- ggplot(corrX, aes(x=edad,y=mean.intensity, colour=casecontrol)) + geom_point() + geom_smooth(alpha=.2) #+ ggtitle("xx") + geom_line() + # group=con_exp

EdadBreadth3 <- ggplot(corrX, aes(x=edad,y=breadth.score, colour=con_exp)) + geom_point() + geom_smooth(alpha=.2) #+ ggtitle("xx") + geom_line() + # group=con_exp
#ggplot(corrX, aes(x=edad,y=breadth.percentage, colour=casecontrol)) + geom_point() + geom_smooth(alpha=.2,size=.1) #+ ggtitle("xx") + geom_line() + # group=con_exp
EdadBreadth4 <- ggplot(corrX, aes(x=edad,y=mean.intensity, colour=con_exp)) + geom_point() + geom_smooth(alpha=.2) #+ ggtitle("xx") + geom_line() + # group=con_exp


multiplot(EdadBreadth1,EdadBreadth3,EdadBreadth2,EdadBreadth4,cols=2)
```

```{r, fig.height=6, fig.width=14, fig.align='center', warning=FALSE, message=FALSE}
corrX <- subset(pData(esetEXTRA),select= c(casecontrol,con_exp,edad_CAT:mean.intensity))
#kable(head(corrX[order(corrX$edad, decreasing = TRUE),]), align = "c")

#corrX
EdadBreadth1 <- ggplot(corrX, aes(x=edad_CAT,y=breadth.score, fill=casecontrol)) +geom_boxplot() #+ geom_point() + geom_smooth(alpha=.2) #+ ggtitle("xx") + geom_line() + # group=con_exp
#ggplot(corrX, aes(x=edad,y=breadth.percentage, colour=casecontrol)) + geom_point() + geom_smooth(alpha=.2,size=.1) #+ ggtitle("xx") + geom_line() + # group=con_exp
EdadBreadth2 <- ggplot(corrX, aes(x=edad_CAT,y=mean.intensity, fill=casecontrol)) +geom_boxplot() # + geom_point() + geom_smooth(alpha=.2) #+ ggtitle("xx") + geom_line() + # group=con_exp

EdadBreadth3 <- ggplot(corrX, aes(x=edad_CAT,y=breadth.score, fill=con_exp)) +geom_boxplot() # + geom_point() + geom_smooth(alpha=.2) #+ ggtitle("xx") + geom_line() + # group=con_exp
#ggplot(corrX, aes(x=edad,y=breadth.percentage, colour=casecontrol)) + geom_point() + geom_smooth(alpha=.2,size=.1) #+ ggtitle("xx") + geom_line() + # group=con_exp
EdadBreadth4 <- ggplot(corrX, aes(x=edad_CAT,y=mean.intensity, fill=con_exp)) +geom_boxplot() # + geom_point() + geom_smooth(alpha=.2) #+ ggtitle("xx") + geom_line() + # group=con_exp


multiplot(EdadBreadth1,EdadBreadth3,EdadBreadth2,EdadBreadth4,cols=2)
```


***

# 5. Group comparison  
*HIGHLY and DIFFERENCIALLY reactive antigens*

**should have:**

- [analysis] one-side empirical Bayes (*unpaired*) moderated T-test **applying `limma` package**[@limma]
- [visual] bar plots + 95% CI of top ranked antigens by UN-ADJUSTED p_values -----------------------------> justification?
- [visual] "volcano" plots of fold changes between groups (x-axis) by inverse log10 p_values (y-axis)

*BELOW protocol is not required ------------------------------> BUT, When do adjusted p_values for FDR should be applied?*

- [analysis] to identify ASSOCIATION between antigens and trait, 
             AUC per antigen on discriminatory performance 
             with Wilcoxon's Rank Sum Test for approximated significance
             controling for multiple comparisons (?) using Benjamini-Hochbenge (BH) method of ADJUSTING p_values for false discovery rate (FDR) ----> justification?

**justification?**

FROM SAP of ADi & N6 (Analytical plan: 2. first-level analysis)

High-dimensional data can (will) produce false discoveries. 
Therefore, p-values will be adjusted for the false discovery rate 
using the Benjamini-Hochberg method in these comparisons and all multi-dimensional comparisons, henceforth. 
P-value adjustments will be performed separately for each type of test 
(e.g. eBayes T tests, Wilcoxon rank sum tests and regression).

**possible methods to be applied**

1. genefilter ------------------------> traditional t-test
2. limma -----------------------------> LINEAR MODEL + empirical Bayes t-test + BH adjusted p_values
3. PAA -------------------------------> allows all required? ---> FALSE DISCOVERY RATE (FDR) graph
4. DESeq or edgeR (for RNA-seq) ------> GENERALIZED LINEAR MODELS // MORE USED with count data (RNA-seq)


Take a look on the sample covariates or metadata
```{r, warning=FALSE, eval=FALSE, echo=FALSE}
varMetadata(eset)
```

***

## I. Grouped by Symptomatology

**Restricted to Pv 500 chip**

### Heatmap

```{r}
# [START FROM THE BEGINNNING]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
#eset$Group

#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#


########### GROUP "E"
m_data <- exprs(eset[,pData(eset)$Group=="E"])

m_data<-as.data.frame(m_data)

col.mean.m_data<-t(as.matrix(apply(m_data, 2, mean, na.rm=TRUE)))
rownames(col.mean.m_data) <- c("col.mean.m")                 # assign row names 
#col.mean.m_data

#

m_data_col.row.mean<-rbind(m_data,col.mean.m_data)

m_data_col.row.mean<-t(m_data_col.row.mean)
m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)

m_data_col.row.mean<-m_data_col.row.mean[order(m_data_col.row.mean$col.mean.m),]

m_data_col.row.mean$col.mean.m<-NULL
m_data_col.row.mean<-t(m_data_col.row.mean)

m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)



##### sample sorting for GROUP "E"
Ecol<-colnames(m_data_col.row.mean)



########### GROUP "D"
m_data <- exprs(eset[,pData(eset)$Group=="D"])

m_data<-as.data.frame(m_data)

col.mean.m_data<-t(as.matrix(apply(m_data, 2, mean, na.rm=TRUE)))
rownames(col.mean.m_data) <- c("col.mean.m")                 # assign row names 
#col.mean.m_data

#

m_data_col.row.mean<-rbind(m_data,col.mean.m_data)

m_data_col.row.mean<-t(m_data_col.row.mean)
m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)

m_data_col.row.mean<-m_data_col.row.mean[order(m_data_col.row.mean$col.mean.m),]

m_data_col.row.mean$col.mean.m<-NULL
m_data_col.row.mean<-t(m_data_col.row.mean)

m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)


##### sample sorting for GROUP "D" 
Dcol<-colnames(m_data_col.row.mean)


##### CONCATENATE character strings
sample_sort_mean<-c(Ecol,Dcol)


##############################
eset <- eset[feature_sort_mean,sample_sort_mean] ## REORDER samples and features inside the ExpressionSet

#eset$Group
```

```{r, fig.height=8, fig.width=8, fig.align="center", echo=FALSE, eval=FALSE}
#
color.map <- function(Group) { if (Group=="D") "#ff0000" else "#00ffff" }   # GROUP D is RED!!!!
patientcolors.SEV <- unlist(lapply(eset$Group, color.map))
table(patientcolors.SEV)
#


# [JUST A TEST]
#blood <- colorRampPalette(c("white", "#ff0000"))(n = 8)
# [ALTERNATIVE 28jul2016]
#blood <- colorRampPalette(c("#ffe5e5", "#ff0000"))(n = 8)


##### THIS IS THE RIGHT HEATMAP (with ORDERED groups)
heatmap.2(exprs(eset), 
          
          Rowv = FALSE,
          Colv = FALSE,
          dendrogram = "none",
          
          col=bluered(10), 
          scale = "none", 
          ColSideColors = patientcolors.SEV, 
          key = TRUE, symkey = FALSE,
          density.info = "histogram", denscol = "black", 
          trace = "none", cexRow = 0.5)

par(lend = 1)           # square line ends for the color legend
legend("topright",      # location of the legend on the heatmap plot
    legend = c("Severe", "Non-severe"), # category labels
    col = c("#ff0000", "#00ffff"),  # color key
    lty= 1,             # line style
    lwd = 10            # line width
)
#
```

**using `NMF` package**[@Gaujoux2010NMF]
```{r, eval=TRUE, fig.height=8, fig.width=8, fig.align="center"}
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]
#library(NMF)
aheatmap(exprs(eset), Rowv = NA, Colv = NA, annCol = pData(eset))
```

```{r, eval=FALSE, echo=FALSE}
#
png("NEW_SEV_vivax_FILT_GroupOrdered_21jul2016.png", width = 3000, height = 3000, res = 300, pointsize = 8)
heatmap.2(exprs(eset), 
          
          Rowv = FALSE,
          Colv = FALSE,
          dendrogram = "none",
          
          col=blood, 
          scale = "none", 
          ColSideColors = patientcolors.SEV, 
          key = TRUE, symkey = FALSE,
          density.info = "histogram", denscol = "black", 
          trace = "none", cexRow = 0.5)
dev.off()

#
png("SEV_vivax_FILT_neg2zero_GroupOrdered_RED2WHITE_28jul2016.png", width = 3000, height = 3000, res = 300, pointsize = 8)
heatmap.2(exprs(eset), 
          
          Rowv = FALSE,
          Colv = FALSE,
          dendrogram = "none",
          
          col=blood, 
          scale = "none", 
          ColSideColors = patientcolors.SEV, 
          key = TRUE, symkey = FALSE,
          density.info = "histogram", denscol = "black", 
          trace = "none", cexRow = 0.5)
dev.off()
#
```

### Magnitud, Frequency and Breadth

**Magnitud and Frequency of Response per Group (per feature)**
```{r}
# (1.4) SUMMARY TABLE: 
# ID /Description / #Rve=A+B / % sero.prevalence / mean.rve.intensity / #A=Case / % A.sero.prevalence / A.mean.rve.intensity / #B=Control / % B.sero.prevalence / B.mean.rve.intensity /

eset <- esetEXTRA

sero.posit.SUMMARY <- sero.posit
#head(sero.posit.SUMMARY) # ALL the 515 antigens
#rownames(sero.posit.SUMMARY)

#dim(filter.eset)
#colnames(filter.eset)

eset$Group <- factor(eset$Group)

eset.D<-eset[,phenoData(eset)$Group==levels(phenoData(eset)$Group)[1]] # ALWAYS is going to be in alphabetic order
eset.E<-eset[,phenoData(eset)$Group==levels(phenoData(eset)$Group)[2]]

eset.D.samples <- sampleNames(eset.D)
eset.E.samples <- sampleNames(eset.E)

filter.eset.D <- filter.eset[rownames(sero.posit.SUMMARY),eset.D.samples]
filter.eset.E <- filter.eset[rownames(sero.posit.SUMMARY),eset.E.samples]
#
filter.eset.SUB<-filter.eset.D               # LOGICAL statement result

##---- SEROPREVALENCE function WITHOUT ORDER
sero.posit<-as.matrix(apply(filter.eset.SUB, 1, sero.percent))

# order [DONE]
sero.posit<-as.data.frame(sero.posit)
colnames(sero.posit) <- c("D.sero.prevalence") 

sero_mean<-as.matrix(apply(exprs(eset)[rownames(sero.posit.SUMMARY),eset.D.samples], 1, mean, na.rm = TRUE))
colnames(sero_mean) <- c("D.mean.intensity")

sero_sum<-as.matrix(apply(filter.eset.SUB, 1, sum, na.rm = TRUE))
colnames(sero_sum) <- c("D.rve.samples")

sero.posit<-cbind(sero_sum,sero.posit,sero_mean)
#head(sero.posit)

#
sero.posit.SUMMARY<-cbind(sero.posit.SUMMARY,sero.posit)
#head(sero.posit.SUMMARY) # ALL the 515 antigens
#

######

filter.eset.SUB<-filter.eset.E               # LOGICAL statement result

##---- SEROPREVALENCE function WITHOUT ORDER
sero.posit<-as.matrix(apply(filter.eset.SUB, 1, sero.percent))

# order [DONE]
sero.posit<-as.data.frame(sero.posit)
colnames(sero.posit) <- c("E.sero.prevalence") 

sero_mean<-as.matrix(apply(exprs(eset)[rownames(sero.posit.SUMMARY),eset.E.samples], 1, mean, na.rm = TRUE))
colnames(sero_mean) <- c("E.mean.intensity")

sero_sum<-as.matrix(apply(filter.eset.SUB, 1, sum, na.rm = TRUE))
colnames(sero_sum) <- c("E.rve.samples")

sero.posit<-cbind(sero_sum,sero.posit,sero_mean)
#head(sero.posit)

#
sero.posit.SUMMARY<-cbind(sero.posit.SUMMARY,sero.posit)
#kable(head(sero.posit.SUMMARY[,1:6]), align = "c") # ALL the 515 antigens-----------------> HERE is the COMPLETE ANSWER
#kable(head(sero.posit.SUMMARY[,c(1:3,7:9)]), align = "c") # ALL the 515 antigens-----------------> HERE is the COMPLETE ANSWER
kable(head(sero.posit.SUMMARY[,4:9]), align = "c") # ALL the 515 antigens-----------------> HERE is the COMPLETE ANSWER
#
#write.csv(sero.posit.SUMMARY, "SEVERE.sero.posit.SUMMARY.csv")

######
```
**Frequency of Response per Group (per feature)**

*No se rechaza la $H_{o}$ de igualdad de medias. No se rechaza la $H_{o}$ de igualdad de varianzas*
```{r, fig.height=4, fig.width=16, message=FALSE}
# (1.5) GRAPHIC: Venn-diagram --> [MANUALLY done]
#
# (1.6) TEST DIFFERENCES
#
#kable(summary(sero.posit.SUMMARY))
#kable(summary(sero.posit.SUMMARY[sero.posit.over10.LIST,]))
#
#boxplot(sero.posit.SUMMARY[,c("D.rve.samples","E.rve.samples")], main= "Seropositive Antigens", ylab= "number of reactive samples per antigen")
#stripchart(sero.posit.SUMMARY[,c("D.rve.samples","E.rve.samples")], pch = 1, vertical = TRUE, add = TRUE, method = "jitter")
#
#boxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.rve.samples","E.rve.samples")], main= "Reactive Antigens", ylab= "number of reactive samples per antigen")
#stripchart(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.rve.samples","E.rve.samples")], pch = 1, vertical = TRUE, add = TRUE, method = "jitter")

############################## 
#
#png("SEVERE_y_summary_seroprevalence_MeanInt.png",    # create PNG for the heat map        
#    width = 3000,
#    height = 1500,        # DIMENSIONS of final image: 1500x1500 pixels
#    res = 300,            # 300 pixels per inch (RESOLUTION)
#    pointsize = 8)        # smaller font size
#
par(mfrow=c(1,4))
#
boxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.sero.prevalence","E.sero.prevalence")], 
        main= "Seroprevalence of Reactive Antigens", ylab= "percentage of reactive samples per antigen")
stripchart(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.sero.prevalence","E.sero.prevalence")],
           pch =1 , vertical = TRUE, add = TRUE, method = "jitter", col = c(2,4))
beeswarm(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.sero.prevalence","E.sero.prevalence")], col = c(2,4))
bxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.sero.prevalence","E.sero.prevalence")], add = TRUE)
vioplot(sero.posit.SUMMARY[sero.posit.over10.LIST,]$D.sero.prevalence, sero.posit.SUMMARY[sero.posit.over10.LIST,]$E.sero.prevalence, col = "grey80", names = c("D","E"))
beanplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.sero.prevalence","E.sero.prevalence")], log = "", overallline ="median", maxstripline = 0.16)
# TEST for differences
wtSevFREQ1 <- wilcox.test(sero.posit.SUMMARY[sero.posit.over10.LIST,"D.sero.prevalence"],sero.posit.SUMMARY[sero.posit.over10.LIST,"E.sero.prevalence"],exact = FALSE, correct = FALSE)
varSevFREQ1 <- var.test(sero.posit.SUMMARY[sero.posit.over10.LIST,"D.sero.prevalence"],sero.posit.SUMMARY[sero.posit.over10.LIST,"E.sero.prevalence"])

kable(tidy(wtSevFREQ1), align = "c")
kable(glance(varSevFREQ1), align = "c")
############################## 
```
**Magnitud of Response per Group (per feature)**

*No se rechaza la $H_{o}$ de igualdad de medias. No se rechaza la $H_{o}$ de igualdad de varianzas*
```{r, fig.height=4, fig.width=16, message=FALSE}
#
par(mfrow=c(1,4))
#
boxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.mean.intensity","E.mean.intensity")], 
        main= "Mean Intensity of Reactive Antigens", ylab= "mean intensity of reactive antigens")
stripchart(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.mean.intensity","E.mean.intensity")], 
           pch = 1, vertical = TRUE, add = TRUE, method = "jitter", col = c(2,4))
beeswarm(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.mean.intensity","E.mean.intensity")], col = c(2,4))
bxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.mean.intensity","E.mean.intensity")], add = TRUE)
vioplot(sero.posit.SUMMARY[sero.posit.over10.LIST,]$D.mean.intensity,sero.posit.SUMMARY[sero.posit.over10.LIST,]$E.mean.intensity, col = "grey80", names = c("D","E"))
beanplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.mean.intensity","E.mean.intensity")], log = "", overallline ="median")
#
#dev.off()
#
# TEST for differences
wtSevMAG1 <- wilcox.test(sero.posit.SUMMARY[sero.posit.over10.LIST,"D.mean.intensity"],sero.posit.SUMMARY[sero.posit.over10.LIST,"E.mean.intensity"],exact = FALSE, correct = FALSE)
varSevMAG1 <- var.test(sero.posit.SUMMARY[sero.posit.over10.LIST,"D.mean.intensity"],sero.posit.SUMMARY[sero.posit.over10.LIST,"E.mean.intensity"])

kable(tidy(wtSevMAG1), align = "c")
kable(glance(varSevMAG1), align = "c")
############################## 
#
# [PENDING TO KEEP TRYING]
# THIS IS NOT WORKING: http://stackoverflow.com/questions/14975853/how-can-i-create-violin-plot-in-different-colours
# ggplot2 BETTER: http://www.sthda.com/english/wiki/ggplot2-violin-plot-quick-start-guide-r-software-and-data-visualization
#
```


**Breadth (percentage) of Response per Group (per sample)**

*No se rechaza la $H_{o}$ de igualdad de medias. Sí se rechaza la $H_{o}$ de igualdad de varianzas*
```{r, fig.height=4, fig.width=16, message=FALSE}
# GRAPHIC: Bee-swarm scatter plot
#summary(subset(pData(esetEXTRA),Group=="D", breadth.score:mean.intensity))
#summary(subset(pData(esetEXTRA),Group=="E", breadth.score:mean.intensity))
#
#png("SEVERE_x_summary_breadth_intensity.png",    # create PNG for the heat map        
#    width = 3000,
#    height = 1500,        # DIMENSIONS of final image: 1500x1500 pixels
#    res = 300,            # 300 pixels per inch (RESOLUTION)
#    pointsize = 8)        # smaller font size
#
par(mfrow=c(1,4))
#
boxplot(breadth.percentage ~ Group, data= pData(eset), main= "Breadth of Response", ylab= "Percentage of reactive antigens per sample")
stripchart(breadth.percentage ~ Group, data= pData(eset), pch = 1, vertical = TRUE, add = TRUE, method = "jitter", col = c(2,4))
boxplot(breadth.percentage ~ Group, data= pData(eset), main= "Breadth of Response", ylab= "Percentage of reactive antigens per sample")
beeswarm(breadth.percentage ~ Group, data= pData(eset), col = c(2,4), add=TRUE) #, pwcol = 
vioplot(subset(pData(eset),Group=="D")$breadth.percentage,subset(pData(eset),Group=="E")$breadth.percentage, col = "grey80", names = c("D","E"))
beanplot(breadth.percentage ~ Group, data= pData(eset), overallline ="mean")
# TEST for differences
wtSev1 <- wilcox.test(subset(pData(eset),Group=="D")$breadth.percentage,subset(pData(eset),Group=="E")$breadth.percentage,exact = FALSE, correct = FALSE)
varSev1 <- var.test(subset(pData(eset),Group=="D")$breadth.percentage,subset(pData(eset),Group=="E")$breadth.percentage)

kable(tidy(wtSev1), align = "c")
kable(glance(varSev1), align = "c")
############################## 
```
**Breadth (intensity) of Response per Group (per sample)**

*No se rechaza la $H_{o}$ de igualdad de medias. Sí se rechaza la $H_{o}$ de igualdad de varianzas*
```{r, fig.height=4, fig.width=16, message=FALSE}
#
par(mfrow=c(1,4))
#
boxplot(mean.intensity ~ Group, data= pData(eset), main= "Intensity of Response", ylab= "Mean Intensity of reactive antigens per sample")
stripchart(mean.intensity ~ Group, data= pData(eset), pch = 1, vertical = TRUE, add = TRUE, method = "jitter", col = c(2,4))
boxplot(mean.intensity ~ Group, data= pData(eset), main= "Intensity of Response", ylab= "Mean Intensity of reactive antigens per sample")
beeswarm(mean.intensity ~ Group, data= pData(eset), col = c(2,4), add = TRUE)
vioplot(subset(pData(eset),Group=="D")$mean.intensity,subset(pData(eset),Group=="E")$mean.intensity, col = "grey80", names = c("D","E"))
beanplot(mean.intensity ~ Group, data= pData(eset), overallline ="mean")
#
# TEST for differences
wtSev2 <- wilcox.test(subset(pData(eset),Group=="D")$mean.intensity,subset(pData(eset),Group=="E")$mean.intensity,exact = FALSE, correct = FALSE)
varSev2 <- var.test(subset(pData(eset),Group=="D")$mean.intensity,subset(pData(eset),Group=="E")$mean.intensity)
#
#dev.off()
#
kable(tidy(wtSev2), align = "c")
kable(glance(varSev2), align = "c")
############################## 
```


```{r, echo=FALSE, eval=FALSE}
################################## FOR EACH GROUP

# [ADDED 28jul2016]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
eset$Group
# [end 28jul2016]

########### GROUP "E"
m_data <- exprs(eset[,pData(eset)$Group=="E"])

m_data<-as.data.frame(m_data)

row.mean.m_data<-as.matrix(apply(m_data, 1, mean))
colnames(row.mean.m_data) <- c("row.mean.m")                 # assign row names 
Em<-row.mean.m_data

m_data_row.mean<-cbind(m_data,row.mean.m_data)
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean<-m_data_row.mean[order(-m_data_row.mean$row.mean.m),]

##### TOP10 OF HIGHER REACTIVITY in GROUP "E
head(rownames(m_data_col.row.mean), 10)
E<-head(rownames(m_data_col.row.mean), 10)
E2<-rownames(m_data_col.row.mean)





########### GROUP "D"
m_data <- exprs(eset[,pData(eset)$Group=="D"])

m_data<-as.data.frame(m_data)

row.mean.m_data<-as.matrix(apply(m_data, 1, mean))
colnames(row.mean.m_data) <- c("row.mean.m")                 # assign row names 
Dm<-row.mean.m_data

m_data_row.mean<-cbind(m_data,row.mean.m_data)
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean<-m_data_row.mean[order(-m_data_row.mean$row.mean.m),]

##### TOP10 OF HIGHER REACTIVITY in GROUP "D
head(rownames(m_data_col.row.mean), 10)
D<-head(rownames(m_data_col.row.mean), 10)
D2<-rownames(m_data_col.row.mean)



##### TOP10 comparison of BOTH GROUPS
##### compare TOP10 of BOTH GROUPS: sum of equality in position
sum(E==D)
sum(E2==D2)

##### compare TOP10 of BOTH GROUPS: sum of equality in mean(reactivity)
sum(Em==Dm, na.rm = TRUE)

##### compare TOP10 of BOTH GROUPS: sum of higher # of reactive antigens between groups
# THIS SHOULD BE CONFIRMED BY (i) breadth and (ii) volcano plots
sum(Em>=Dm, na.rm = TRUE) # NON.SEVERE higher mean than SEVERE = 90
sum(Dm>=Em, na.rm = TRUE) # SEVERE higher mean than NON.SEVERE = 164
#
```

### Moderated eBayes t-test

```{r}
############################################################################ SEVERE
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER      ## Samples vs Controls
#summary(pData(eset))

eset$Study <- factor(eset$Study)
table(eset$Study)

eset$Group <- factor(eset$Group)
#table(eset$Group)

#varMetadata(eset) # D and E still unknown!!!!!!

#

library(limma)
######################
eset$Group <- factor(eset$Group)
# > eset$Sample.Type
# Levels: ALL NoL

#table(eset$Group)

# RELEVEL --> the first level is the REFERENCE level
eset$Group <- relevel(eset$Group, ref = "E") # CONTROL: E (NOT-SEVERE) -> CASE: D (severe)
# > eset$Sample.Type
# Levels: NoL ALL
table(eset$Group)
#

# perform the standart limma model
design <- model.matrix(~eset$Group)    # design the experiment
fit <- lmFit(eset, design)                    # fit the model
fit <- eBayes(fit)                               # empirical Bayes for moderate t-statistics
#topTable(fit)                                    # evaluate
#######################

# [DEFAULT sorting] "B" for the lods or B-statistic
#topTable(fit, coef = 2, genelist = fit$genes$Description)


```
**Differentially reactive features**
```{r}
#-----------standard-----------
# DIFFERENTIALLY rve [sorted by p.value] (SAME as default)
#topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$ID, number = 23)
td <- topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description)
td <- format(td, digits=3)
#htmlTable(td, align = "c", col.rgroup = c("none", "#F7F7F7"), col.columns = c("none", "#F7F7F7"))
kable(td, align = "c")
#
ta <- topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description, number = Inf)
ta <- format(ta, digits=3)
```
**Highly reactive features**
```{r, eval=TRUE}
# HIGHLY rve [sorted by average expression level (over all arrays) in descending order]
#topTable(fit, coef = 2, sort.by ="AveExpr", genelist = fit$genes$Description)
t <- topTable(fit, coef = 2, sort.by ="AveExpr", genelist = fit$genes$Description)
t <- format(t, digits=3)
#htmlTable(t, align = "c", col.rgroup = c("none", "#F7F7F7"), col.columns = c("none", "#F7F7F7"))
kable(t, align = "c")
#
ha <- topTable(fit, coef = 2, sort.by ="AveExpr", genelist = fit$genes$Description, number = Inf)
ha <- format(ha, digits=3)
#------------------------------
```

```{r, eval=FALSE}
#-----------explore------------
# DIFFERENTIALLY rve ONLY [sorted by FOC in descending order, without considering p.value]
topTable(fit, coef = 2, sort.by ="logFC", genelist = fit$genes$Description)
# FIRST sort by default (p_value) and THEN by average
topTable(fit, coef = 2, resort.by ="AveExpr", genelist = fit$genes$Description)
# FIRST sort by default (p_value) and THEN by the (absolute) coefficient representing the log-fold-change
topTable(fit, coef = 2,resort.by ="logFC", genelist = fit$genes$Description)
# to sort by the AVERAGE EXPRESSION LEVEL, then resort by LOG-FOLD-CHANGE
topTable(fit, coef = 2, sort.by ="AveExpr", resort.by ="logFC", genelist = fit$genes$Description)
#------------------------------
```
**p_value distribution [required evaluation prior to FDR]**
```{r, eval=FALSE, fig.align="center", fig.height=2, fig.width=4}
# [31ago2016] before any FDR in MULTIPLE COMPARISONS
# suggested by: http://varianceexplained.org/statistics/interpreting-pvalue-histogram/
test <- topTable(fit, coef = 2,
                 sort.by ="P", 
                 #sort.by ="AveExpr", 
                 genelist = fit$genes$Gene.ID, number = Inf)
library(ggplot2)
ax <- qplot(test$P.Value, binwidth=.05)
bx <- qplot(test$adj.P.Val, binwidth=.05)
multiplot(ax,bx,cols = 2)
```

```{r, eval=FALSE, fig.align="center"}
test2 <- -log10(as.matrix(test[,5]))
rownames(test2) <- as.matrix(test[,1])
colnames(test2) <- c("p.values")
test3 <- t(test2)
barplot(test3, las=2, cex.names = .5)

test4 <- test3[,test3>-log10(0.05)]
barplot(test4, las=2, cex.names = .8)
#barplot(test[,"adj.P.Val"], las=2)
#

# P_adj FILTERING
selected  <- p.adjust(fit$p.value[, 2]) <0.05
esetSel <- eset [selected, ]
#
featureNames(esetSel) 
#
```


### Summary plots

```{r, eval=FALSE, echo=FALSE}
#### NEW FEATURES: FROM help(lmFit) [START]

# Ordinary fit
#fit <- lmFit(y,design)
#fit <- eBayes(fit)
#topTable(fit,coef=2)
dim(fit)
colnames(fit)
rownames(fit)[1:10]
names(fit)

# Fold-change thresholding
fit2 <- treat(fit,lfc=0.1)
topTreat(fit2,coef=2)

# Volcano plot
volcanoplot(fit,coef=2,highlight=10,
            names=fit$genes$Description,
            pch = 19)

# Mean-difference plot
plotMD(fit,column=2)

# Q-Q plot of moderated t-statistics
qqt(fit$t[,2],df=fit$df.residual+fit$df.prior)
abline(0,1)


############### 

```

```{r, eval=FALSE, echo=FALSE}
# (1) Volcano plot: Log Fold Change x Log Odds (B)
volcanoplot(fit,coef=2,highlight=10,
            names=fit$genes$Description,
            pch = 19)

```

**Volcano plot I**
```{r, eval=TRUE, fig.align="center"}

# (2) Volcano plot: Log Fold Change x -log10(p.value)
#source: http://genomicsclass.github.io/book/pages/using_limma.html
h<-topTable(fit, coef = 2, sort.by =NULL, resort.by =NULL, genelist = NULL, number = Inf, confint=0.95)
h<-h[rownames(coef(fit)),]
limmares <- data.frame(Gene.ID=fit$genes$ID,logFC=coef(fit)[,2],AveExpr=h$AveExpr, p.value=fit$p.value[,2], padj=h$adj.P.Val, B=h$B)

#
par(mfrow=c(1,2))
with(limmares, plot(logFC, -log10(p.value),cex=.7, pch=20,
                    main = "Volcano plot",
                    #col=cols,
                    xlim=c(-1.5,1.5), ylim=c(0,12),
                    xlab="log Fold Change"))
abline(h=c(-log10(0.05),-log10(0.001)),
       v=c(-1,1),
       lty=2, col= "grey55")

# Add colored points: red if padj<0.05, orange of log2FC>1, green if both)
###source: http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html
#with(subset(limmares, p.value<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
#with(subset(limmares, abs(AveExpr)>1.738), points(logFC, -log10(p.value), pch=20, col="green"))
#with(subset(limmares, abs(AveExpr)>2), points(logFC, -log10(p.value), pch=20, col="orange"))
#with(subset(limmares, p.value<.0224), points(logFC, -log10(p.value), pch=19, col="red"))

with(subset(limmares, padj<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
#with(subset(limmares, p.value<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
with(subset(limmares, abs(AveExpr)>1.738), points(logFC, -log10(p.value), pch=20, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, -log10(p.value), pch=20, col="orange"))
with(subset(limmares, B>0 ), points(logFC, -log10(p.value), pch=20, col="red"))

# (2.2) Volcano plot: Log Fold Change x AveExpr
#source: http://genomicsclass.github.io/book/pages/using_limma.html

#
with(limmares, plot(logFC, AveExpr,cex=.7, pch=20,
                    main = "logFC x AveExpr plot",
                    #col=cols,
                    xlim=c(-1.5,1.5), #ylim=c(-1,2.5),
                    xlab="log Fold Change"))
abline(h=c(0,2,-2),
       v=c(-1,0,1),
       lty=2, col= "grey55")

# Add colored points: red if padj<0.05, orange of log2FC>1, green if both)
###source: http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html

#with(subset(limmares, p.value<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
with(subset(limmares, padj<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
with(subset(limmares, B>0 ), points(logFC, AveExpr, pch=20, col="red"))
#with(subset(limmares, abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="orange"))
#with(subset(limmares, p.value<.05 & abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="green"))

with(subset(limmares, abs(AveExpr)>1.738), points(logFC, AveExpr, pch=19, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, AveExpr, pch=19, col="orange"))
#with(subset(limmares, p.value<.0224), points(logFC, AveExpr, pch=20, col="red"))






```

```{r, eval=FALSE, echo=FALSE}
# (2.3) SCATTER plot: -log10(p.value) x AveExpr
#source: http://genomicsclass.github.io/book/pages/using_limma.html

#
with(limmares, plot(AveExpr, -log10(p.value),cex=.7, pch=20,
                    main = "AveExpr x p.value plot",
                    #col=cols,
                    #xlim=c(-1,1), #ylim=c(-1,2.5),
                    xlab="mean reactivity"))

# Add colored points: red if padj<0.05, orange of log2FC>1, green if both)
###source: http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html
with(subset(limmares, p.value<.05 ), points(AveExpr, -log10(p.value), pch=19, col="blue"))
#with(subset(limmares, padj<.05 ), points(logFC, -log10(p.value), pch=20, col="red"))
#with(subset(limmares, abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="orange"))
#with(subset(limmares, p.value<.05 & abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="green"))

with(subset(limmares, abs(AveExpr)>1.738), points(AveExpr, -log10(p.value), pch=19, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(AveExpr, -log10(p.value), pch=19, col="orange"))
with(subset(limmares, p.value<.0224), points(AveExpr, -log10(p.value), pch=19, col="red"))


```
**Volcano plot II**

Using `dplyr`[@dplyr], `ggplot2`[@ggplot2], `ggrepel`[@ggrepel] and `Rmisc`[@Rmisc] packages
```{r, eval=TRUE, message=FALSE, warning=FALSE, fig.align="center", fig.height=8, fig.width=15}

# (3) Volcano plot: Log Fold Change x -log10(p.value) [W/ ANNOTATION]
###source: http://www.gettinggeneticsdone.com/2016/01/repel-overlapping-text-labels-in-ggplot2.html
#library(dplyr)
#library(ggplot2)

results <- mutate(limmares, sig=ifelse(limmares$p.value<.0224, "diff rve","below Top10"))

p <- ggplot(results, aes(logFC, -log10(p.value))) + 
  geom_point(aes(col=sig)) +
  scale_color_manual(values=c("black","red"))

#p
#p+geom_text(data=filter(results, p.value<.0224), aes(label=Gene.ID))


# Install ggrepel package if needed
# install.packages("devtools")
# devtools::install_github("slowkow/ggrepel")
#library(ggrepel)

#p+geom_text_repel(data=filter(results, p.value<.0224), aes(label=Gene.ID))+xlim(-1,1)
p_rep<-p+geom_text_repel(data=filter(results, p.value<.0224), aes(label=Gene.ID))#+xlim=c(-1.5,1.5)


# (3.2) Volcano plot: Log Fold Change x AveExprs [W/ ANNOTATION]
###source: http://www.gettinggeneticsdone.com/2016/01/repel-overlapping-text-labels-in-ggplot2.html
#library(dplyr)
#library(ggplot2)

resultsq <- mutate(limmares, sigq=ifelse(limmares$AveExpr>1.738, "highly rve","below Top10"))

q <- ggplot(resultsq, aes(logFC, AveExpr)) + 
  geom_point(aes(col=sigq)) +
  scale_color_manual(values=c("black","green"))

#q
#q+geom_text(data=filter(resultsq, AveExpr>1.738), aes(label=Gene.ID))



# Install ggrepel package if needed
# install.packages("devtools")
# devtools::install_github("slowkow/ggrepel")
#library(ggrepel)

#q+geom_text_repel(data=filter(resultsq, AveExpr>1.738), aes(label=Gene.ID))+xlim(-1,1)
q_rep<-q+geom_text_repel(data=filter(resultsq, AveExpr>1.738), aes(label=Gene.ID))#+xlim=c(-1.5,1.5)

#library(Rmisc)   #multiploting ggplots


# RECORDING images
multiplot(p_rep,q_rep,cols=2)


```

```{r, eval=FALSE, echo=FALSE}

#library(Rmisc)   #multiploting ggplots
# RECORDING images
png("VOLCANO_2_SEV_vivax_25jul2016.png", width = 8000, height = 3000, res = 300, pointsize = 16)
multiplot(p_rep,q_rep,cols=2)
dev.off()


#RECORDING IMAGES
png("VOLCANO_1_SEV_vivax_25jul2016.png", width = 2000, height = 1500, res = 300, pointsize = 8)
par(mfrow=c(1,2))
with(limmares, plot(logFC, -log10(p.value),cex=.7, pch=20,
                    main = "Volcano plot",
                    #col=cols,
                    xlim=c(-1,1), #ylim=c(0,5),
                    xlab="log Fold Change"))
abline(h=c(-log10(0.05),-log10(0.001)),
       v=c(-1,1),
       lty=2)

with(subset(limmares, p.value<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
with(subset(limmares, abs(AveExpr)>1.738), points(logFC, -log10(p.value), pch=20, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, -log10(p.value), pch=20, col="orange"))
with(subset(limmares, p.value<.0224), points(logFC, -log10(p.value), pch=19, col="red"))

# (2.2) Volcano plot: Log Fold Change x AveExpr
#source: http://genomicsclass.github.io/book/pages/using_limma.html

#
with(limmares, plot(logFC, AveExpr,cex=.7, pch=20,
                    main = "logFC x AveExpr plot",
                    #col=cols,
                    xlim=c(-1,1), #ylim=c(-1,2.5),
                    xlab="log Fold Change"))
abline(h=c(0,2,-2),
       v=c(-1,0,1),
       lty=2)

with(subset(limmares, p.value<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
with(subset(limmares, abs(AveExpr)>1.738), points(logFC, AveExpr, pch=19, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, AveExpr, pch=19, col="orange"))
with(subset(limmares, p.value<.0224), points(logFC, AveExpr, pch=20, col="red"))

dev.off()

# (ADi) Volcano plot: MEAN difference x -log10(p.value)
# xlab="mean(D=severe) - mean(E=control)"


#### NEW FEATURES: FROM help(lmFit) [END]

```

**Magnitud and Frequency per feature**

### Correlations 
**between Magnitud and Symptomatology**
```{r, eval=FALSE}
messz <- exprs(eset)[rownames(subset(limmares, p.value<.0022)),]
dim(t(messz))
doom <- pData(eset)[,c(9:ncol(pData(eset)))]
dim(doom)

corrSEV <- cbind(t(messz), doom)
#colnames(corrSEV)
corrSEV$ID <- rownames(corrSEV)
#str(corrSEV)
```

```{r, fig.height=2, fig.width=8, fig.align='center', warning=FALSE, message=FALSE, eval=FALSE}
#library(tidyr)
#library(dplyr)
CorrSev7 <- corrSEV %>%
  gather(feature, reactivity, PVX_092070_1o1.176:PVX_099080_2o2.439) %>%
  filter(casecontrol=="Severe") %>%
  ggplot(aes(reactivity, CumSumSevere7)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature)# + geom_line()
#CorrSev7

CorrSev10 <- corrSEV %>%
  gather(feature, reactivity, PVX_092070_1o1.176:PVX_099080_2o2.439) %>%
  filter(casecontrol=="Severe") %>%
  ggplot(aes(reactivity, CumSumSevere10)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature)# + geom_line()
#CorrSev10

multiplot(CorrSev7,CorrSev10, cols = 2)
```

```{r, fig.height=2, fig.width=8, fig.align='center', warning=FALSE, message=FALSE, eval=FALSE}
#library(tidyr)
#library(dplyr)
CorrSev7 <- corrSEV %>%
  gather(feature, reactivity, PVX_092070_1o1.176:PVX_099080_2o2.439) %>%
  filter(casecontrol=="Severe") %>%
  ggplot(aes(reactivity, CumSumSevere7)) + geom_point() + geom_smooth() + facet_wrap(~ feature)# + geom_line()
#CorrSev7

CorrSev10 <- corrSEV %>%
  gather(feature, reactivity, PVX_092070_1o1.176:PVX_099080_2o2.439) %>%
  filter(casecontrol=="Severe") %>%
  ggplot(aes(reactivity, CumSumSevere10)) + geom_point() + geom_smooth() + facet_wrap(~ feature)# + geom_line()
#CorrSev10

multiplot(CorrSev7,CorrSev10, cols = 2)
```

```{r, fig.height=2, fig.width=11, fig.align='center', warning=FALSE, message=FALSE, eval=FALSE}
CorrSev7x <- corrSEV %>%
  gather(feature, reactivity, PVX_092070_1o1.176:PVX_099080_2o2.439) %>%
  filter(casecontrol=="Severe") %>%
  ggplot(aes(reactivity, CumSumSevere7, group=con_exp, colour=con_exp)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature)# + geom_line()
#CorrSev7x

CorrSev10x <- corrSEV %>%
  gather(feature, reactivity, PVX_092070_1o1.176:PVX_099080_2o2.439) %>%
  filter(casecontrol=="Severe") %>%
  ggplot(aes(reactivity, CumSumSevere10, group=con_exp, colour=con_exp)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature)# + geom_line()
#CorrSev10x

#str(cleaned)
multiplot(CorrSev7x,CorrSev10x, cols = 2)
```

```{r, fig.height=2, fig.width=11, fig.align='center', warning=FALSE, message=FALSE, eval=FALSE}
CorrSev7x <- corrSEV %>%
  gather(feature, reactivity, PVX_092070_1o1.176:PVX_099080_2o2.439) %>%
  filter(casecontrol=="Severe") %>%
  ggplot(aes(reactivity, CumSumSevere7, group=con_exp, colour=con_exp)) + geom_point() + geom_smooth() + facet_wrap(~ feature)# + geom_line()
#CorrSev7x

CorrSev10x <- corrSEV %>%
  gather(feature, reactivity, PVX_092070_1o1.176:PVX_099080_2o2.439) %>%
  filter(casecontrol=="Severe") %>%
  ggplot(aes(reactivity, CumSumSevere10, group=con_exp, colour=con_exp)) + geom_point() + geom_smooth() + facet_wrap(~ feature)# + geom_line()
#CorrSev10x

#str(cleaned)
multiplot(CorrSev7x,CorrSev10x, cols = 2)
```


***

### Pf SEVERITY
*For breadth comparison*

### Pv PRIMAQUINE
*For methodology validation*
```{r, eval=FALSE, echo=FALSE}
############################################################################ PRIMAQUINE
eset<-normal.log2.raw.eSet.VIVAX.PQ.FILTER      ## Samples vs Controls
summary(pData(eset))

eset$Study <- factor(eset$Study)
table(eset$Study)

eset$Group <- factor(eset$Group)
table(eset$Group)

varMetadata(eset) # D and E still unknown!!!!!!

#

library(limma)
######################
eset$Group <- factor(eset$Group)
# > eset$Sample.Type
# Levels: ALL NoL

table(eset$Group)     #------->>>> HOW to make sense of this ORDER?????
# if A are recurrent group, then they should be more susceptible, isn´t it?
# but A was named the CASE GROUP
# and B was named the CONTROL GROUP




# RELEVEL --> the first level is the REFERENCE level    ------->>>> HOW to make sense of this ORDER?????
eset$Group <- relevel(eset$Group, ref = "B") # CONTROL: B (NON-recurrent) -> CASE: A (recurrent)
# > eset$Sample.Type
# Levels: NoL ALL
table(eset$Group)
#

# perform the standart limma model
design <- model.matrix(~eset$Group)    # design the experiment
fit <- lmFit(eset, design)                    # fit the model
fit <- eBayes(fit)                               # empirical Bayes for moderate t-statistics
topTable(fit)                                    # evaluate
#######################

# [DEFAULT sorting] "B" for the lods or B-statistic
topTable(fit, coef = 2, genelist = fit$genes$Description)

#-----------standard-----------
# DIFFERENTIALLY rve [sorted by p.value] (SAME as default)
topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description)
# HIGHLY rve [sorted by average expression level (over all arrays) in descending order]
topTable(fit, coef = 2, sort.by ="AveExpr", genelist = fit$genes$Description)
#------------------------------

#-----------explore------------
# DIFFERENTIALLY rve ONLY [sorted by FOC in descending order, without considering p.value]
topTable(fit, coef = 2, sort.by ="logFC", genelist = fit$genes$Description)
# FIRST sort by default (p_value) and THEN by average
topTable(fit, coef = 2, resort.by ="AveExpr", genelist = fit$genes$Description)
# FIRST sort by default (p_value) and THEN by the (absolute) coefficient representing the log-fold-change
topTable(fit, coef = 2,resort.by ="logFC", genelist = fit$genes$Description)
# to sort by the AVERAGE EXPRESSION LEVEL, then resort by LOG-FOLD-CHANGE
topTable(fit, coef = 2, sort.by ="AveExpr", resort.by ="logFC", genelist = fit$genes$Description)
#------------------------------



# [31ago2016] before any FDR in MULTIPLE COMPARISONS
# suggested by: http://varianceexplained.org/statistics/interpreting-pvalue-histogram/
test <- topTable(fit, coef = 2,
                 sort.by ="P", 
                 #sort.by ="AveExpr", 
                 genelist = fit$genes$Gene.ID, number = Inf)
library(ggplot2)
qplot(test$P.Value, binwidth=.05)
qplot(test$adj.P.Val, binwidth=.05)

test2 <- -log10(as.matrix(test[,5]))
rownames(test2) <- as.matrix(test[,1])
colnames(test2) <- c("p.values")
test3 <- t(test2)
barplot(test3, las=2, cex.names = .5)

test4 <- test3[,test3>-log10(0.05)]
barplot(test4, las=2, cex.names = .8)
#barplot(test[,"adj.P.Val"], las=2)
#




# P_adj FILTERING
selected  <- p.adjust(fit$p.value[, 2]) <0.05
esetSel <- eset [selected, ]

featureNames(esetSel) 


#### NEW FEATURES: FROM help(lmFit) [START]

# Ordinary fit
#fit <- lmFit(y,design)
#fit <- eBayes(fit)
#topTable(fit,coef=2)
dim(fit)
colnames(fit)
rownames(fit)[1:10]
names(fit)

# Fold-change thresholding
fit2 <- treat(fit,lfc=0.1)
topTreat(fit2,coef=2)

# Mean-difference plot
plotMD(fit,column=2)

# Q-Q plot of moderated t-statistics
qqt(fit$t[,2],df=fit$df.residual+fit$df.prior)
abline(0,1)

############### 

# (1) Volcano plot: Log Fold Change x Log Odds (B)
volcanoplot(fit,coef=2,highlight=10,
            names=fit$genes$Description,
            pch = 19)

# (2) Volcano plot: Log Fold Change x -log10(p.value)
#source: http://genomicsclass.github.io/book/pages/using_limma.html
h<-topTable(fit, coef = 2, sort.by =NULL, resort.by =NULL, genelist = NULL, number = Inf, confint=0.95)
h<-h[rownames(coef(fit)),]
limmares <- data.frame(Gene.ID=fit$genes$ID,logFC=coef(fit)[,2],AveExpr=h$AveExpr, p.value=fit$p.value[,2], padj=h$adj.P.Val)

#
par(mfrow=c(1,2))
with(limmares, plot(logFC, -log10(p.value),cex=.7, pch=20,
                    main = "Volcano plot",
                    #col=cols,
                    xlim=c(-1,1), #ylim=c(0,5),
                    xlab="log Fold Change"))
abline(h=c(-log10(0.05),-log10(0.001)),
       v=c(-1,1),
       lty=2)

# Add colored points: red if padj<0.05, orange of log2FC>1, green if both)
###source: http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html
with(subset(limmares, p.value<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
#with(subset(limmares, padj<.05 ), points(logFC, -log10(p.value), pch=20, col="red"))
#with(subset(limmares, abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="orange"))
#with(subset(limmares, p.value<.05 & abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="green"))

with(subset(limmares, abs(AveExpr)>1.865), points(logFC, -log10(p.value), pch=20, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, -log10(p.value), pch=20, col="orange"))
with(subset(limmares, p.value<.0384), points(logFC, -log10(p.value), pch=19, col="red"))

# (2.2) Volcano plot: Log Fold Change x AveExpr
#source: http://genomicsclass.github.io/book/pages/using_limma.html

#
with(limmares, plot(logFC, AveExpr,cex=.7, pch=20,
                    main = "logFC x AveExpr plot",
                    #col=cols,
                    xlim=c(-1,1), #ylim=c(-1,2.5),
                    xlab="log Fold Change"))
abline(h=c(0,2,-2),
       v=c(-1,0,1),
       lty=2)

# Add colored points: red if padj<0.05, orange of log2FC>1, green if both)
###source: http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html
with(subset(limmares, p.value<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
#with(subset(limmares, padj<.05 ), points(logFC, -log10(p.value), pch=20, col="red"))
#with(subset(limmares, abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="orange"))
#with(subset(limmares, p.value<.05 & abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="green"))

with(subset(limmares, abs(AveExpr)>1.865), points(logFC, AveExpr, pch=19, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, AveExpr, pch=19, col="orange"))
with(subset(limmares, p.value<.0384), points(logFC, AveExpr, pch=20, col="red"))



# (3) Volcano plot: Log Fold Change x -log10(p.value) [W/ ANNOTATION]
###source: http://www.gettinggeneticsdone.com/2016/01/repel-overlapping-text-labels-in-ggplot2.html
library(dplyr)
library(ggplot2)

results <- mutate(limmares, sig=ifelse(limmares$p.value<.0384, "diff rve","below Top10"))

p <- ggplot(results, aes(logFC, -log10(p.value))) + 
  geom_point(aes(col=sig)) +
  scale_color_manual(values=c("black","red"))

p
p+geom_text(data=filter(results, p.value<.0384), aes(label=Gene.ID))


# Install ggrepel package if needed
# install.packages("devtools")
# devtools::install_github("slowkow/ggrepel")
library(ggrepel)

p+geom_text_repel(data=filter(results, p.value<.0384), aes(label=Gene.ID))+xlim(-1,1)
p_rep<-p+geom_text_repel(data=filter(results, p.value<.0384), aes(label=Gene.ID))+xlim(-1,1)


# (3.2) Volcano plot: Log Fold Change x AveExprs [W/ ANNOTATION]
###source: http://www.gettinggeneticsdone.com/2016/01/repel-overlapping-text-labels-in-ggplot2.html
library(dplyr)
library(ggplot2)

resultsq <- mutate(limmares, sigq=ifelse(limmares$AveExpr>1.865, "highly rve","below Top10"))

q <- ggplot(resultsq, aes(logFC, AveExpr)) + 
  geom_point(aes(col=sigq)) +
  scale_color_manual(values=c("black","green"))

q
q+geom_text(data=filter(resultsq, AveExpr>1.865), aes(label=Gene.ID))



# Install ggrepel package if needed
# install.packages("devtools")
# devtools::install_github("slowkow/ggrepel")
library(ggrepel)

q+geom_text_repel(data=filter(resultsq, AveExpr>1.865), aes(label=Gene.ID))+xlim(-1,1)
q_rep<-q+geom_text_repel(data=filter(resultsq, AveExpr>1.865), aes(label=Gene.ID))+xlim(-1,1)




library(Rmisc)   #multiploting ggplots
# RECORDING images
png("VOLCANO_2_PQ_vivax_25jul2016.png", width = 8000, height = 3000, res = 300, pointsize = 16)
multiplot(p_rep,q_rep,cols=2)
dev.off()

png("VOLCANO_1_PQ_vivax_25jul2016.png", width = 2000, height = 1500, res = 300, pointsize = 8)
par(mfrow=c(1,2))
with(limmares, plot(logFC, -log10(p.value),cex=.7, pch=20,
                    main = "Volcano plot",
                    #col=cols,
                    xlim=c(-1,1), #ylim=c(0,5),
                    xlab="log Fold Change"))
abline(h=c(-log10(0.05),-log10(0.001)),
       v=c(-1,1),
       lty=2)

with(subset(limmares, p.value<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
with(subset(limmares, abs(AveExpr)>1.865), points(logFC, -log10(p.value), pch=20, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, -log10(p.value), pch=20, col="orange"))
with(subset(limmares, p.value<.0384), points(logFC, -log10(p.value), pch=19, col="red"))

# (2.2) Volcano plot: Log Fold Change x AveExpr
#source: http://genomicsclass.github.io/book/pages/using_limma.html

#
with(limmares, plot(logFC, AveExpr,cex=.7, pch=20,
                    main = "logFC x AveExpr plot",
                    #col=cols,
                    xlim=c(-1,1), #ylim=c(-1,2.5),
                    xlab="log Fold Change"))
abline(h=c(0,2,-2),
       v=c(-1,0,1),
       lty=2)

with(subset(limmares, p.value<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
with(subset(limmares, abs(AveExpr)>1.865), points(logFC, AveExpr, pch=19, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, AveExpr, pch=19, col="orange"))
with(subset(limmares, p.value<.0384), points(logFC, AveExpr, pch=20, col="red"))

dev.off()

# (ADi) Volcano plot: MEAN difference x -log10(p.value)
# xlab="mean(D=severe) - mean(E=control)"


#### NEW FEATURES: FROM help(lmFit) [END]

```


***

## II. Grouped by Exposure

**Restricted to Pv 500 chip**

### Heatmap
```{r}
# [MODIFIED 16sep2016]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
levels(eset$con_exp) <- c("CON","SIN") # CHANGE: D: CON exposicion, E: SIN exposicion
summary(eset$con_exp)

#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#


########### con_exp "SIN"
m_data <- exprs(eset[,pData(eset)$con_exp=="SIN"])

m_data<-as.data.frame(m_data)

col.mean.m_data<-t(as.matrix(apply(m_data, 2, mean, na.rm=TRUE)))
rownames(col.mean.m_data) <- c("col.mean.m")                 # assign row names 
#col.mean.m_data

#

m_data_col.row.mean<-rbind(m_data,col.mean.m_data)

m_data_col.row.mean<-t(m_data_col.row.mean)
m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)

m_data_col.row.mean<-m_data_col.row.mean[order(m_data_col.row.mean$col.mean.m),]

m_data_col.row.mean$col.mean.m<-NULL
m_data_col.row.mean<-t(m_data_col.row.mean)

m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)



##### sample sorting for con_exp "SIN"
Ecol<-colnames(m_data_col.row.mean)



########### con_exp "CON"
m_data <- exprs(eset[,pData(eset)$con_exp=="CON"])

m_data<-as.data.frame(m_data)

col.mean.m_data<-t(as.matrix(apply(m_data, 2, mean, na.rm=TRUE)))
rownames(col.mean.m_data) <- c("col.mean.m")                 # assign row names 
#col.mean.m_data

#

m_data_col.row.mean<-rbind(m_data,col.mean.m_data)

m_data_col.row.mean<-t(m_data_col.row.mean)
m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)

m_data_col.row.mean<-m_data_col.row.mean[order(m_data_col.row.mean$col.mean.m),]

m_data_col.row.mean$col.mean.m<-NULL
m_data_col.row.mean<-t(m_data_col.row.mean)

m_data_col.row.mean<-as.data.frame(m_data_col.row.mean)


##### sample sorting for con_exp "CON" 
Dcol<-colnames(m_data_col.row.mean)


##### CONCATENATE character strings
sample_sort_mean<-c(Ecol,Dcol)

##############################
eset <- eset[feature_sort_mean,sample_sort_mean] ## REORDER samples and features inside the ExpressionSet

#eset$con_exp
```

```{r, fig.height=8, fig.width=8, fig.align="center", echo=FALSE, eval=FALSE}
#
color.map <- function(con_exp) { if (con_exp=="CON") "#ff00ff" else "#00ff00" }   # con_exp D is RED!!!!
patientcolors.EXPO <- unlist(lapply(eset$con_exp, color.map))
table(patientcolors.EXPO)
#

# [JUST A TEST]
#blood <- colorRampPalette(c("white", "#ff0000"))(n = 8)
# [ALTERNATIVE 28jul2016]
#blood <- colorRampPalette(c("#ffe5e5", "#ff0000"))(n = 8)


##### THIS IS THE RIGHT HEATMAP (with ORDERED con_exps)
heatmap.2(exprs(eset), 
          
          Rowv = FALSE,
          Colv = FALSE,
          dendrogram = "none",
          
          col=bluered(10), 
          scale = "none", 
          ColSideColors = patientcolors.EXPO, 
          key = TRUE, symkey = FALSE,
          density.info = "histogram", denscol = "black", 
          trace = "none", cexRow = 0.5)

par(lend = 1)           # square line ends for the color legend
legend("topright",      # location of the legend on the heatmap plot
    legend = c("Con eventos previos", "Sin eventos previos"), # category labels
    col = c("#ff00ff", "#00ff00"),  # color key
    lty= 1,             # line style
    lwd = 10            # line width
)
#
```

**using `NMF` package**[@Gaujoux2010NMF]
```{r, eval=TRUE, fig.height=8, fig.width=8, fig.align="center"}
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]
#library(NMF)
aheatmap(exprs(eset), Rowv = NA, Colv = NA, annCol = pData(eset))
```

```{r, eval=FALSE, echo=FALSE}
#
png("NEW_SEV_vivax_FILT_con_expOrdered_21jul2016.png", width = 3000, height = 3000, res = 300, pointsize = 8)
heatmap.2(exprs(eset), 
          
          Rowv = FALSE,
          Colv = FALSE,
          dendrogram = "none",
          
          col=blood, 
          scale = "none", 
          ColSideColors = patientcolors.EXPO, 
          key = TRUE, symkey = FALSE,
          density.info = "histogram", denscol = "black", 
          trace = "none", cexRow = 0.5)
dev.off()

#
png("SEV_vivax_FILT_neg2zero_con_expOrdered_RED2WHITE_28jul2016.png", width = 3000, height = 3000, res = 300, pointsize = 8)
heatmap.2(exprs(eset), 
          
          Rowv = FALSE,
          Colv = FALSE,
          dendrogram = "none",
          
          col=blood, 
          scale = "none", 
          ColSideColors = patientcolors.EXPO, 
          key = TRUE, symkey = FALSE,
          density.info = "histogram", denscol = "black", 
          trace = "none", cexRow = 0.5)
dev.off()
#
```

```{r, echo=FALSE, eval=FALSE}
################################## FOR EACH con_exp

# [MODIFIED 16sep2016]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
levels(eset$con_exp) <- c("CON","SIN") # CHANGE: D: CON exposicion, E: SIN exposicion
summary(eset$con_exp)
# [end 16sep2016]

########### con_exp "SIN"
m_data <- exprs(eset[,pData(eset)$con_exp=="SIN"])

m_data<-as.data.frame(m_data)

row.mean.m_data<-as.matrix(apply(m_data, 1, mean))
colnames(row.mean.m_data) <- c("row.mean.m")                 # assign row names 
Em<-row.mean.m_data

m_data_row.mean<-cbind(m_data,row.mean.m_data)
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean<-m_data_row.mean[order(-m_data_row.mean$row.mean.m),]

##### TOP10 OF HIGHER REACTIVITY in con_exp "E
head(rownames(m_data_col.row.mean), 10)
E<-head(rownames(m_data_col.row.mean), 10)
E2<-rownames(m_data_col.row.mean)





########### con_exp "CON"
m_data <- exprs(eset[,pData(eset)$con_exp=="CON"])

m_data<-as.data.frame(m_data)

row.mean.m_data<-as.matrix(apply(m_data, 1, mean))
colnames(row.mean.m_data) <- c("row.mean.m")                 # assign row names 
Dm<-row.mean.m_data

m_data_row.mean<-cbind(m_data,row.mean.m_data)
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean<-m_data_row.mean[order(-m_data_row.mean$row.mean.m),]

##### TOP10 OF HIGHER REACTIVITY in con_exp "D
head(rownames(m_data_col.row.mean), 10)
D<-head(rownames(m_data_col.row.mean), 10)
D2<-rownames(m_data_col.row.mean)



##### TOP10 comparison of BOTH con_expS
##### compare TOP10 of BOTH con_expS: sum of equality in position
sum(E==D)
sum(E2==D2)

##### compare TOP10 of BOTH con_expS: sum of equality in mean(reactivity)
sum(Em==Dm, na.rm = TRUE)

##### compare TOP10 of BOTH con_expS: sum of higher # of reactive antigens between con_exps
# THIS SHOULD BE CONFIRMED BY (i) breadth and (ii) volcano plots
sum(Em>=Dm, na.rm = TRUE) # NON.SEVERE higher mean than SEVERE = 90
sum(Dm>=Em, na.rm = TRUE) # SEVERE higher mean than NON.SEVERE = 164
#
```

### Magnitud, Frequency and Breadth

**Magnitud and Frequency of Response per Group (per feature)**
```{r}

#
##### CAMBIAR A COMPARAR PERCENTILES SUPERIORES
#


# (1.4) SUMMARY TABLE: 
# ID /Description / #Rve=A+B / % sero.prevalence / mean.rve.intensity / #A=Case / % A.sero.prevalence / A.mean.rve.intensity / #B=Control / % B.sero.prevalence / B.mean.rve.intensity /

eset <- esetEXTRA

sero.posit.SUMMARY <- sero.posit
#head(sero.posit.SUMMARY) # ALL the 515 antigens
#rownames(sero.posit.SUMMARY)

#dim(filter.eset)
#colnames(filter.eset)

#eset$Group <- factor(eset$Group)
eset$con_exp <- relevel(eset$con_exp, ref = "Con exposicion") # CONTROL: E (NOT-SEVERE) -> CASE: D (severe)
levels(eset$con_exp) <- c("CON","SIN") # CHANGE: D: CON exposicion, E: SIN exposicion
summary(eset$con_exp)


eset.CON<-eset[,phenoData(eset)$con_exp==levels(phenoData(eset)$con_exp)[1]] # ALWAYS is going to be in alphabetic order
eset.SIN<-eset[,phenoData(eset)$con_exp==levels(phenoData(eset)$con_exp)[2]]

eset.CON.samples <- sampleNames(eset.CON)
eset.SIN.samples <- sampleNames(eset.SIN)

filter.eset.CON <- filter.eset[rownames(sero.posit.SUMMARY),eset.CON.samples]
filter.eset.SIN <- filter.eset[rownames(sero.posit.SUMMARY),eset.SIN.samples]
#
filter.eset.SUB<-filter.eset.CON               # LOGICAL statement result

##---- SEROPREVALENCE function WITHOUT ORDER
sero.posit<-as.matrix(apply(filter.eset.SUB, 1, sero.percent))

# order [DONE]
sero.posit<-as.data.frame(sero.posit)
colnames(sero.posit) <- c("CON.sero.prevalence") 

sero_mean<-as.matrix(apply(exprs(eset)[rownames(sero.posit.SUMMARY),eset.CON.samples], 1, mean, na.rm = TRUE))
colnames(sero_mean) <- c("CON.mean.intensity")

sero_sum<-as.matrix(apply(filter.eset.SUB, 1, sum, na.rm = TRUE))
colnames(sero_sum) <- c("CON.rve.samples")

sero.posit<-cbind(sero_sum,sero.posit,sero_mean)
#head(sero.posit)

#
sero.posit.SUMMARY<-cbind(sero.posit.SUMMARY,sero.posit)
#head(sero.posit.SUMMARY) # ALL the 515 antigens
#

######

filter.eset.SUB<-filter.eset.SIN               # LOGICAL statement result

##---- SEROPREVALENCE function WITHOUT ORDER
sero.posit<-as.matrix(apply(filter.eset.SUB, 1, sero.percent))

# order [DONE]
sero.posit<-as.data.frame(sero.posit)
colnames(sero.posit) <- c("SIN.sero.prevalence") 

sero_mean<-as.matrix(apply(exprs(eset)[rownames(sero.posit.SUMMARY),eset.SIN.samples], 1, mean, na.rm = TRUE))
colnames(sero_mean) <- c("SIN.mean.intensity")

sero_sum<-as.matrix(apply(filter.eset.SUB, 1, sum, na.rm = TRUE))
colnames(sero_sum) <- c("SIN.rve.samples")

sero.posit<-cbind(sero_sum,sero.posit,sero_mean)
#head(sero.posit)

#
sero.posit.SUMMARY<-cbind(sero.posit.SUMMARY,sero.posit) #FIX TABLES BELOW!
#kable(head(sero.posit.SUMMARY[,1:6]), align = "c") # ALL the 515 antigens-----------------> HERE is the COMPLETE ANSWER
#kable(head(sero.posit.SUMMARY[,c(1:3,7:9)]), align = "c") # ALL the 515 antigens-----------------> HERE is the COMPLETE ANSWER
kable(head(sero.posit.SUMMARY[,4:6]), align = "c") # ALL the 515 antigens-----------------> HERE is the COMPLETE ANSWER
kable(head(sero.posit.SUMMARY[,7:9]), align = "c") # ALL the 515 antigens-----------------> HERE is the COMPLETE ANSWER
#
#write.csv(sero.posit.SUMMARY, "SEVERE.sero.posit.SUMMARY.csv")

######
```
**Frequency of Response per Group (per feature)**

*No se rechaza la $H_{o}$ de igualdad de medias. No se rechaza la $H_{o}$ de igualdad de varianzas*
```{r, fig.height=4, fig.width=16, message=FALSE}
# (1.5) GRAPHIC: Venn-diagram --> [MANUALLY done]
#
# (1.6) TEST DIFFERENCES
#
#kable(summary(sero.posit.SUMMARY))
#kable(summary(sero.posit.SUMMARY[sero.posit.over10.LIST,]))
#
#boxplot(sero.posit.SUMMARY[,c("D.rve.samples","E.rve.samples")], main= "Seropositive Antigens", ylab= "number of reactive samples per antigen")
#stripchart(sero.posit.SUMMARY[,c("D.rve.samples","E.rve.samples")], pch = 1, vertical = TRUE, add = TRUE, method = "jitter")
#
#boxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.rve.samples","E.rve.samples")], main= "Reactive Antigens", ylab= "number of reactive samples per antigen")
#stripchart(sero.posit.SUMMARY[sero.posit.over10.LIST,c("D.rve.samples","E.rve.samples")], pch = 1, vertical = TRUE, add = TRUE, method = "jitter")

############################## 
#
#png("SEVERE_y_summary_seroprevalence_MeanInt.png",    # create PNG for the heat map        
#    width = 3000,
#    height = 1500,        # DIMENSIONS of final image: 1500x1500 pixels
#    res = 300,            # 300 pixels per inch (RESOLUTION)
#    pointsize = 8)        # smaller font size
#
par(mfrow=c(1,4))
#
boxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.sero.prevalence","SIN.sero.prevalence")], 
        main= "Seroprevalence of Reactive Antigens", ylab= "percentage of reactive samples per antigen")
stripchart(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.sero.prevalence","SIN.sero.prevalence")],
           pch =1 , vertical = TRUE, add = TRUE, method = "jitter", col = c(2,4))
beeswarm(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.sero.prevalence","SIN.sero.prevalence")], col = c(2,4))
bxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.sero.prevalence","SIN.sero.prevalence")], add = TRUE)
vioplot(sero.posit.SUMMARY[sero.posit.over10.LIST,]$CON.sero.prevalence, sero.posit.SUMMARY[sero.posit.over10.LIST,]$SIN.sero.prevalence, col = "grey80", names = c("CON","SIN"))
beanplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.sero.prevalence","SIN.sero.prevalence")], log = "", overallline ="median", maxstripline = 0.16)

# TEST for differences
wtSevFREQ1 <- wilcox.test(sero.posit.SUMMARY[sero.posit.over10.LIST,"CON.sero.prevalence"],sero.posit.SUMMARY[sero.posit.over10.LIST,"SIN.sero.prevalence"],exact = FALSE, correct = FALSE)
varSevFREQ1 <- var.test(sero.posit.SUMMARY[sero.posit.over10.LIST,"CON.sero.prevalence"],sero.posit.SUMMARY[sero.posit.over10.LIST,"SIN.sero.prevalence"])

kable(tidy(wtSevFREQ1), align = "c")
kable(glance(varSevFREQ1), align = "c")
############################## 
```
**Magnitud of Response per Group (per feature)**

*No se rechaza la $H_{o}$ de igualdad de medias. No se rechaza la $H_{o}$ de igualdad de varianzas*
```{r, fig.height=4, fig.width=16, message=FALSE}
#
par(mfrow=c(1,4))
#
boxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.mean.intensity","SIN.mean.intensity")], 
        main= "Mean Intensity of Reactive Antigens", ylab= "mean intensity of reactive antigens")
stripchart(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.mean.intensity","SIN.mean.intensity")], 
           pch = 1, vertical = TRUE, add = TRUE, method = "jitter", col = c(2,4))
beeswarm(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.mean.intensity","SIN.mean.intensity")], col = c(2,4))
bxplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.mean.intensity","SIN.mean.intensity")], add = TRUE)
vioplot(sero.posit.SUMMARY[sero.posit.over10.LIST,]$CON.mean.intensity,sero.posit.SUMMARY[sero.posit.over10.LIST,]$SIN.mean.intensity, col = "grey80", names = c("CON","SIN"))
beanplot(sero.posit.SUMMARY[sero.posit.over10.LIST,c("CON.mean.intensity","SIN.mean.intensity")], log = "", overallline ="median")
#
#dev.off()
#
# TEST for differences
wtSevMAG1 <- wilcox.test(sero.posit.SUMMARY[sero.posit.over10.LIST,"CON.mean.intensity"],sero.posit.SUMMARY[sero.posit.over10.LIST,"SIN.mean.intensity"],exact = FALSE, correct = FALSE)
varSevMAG1 <- var.test(sero.posit.SUMMARY[sero.posit.over10.LIST,"CON.mean.intensity"],sero.posit.SUMMARY[sero.posit.over10.LIST,"SIN.mean.intensity"])

kable(tidy(wtSevMAG1), align = "c")
kable(glance(varSevMAG1), align = "c")
############################## 
#
# [PENDING TO KEEP TRYING]
# THIS IS NOT WORKING: http://stackoverflow.com/questions/14975853/how-can-i-create-violin-plot-in-different-colours
# ggplot2 BETTER: http://www.sthda.com/english/wiki/ggplot2-violin-plot-quick-start-guide-r-software-and-data-visualization
#
```


**Breadth (percentage) of Response per Group (per sample)**

*No se rechaza la $H_{o}$ de igualdad de medias. Sí se rechaza la $H_{o}$ de igualdad de varianzas*
```{r, fig.height=4, fig.width=16, message=FALSE}
# GRAPHIC: Bee-swarm scatter plot
#summary(subset(pData(esetEXTRA),Group=="D", breadth.score:mean.intensity))
#summary(subset(pData(esetEXTRA),Group=="E", breadth.score:mean.intensity))
#
#png("SEVERE_x_summary_breadth_intensity.png",    # create PNG for the heat map        
#    width = 3000,
#    height = 1500,        # DIMENSIONS of final image: 1500x1500 pixels
#    res = 300,            # 300 pixels per inch (RESOLUTION)
#    pointsize = 8)        # smaller font size
#
par(mfrow=c(1,4))
#
boxplot(breadth.percentage ~ con_exp, data= pData(eset), main= "Breadth of Response", ylab= "Percentage of reactive antigens per sample")
stripchart(breadth.percentage ~ con_exp, data= pData(eset), pch = 1, vertical = TRUE, add = TRUE, method = "jitter", col = c(2,4))
boxplot(breadth.percentage ~ con_exp, data= pData(eset), main= "Breadth of Response", ylab= "Percentage of reactive antigens per sample")
beeswarm(breadth.percentage ~ con_exp, data= pData(eset), col = c(2,4), add=TRUE) #, pwcol = 
vioplot(subset(pData(eset),con_exp=="CON")$breadth.percentage,subset(pData(eset),con_exp=="SIN")$breadth.percentage, col = "grey80", names = c("CON","SIN"))
beanplot(breadth.percentage ~ con_exp, data= pData(eset), overallline ="mean")
# TEST for differences
wtSev1 <- wilcox.test(subset(pData(eset),con_exp=="CON")$breadth.percentage,subset(pData(eset),con_exp=="SIN")$breadth.percentage,exact = FALSE, correct = FALSE)
varSev1 <- var.test(subset(pData(eset),con_exp=="CON")$breadth.percentage,subset(pData(eset),con_exp=="SIN")$breadth.percentage)

kable(tidy(wtSev1), align = "c")
kable(glance(varSev1), align = "c")
############################## 
```
**Breadth (intensity) of Response per Group (per sample)**

*No se rechaza la $H_{o}$ de igualdad de medias. Sí se rechaza la $H_{o}$ de igualdad de varianzas*
```{r, fig.height=4, fig.width=16, message=FALSE}
#
par(mfrow=c(1,4))
#
boxplot(mean.intensity ~ con_exp, data= pData(eset), main= "Intensity of Response", ylab= "Mean Intensity of reactive antigens per sample")
stripchart(mean.intensity ~ con_exp, data= pData(eset), pch = 1, vertical = TRUE, add = TRUE, method = "jitter", col = c(2,4))
boxplot(mean.intensity ~ con_exp, data= pData(eset), main= "Intensity of Response", ylab= "Mean Intensity of reactive antigens per sample")
beeswarm(mean.intensity ~ con_exp, data= pData(eset), col = c(2,4), add = TRUE)
vioplot(subset(pData(eset),con_exp=="CON")$mean.intensity,subset(pData(eset),con_exp=="SIN")$mean.intensity, col = "grey80", names = c("CON","SIN"))
beanplot(mean.intensity ~ con_exp, data= pData(eset), overallline ="mean")
#
# TEST for differences
wtSev2 <- wilcox.test(subset(pData(eset),con_exp=="CON")$mean.intensity,subset(pData(eset),con_exp=="SIN")$mean.intensity,exact = FALSE, correct = FALSE)
varSev2 <- var.test(subset(pData(eset),con_exp=="CON")$mean.intensity,subset(pData(eset),con_exp=="SIN")$mean.intensity)
#
#dev.off()
#
kable(tidy(wtSev2), align = "c")
kable(glance(varSev2), align = "c")
############################## 
```


```{r, echo=FALSE, eval=FALSE}
################################## FOR EACH GROUP

# [ADDED 28jul2016]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
eset$Group
# [end 28jul2016]

########### GROUP "E"
m_data <- exprs(eset[,pData(eset)$Group=="E"])

m_data<-as.data.frame(m_data)

row.mean.m_data<-as.matrix(apply(m_data, 1, mean))
colnames(row.mean.m_data) <- c("row.mean.m")                 # assign row names 
Em<-row.mean.m_data

m_data_row.mean<-cbind(m_data,row.mean.m_data)
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean<-m_data_row.mean[order(-m_data_row.mean$row.mean.m),]

##### TOP10 OF HIGHER REACTIVITY in GROUP "E
head(rownames(m_data_col.row.mean), 10)
E<-head(rownames(m_data_col.row.mean), 10)
E2<-rownames(m_data_col.row.mean)





########### GROUP "D"
m_data <- exprs(eset[,pData(eset)$Group=="D"])

m_data<-as.data.frame(m_data)

row.mean.m_data<-as.matrix(apply(m_data, 1, mean))
colnames(row.mean.m_data) <- c("row.mean.m")                 # assign row names 
Dm<-row.mean.m_data

m_data_row.mean<-cbind(m_data,row.mean.m_data)
m_data_row.mean<-as.matrix(m_data_row.mean)

m_data_row.mean<-as.data.frame(m_data_row.mean)
m_data_row.mean<-m_data_row.mean[order(-m_data_row.mean$row.mean.m),]

##### TOP10 OF HIGHER REACTIVITY in GROUP "D
head(rownames(m_data_col.row.mean), 10)
D<-head(rownames(m_data_col.row.mean), 10)
D2<-rownames(m_data_col.row.mean)



##### TOP10 comparison of BOTH GROUPS
##### compare TOP10 of BOTH GROUPS: sum of equality in position
sum(E==D)
sum(E2==D2)

##### compare TOP10 of BOTH GROUPS: sum of equality in mean(reactivity)
sum(Em==Dm, na.rm = TRUE)

##### compare TOP10 of BOTH GROUPS: sum of higher # of reactive antigens between groups
# THIS SHOULD BE CONFIRMED BY (i) breadth and (ii) volcano plots
sum(Em>=Dm, na.rm = TRUE) # NON.SEVERE higher mean than SEVERE = 90
sum(Dm>=Em, na.rm = TRUE) # SEVERE higher mean than NON.SEVERE = 164
#
```


### Moderated eBayes t-test
```{r}
############################################################################ SEVERE
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER      ## Samples vs Controls
#summary(pData(eset))

eset$Study <- factor(eset$Study)
table(eset$Study)

# [MODIFIED 16sep2016]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#
levels(eset$con_exp) <- c("CON","SIN") # CHANGE: D: CON exposicion, E: SIN exposicion
#summary(eset$con_exp)


eset$con_exp <- factor(eset$con_exp)
#table(eset$con_exp)

#varMetadata(eset) # D and E still unknown!!!!!!

#

library(limma)
######################
eset$con_exp <- factor(eset$con_exp)
# > eset$Sample.Type
# Levels: ALL NoL

#table(eset$con_exp)

# RELEVEL --> the first level is the REFERENCE level
eset$con_exp <- relevel(eset$con_exp, ref = "SIN") # CONTROL: E (NOT-SEVERE) -> CASE: D (severe)
# > eset$Sample.Type
# Levels: NoL ALL
table(eset$con_exp)
#

# perform the standart limma model
design <- model.matrix(~eset$con_exp)    # design the experiment
fit <- lmFit(eset, design)                    # fit the model
fit <- eBayes(fit)                               # empirical Bayes for moderate t-statistics
#topTable(fit)                                    # evaluate
#######################

# [DEFAULT sorting] "B" for the lods or B-statistic
#topTable(fit, coef = 2, genelist = fit$genes$Description)


```

**Differentially reactive features**
```{r}
#-----------standard-----------
# DIFFERENTIALLY rve [sorted by p.value] (SAME as default)
#topTable(fit, coef = 2, sort.by ="P", genelist = fit$genes$ID, number = 15)
ed <- topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description, number = 15)
ed <- format(ed, digits=3)
#htmlTable(ed, col.rgroup = c("none", "#F7F7F7"), col.columns = c("none", "#F7F7F7"))
kable(ed, align = "c")
#topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$ID, number = 20)
#
ea <- topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description, number = Inf)
ea <- format(ea, digits=3)
```
**Highly reactive features**
```{r, eval=TRUE}
# HIGHLY rve [sorted by average expression level (over all arrays) in descending order]
#topTable(fit, coef = 2, sort.by ="AveExpr", genelist = fit$genes$Description)
t <- topTable(fit, coef = 2, sort.by ="AveExpr", genelist = fit$genes$Description)
t <- format(t, digits=3)
#htmlTable(t, col.rgroup = c("none", "#F7F7F7"), col.columns = c("none", "#F7F7F7"))
kable(t, align = "c")
#------------------------------
```

```{r, eval=FALSE}
#-----------explore------------
# DIFFERENTIALLY rve ONLY [sorted by FOC in descending order, without considering p.value]
topTable(fit, coef = 2, sort.by ="logFC", genelist = fit$genes$Description)
# FIRST sort by default (p_value) and THEN by average
topTable(fit, coef = 2, resort.by ="AveExpr", genelist = fit$genes$Description)
# FIRST sort by default (p_value) and THEN by the (absolute) coefficient representing the log-fold-change
topTable(fit, coef = 2,resort.by ="logFC", genelist = fit$genes$Description)
# to sort by the AVERAGE EXPRESSION LEVEL, then resort by LOG-FOLD-CHANGE
topTable(fit, coef = 2, sort.by ="AveExpr", resort.by ="logFC", genelist = fit$genes$Description)
#------------------------------
```

**p_value distribution [required evaluation prior to FDR]**
```{r, eval=FALSE, fig.align="center", fig.height=2, fig.width=4}
# [31ago2016] before any FDR in MULTIPLE COMPARISONS
# suggested by: http://varianceexplained.org/statistics/interpreting-pvalue-histogram/
test <- topTable(fit, coef = 2,
                 sort.by ="P", 
                 #sort.by ="AveExpr", 
                 genelist = fit$genes$Gene.ID, number = Inf)
library(ggplot2)
ax <- qplot(test$P.Value, binwidth=.05)
bx <- qplot(test$adj.P.Val, binwidth=.05)
multiplot(ax,bx,cols = 2)
```

```{r, eval=FALSE, fig.align="center"}
test2 <- -log10(as.matrix(test[,5]))
rownames(test2) <- as.matrix(test[,1])
colnames(test2) <- c("p.values")
test3 <- t(test2)
barplot(test3, las=2, cex.names = .5)

test4 <- test3[,test3>-log10(0.05)]
barplot(test4, las=2, cex.names = .8)
#barplot(test[,"adj.P.Val"], las=2)
#

# P_adj FILTERING
selected  <- p.adjust(fit$p.value[, 2]) <0.05
esetSel <- eset [selected, ]
#
featureNames(esetSel) 
#
```

### Summary plots

```{r, eval=FALSE, echo=FALSE}
#### NEW FEATURES: FROM help(lmFit) [START]

# Ordinary fit
#fit <- lmFit(y,design)
#fit <- eBayes(fit)
#topTable(fit,coef=2)
dim(fit)
colnames(fit)
rownames(fit)[1:10]
names(fit)

# Fold-change thresholding
fit2 <- treat(fit,lfc=0.1)
topTreat(fit2,coef=2)

# Volcano plot
volcanoplot(fit,coef=2,highlight=10,
            names=fit$genes$Description,
            pch = 19)

# Mean-difference plot
plotMD(fit,column=2)

# Q-Q plot of moderated t-statistics
qqt(fit$t[,2],df=fit$df.residual+fit$df.prior)
abline(0,1)


############### 

```

```{r, eval=FALSE, echo=FALSE}
# (1) Volcano plot: Log Fold Change x Log Odds (B)
volcanoplot(fit,coef=2,highlight=10,
            names=fit$genes$Description,
            pch = 19)

```


**Volcano plot I**
```{r, eval=TRUE, fig.align="center"}

# (2) Volcano plot: Log Fold Change x -log10(p.value)
#source: http://genomicsclass.github.io/book/pages/using_limma.html
h<-topTable(fit, coef = 2, sort.by =NULL, resort.by =NULL, genelist = NULL, number = Inf, confint=0.95)
h<-h[rownames(coef(fit)),]
limmares <- data.frame(Gene.ID=fit$genes$ID,logFC=coef(fit)[,2],AveExpr=h$AveExpr, p.value=fit$p.value[,2], padj=h$adj.P.Val, B=h$B)

#
par(mfrow=c(1,2))
with(limmares, plot(logFC, -log10(p.value),cex=.7, pch=20,
                    main = "Volcano plot",
                    #col=cols,
                    xlim=c(-1.5,1.5), ylim=c(0,12),
                    xlab="log Fold Change"))
abline(h=c(-log10(0.05),-log10(0.001)),
       v=c(-1,1),
       lty=2, col= "grey55")

# Add colored points: red if padj<0.05, orange of log2FC>1, green if both)
###source: http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html

#with(subset(limmares, p.value<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
#with(subset(limmares, padj<.05 ), points(logFC, -log10(p.value), pch=19, col="red"))
#with(subset(limmares, abs(AveExpr)>1.738), points(logFC, -log10(p.value), pch=20, col="green"))
#with(subset(limmares, abs(AveExpr)>2), points(logFC, -log10(p.value), pch=20, col="orange"))
#with(subset(limmares, B>0 ), points(logFC, -log10(p.value), pch=20, col="yellow"))

#with(subset(limmares, p.value<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
with(subset(limmares, padj<.05 ), points(logFC, -log10(p.value), pch=20, col="blue"))
with(subset(limmares, abs(AveExpr)>1.738), points(logFC, -log10(p.value), pch=20, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, -log10(p.value), pch=20, col="orange"))
with(subset(limmares, B>0 ), points(logFC, -log10(p.value), pch=19, col="red"))



# (2.2) Volcano plot: Log Fold Change x AveExpr
#source: http://genomicsclass.github.io/book/pages/using_limma.html

#
with(limmares, plot(logFC, AveExpr,cex=.7, pch=20,
                    main = "logFC x AveExpr plot",
                    #col="grey40",
                    xlim=c(-1.5,1.5), #ylim=c(-1,2.5),
                    xlab="log Fold Change"))
abline(h=c(0,2,-2),
       v=c(-1,0,1),
       lty=2, col= "grey55")

# Add colored points: red if padj<0.05, orange of log2FC>1, green if both)
###source: http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html

#with(subset(limmares, abs(AveExpr)>1.738), points(logFC, AveExpr, pch=19, col="green"))
#with(subset(limmares, abs(AveExpr)>2), points(logFC, AveExpr, pch=19, col="orange"))
#with(subset(limmares, p.value<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
#with(subset(limmares, padj<.05 ), points(logFC, AveExpr, pch=19, col="red"))
#with(subset(limmares, B>0 ), points(logFC, AveExpr, pch=20, col="yellow"))

with(subset(limmares, abs(AveExpr)>1.738), points(logFC, AveExpr, pch=19, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, AveExpr, pch=19, col="orange"))
#with(subset(limmares, p.value<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
with(subset(limmares, padj<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
with(subset(limmares, B>0 ), points(logFC, AveExpr, pch=19, col="red"))

```

```{r, eval=FALSE, echo=FALSE}
# (2.3) SCATTER plot: -log10(p.value) x AveExpr
#source: http://genomicsclass.github.io/book/pages/using_limma.html

#
with(limmares, plot(AveExpr, -log10(p.value),cex=.7, pch=20,
                    main = "AveExpr x p.value plot",
                    #col=cols,
                    #xlim=c(-1,1), #ylim=c(-1,2.5),
                    xlab="mean reactivity"))

# Add colored points: red if padj<0.05, orange of log2FC>1, green if both)
###source: http://www.gettinggeneticsdone.com/2014/05/r-volcano-plots-to-visualize-rnaseq-microarray.html
with(subset(limmares, p.value<.05 ), points(AveExpr, -log10(p.value), pch=19, col="blue"))
#with(subset(limmares, padj<.05 ), points(logFC, -log10(p.value), pch=20, col="red"))
#with(subset(limmares, abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="orange"))
#with(subset(limmares, p.value<.05 & abs(logFC)>1), points(logFC, -log10(p.value), pch=20, col="green"))

with(subset(limmares, abs(AveExpr)>1.738), points(AveExpr, -log10(p.value), pch=19, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(AveExpr, -log10(p.value), pch=19, col="orange"))
with(subset(limmares, p.value<.0224), points(AveExpr, -log10(p.value), pch=19, col="red"))


```

**Volcano plot II**

Using `dplyr`[@dplyr], `ggplot2`[@ggplot2], `ggrepel`[@ggrepel] and `Rmisc`[@Rmisc] packages
```{r, eval=TRUE, message=FALSE, warning=FALSE, fig.align="center", fig.height=8, fig.width=15}

# (3) Volcano plot: Log Fold Change x -log10(p.value) [W/ ANNOTATION]
###source: http://www.gettinggeneticsdone.com/2016/01/repel-overlapping-text-labels-in-ggplot2.html
#library(dplyr)
#library(ggplot2)

results <- mutate(limmares, sig=ifelse(limmares$padj<.004, "diff rve","below Top10"))

p <- ggplot(results, aes(logFC, -log10(p.value))) + 
  geom_point(aes(col=sig)) +
  scale_color_manual(values=c("black","red"))

#p
#p+geom_text(data=filter(results, p.value<.0224), aes(label=Gene.ID))


# Install ggrepel package if needed
# install.packages("devtools")
# devtools::install_github("slowkow/ggrepel")
#library(ggrepel)

#p+geom_text_repel(data=filter(results, p.value<.0224), aes(label=Gene.ID))+xlim(-1,1)
p_rep<-p+geom_text_repel(data=filter(results, padj<.004), aes(label=Gene.ID))#+xlim=c(-1.5,1.5)


# (3.2) Volcano plot: Log Fold Change x AveExprs [W/ ANNOTATION]
###source: http://www.gettinggeneticsdone.com/2016/01/repel-overlapping-text-labels-in-ggplot2.html
#library(dplyr)
#library(ggplot2)

resultsq <- mutate(limmares, sigq=ifelse(limmares$AveExpr>1.738, "highly rve","below Top10"))

q <- ggplot(resultsq, aes(logFC, AveExpr)) + 
  geom_point(aes(col=sigq)) +
  scale_color_manual(values=c("black","green"))

#q
#q+geom_text(data=filter(resultsq, AveExpr>1.738), aes(label=Gene.ID))



# Install ggrepel package if needed
# install.packages("devtools")
# devtools::install_github("slowkow/ggrepel")
#library(ggrepel)

#q+geom_text_repel(data=filter(resultsq, AveExpr>1.738), aes(label=Gene.ID))+xlim(-1,1)
q_rep<-q+geom_text_repel(data=filter(resultsq, AveExpr>1.738), aes(label=Gene.ID))#+xlim=c(-1.5,1.5)

#library(Rmisc)   #multiploting ggplots


# RECORDING images
multiplot(p_rep,q_rep,cols=2)


```

```{r, eval=FALSE, echo=FALSE}

#library(Rmisc)   #multiploting ggplots
# RECORDING images
png("VOLCANO_2_SEV_vivax_25jul2016.png", width = 8000, height = 3000, res = 300, pointsize = 16)
multiplot(p_rep,q_rep,cols=2)
dev.off()


#RECORDING IMAGES
png("VOLCANO_1_SEV_vivax_25jul2016.png", width = 2000, height = 1500, res = 300, pointsize = 8)
par(mfrow=c(1,2))
with(limmares, plot(logFC, -log10(p.value),cex=.7, pch=20,
                    main = "Volcano plot",
                    #col=cols,
                    xlim=c(-1,1), #ylim=c(0,5),
                    xlab="log Fold Change"))
abline(h=c(-log10(0.05),-log10(0.001)),
       v=c(-1,1),
       lty=2)

with(subset(limmares, p.value<.05 ), points(logFC, -log10(p.value), pch=19, col="blue"))
with(subset(limmares, abs(AveExpr)>1.738), points(logFC, -log10(p.value), pch=20, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, -log10(p.value), pch=20, col="orange"))
with(subset(limmares, p.value<.0224), points(logFC, -log10(p.value), pch=19, col="red"))

# (2.2) Volcano plot: Log Fold Change x AveExpr
#source: http://genomicsclass.github.io/book/pages/using_limma.html

#
with(limmares, plot(logFC, AveExpr,cex=.7, pch=20,
                    main = "logFC x AveExpr plot",
                    #col=cols,
                    xlim=c(-1,1), #ylim=c(-1,2.5),
                    xlab="log Fold Change"))
abline(h=c(0,2,-2),
       v=c(-1,0,1),
       lty=2)

with(subset(limmares, p.value<.05 ), points(logFC, AveExpr, pch=20, col="blue"))
with(subset(limmares, abs(AveExpr)>1.738), points(logFC, AveExpr, pch=19, col="green"))
with(subset(limmares, abs(AveExpr)>2), points(logFC, AveExpr, pch=19, col="orange"))
with(subset(limmares, p.value<.0224), points(logFC, AveExpr, pch=20, col="red"))

dev.off()

# (ADi) Volcano plot: MEAN difference x -log10(p.value)
# xlab="mean(D=severe) - mean(E=control)"


#### NEW FEATURES: FROM help(lmFit) [END]

```

**Differentially reactive features**
*using `NMF` package*[@Gaujoux2010NMF]
```{r, eval=TRUE, fig.height=5, fig.width=10, fig.align="center"}
#
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]
#exprs(eset)[rownames(subset(limmares, B>0)),]

eset <- eset[feature_sort_mean,sample_sort_mean] ## REORDER samples and features inside the ExpressionSet

#write.csv(subset(limmares[order(limmares$B, decreasing = TRUE),], B>0), file = "EXPOSURE_DiffRve.csv")

#library(NMF)
aheatmap(exprs(eset)[rownames(subset(limmares[order(limmares$B, decreasing = TRUE),], B>0)),], Rowv = NA, Colv = NA, annCol = pData(eset), layout = "_")
```


**Magnitud and Frequency per feature**

### Correlations
**Correlation between Magnitud and Exposure**

Resumen:

- De 15 antígenos diferencialmete reactivos, 12 están diferencialmente reactivos en pacientes con exposiciones previas a malaria y 3 en pacientes sin previa exposición (*naive*).

En relación a la exposición previa a malaria:

- 10 de los 12 antígenos tienen un incremento en la magnitud con respecto al número de exposiciones pasadas (regresión lineal con pendiente positiva).
- PVX_113775_1o1.455 (*membrane protein pf12 precursor, putative* o *6-cysteine protein **(P12)***) tiene la mayor pendiente positiva.
- PVX_003775_202.150 y PVX_003775_202.532 son réplicas del mismo antígeno, **MSP4**. Poseen una alta correlación en su reactividad **[requiere stat]** y ambos están diferencialmente reactivos en pacientes con exposiciones previas a malaria.
- **MSP1** posee dos antígenos con diferentes patrones de respuesta: **(i)** PVX_099980_1o1_S1.793 solo es diferencialmente reactivo en pacientes con exposición previa y **(ii)** PVX_099980_1o1_S2.504 solo es el segundo antígeno con mayor reactividad, independiente a la exposición de los pacientes.

En relación a la sintomatología:

- En pacientes con sintomatología severa, 9 de los 12 antígenos tienen un mayor incremento en la magnitud con respecto al número de exposiciones pasadas, en comparación con los pacientes no severos (malaria no complicada). **[requiere stat]**
- Entre los tres restantes están PVX_092995_2o2.259 (*tryptophan/threonine-rich antigen* o *tryptophan-rich antigen* **(Pv-fam-a)**) y PVX_092070_1o1.176 (*hypothetical protein, conserved* o *parasitophorous vacuolar protein 1, putative* **(PV1)**). Este último es el antígeno con menor *p value* (1.30e-12, 3.33e-10 BH-corregido).
- Caso contrario, PVX_118705_1o1.263 (*liver stage antigen 3* o **Pv-fam-d protein**) decae con respecto al número de exposiciones. **[requiere stat]**
- Todos los antígenos diferencialmente reactivos en pacientes con exposiciones previas a malaria tienen una magnitud independiente a la edad de los pacientes.

En relación a la edad:

- PVX_085590_101_S5.814 es **(i)** el antígeno de mayor magnitud en el grupo de los 3 antígenos diferencialmente reactivos en pacientes sin previa exposición (*naive*), y **(ii)** el antígeno con mayor decadencia con respecto a la edad de los pacientes.
- Los 2 antígenos restantes tienen una magnitud independiente con respecto a la edad.


```{r}
messz <- exprs(eset)[rownames(subset(limmares, B>0 )),]
dim(t(messz))
doom <- pData(eset)#[,c(9:ncol(pData(eset)))]
dim(doom)

corrEXP <- cbind(t(messz), doom)
#colnames(corrSEV)
corrEXP$ID <- rownames(corrEXP)
#str(corrEXP)
```


```{r, fig.height=6, fig.width=8, fig.align='center', warning=FALSE, message=FALSE}
#library(tidyr)
#library(dplyr)

#
##### CAMBIAR A BOXPLOT POR CATEGORIAS
#


#12
corrEXPcon1 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(malaria_pasado_num, reactivity)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature)# + geom_line()
corrEXPcon1

corrEXPcon1 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(malaria_pasado_num, reactivity)) + geom_point() + geom_smooth() + facet_wrap(~ feature)# + geom_line()
corrEXPcon1

corrEXPcon1 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(expo_CAT, reactivity)) + geom_boxplot() + facet_wrap(~ feature)# + geom_line()
corrEXPcon1

```

```{r, fig.height=6, fig.width=9, fig.align='center', warning=FALSE, message=FALSE}
#
##### CAMBIAR A BOXPLOT POR CATEGORIAS
#
corrEXPcon1x <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(malaria_pasado_num, reactivity, group=casecontrol, colour=casecontrol)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature)# + geom_line()
corrEXPcon1x

corrEXPcon1x <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(malaria_pasado_num, reactivity, group=casecontrol, colour=casecontrol)) + geom_point() + geom_smooth() + facet_wrap(~ feature)# + geom_line()
corrEXPcon1x

corrEXPcon1x <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(expo_CAT, reactivity, fill=casecontrol)) + geom_boxplot() + facet_wrap(~ feature)# + geom_line()
corrEXPcon1x

corrEXPcon1xx <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(edad, reactivity, group=casecontrol, colour=casecontrol)) + geom_point() + geom_smooth() + facet_wrap(~ feature)# + geom_line()
corrEXPcon1xx

corrEXPcon1xx <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(edad_CAT, reactivity, fill=casecontrol))+ geom_boxplot() + facet_wrap(~ feature)# + geom_line()
corrEXPcon1xx
```


```{r, fig.height=12, fig.width=8, fig.align='center', warning=FALSE, message=FALSE, eval=FALSE}
corrEXPcon2 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="CON" & feature!="PVX_085025_1o1_S2.1143" & feature!="PVX_099080_2o2.439" & feature!="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(malaria_pasado_num, reactivity)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(casecontrol ~ feature, ncol = 4)# + geom_line()
corrEXPcon2
```

```{r, fig.height=2, fig.width=6, fig.align='center', warning=FALSE, message=FALSE}
#
##### CAMBIAR A BOXPLOT POR CATEGORIAS
#
#3
corrEXPsin1 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="SIN" & feature=="PVX_085025_1o1_S2.1143" | con_exp=="SIN" & feature=="PVX_099080_2o2.439" | con_exp=="SIN" & feature=="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(edad, reactivity)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature)# + geom_line()
corrEXPsin1

corrEXPsin1 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="SIN" & feature=="PVX_085025_1o1_S2.1143" | con_exp=="SIN" & feature=="PVX_099080_2o2.439" | con_exp=="SIN" & feature=="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(edad, reactivity)) + geom_point() + geom_smooth() + facet_wrap(~ feature)# + geom_line()
corrEXPsin1

corrEXPsin1 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="SIN" & feature=="PVX_085025_1o1_S2.1143" | con_exp=="SIN" & feature=="PVX_099080_2o2.439" | con_exp=="SIN" & feature=="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(edad_CAT, reactivity)) + geom_boxplot() + facet_wrap(~ feature)# + geom_line()
corrEXPsin1

```

```{r, fig.height=2, fig.width=7, fig.align='center', warning=FALSE, message=FALSE, eval=FALSE}
#
#3
corrEXPsin1x <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="SIN" & feature=="PVX_085025_1o1_S2.1143" | con_exp=="SIN" & feature=="PVX_099080_2o2.439" | con_exp=="SIN" & feature=="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(edad, reactivity, group=casecontrol, colour=casecontrol)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature)# + geom_line()
corrEXPsin1x

#weird graph -> does not work with geom_smooth
corrEXPsin1x <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(con_exp=="SIN" & feature=="PVX_085025_1o1_S2.1143" | con_exp=="SIN" & feature=="PVX_099080_2o2.439" | con_exp=="SIN" & feature=="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(edad, reactivity, group=casecontrol, colour=casecontrol)) + geom_point()+ facet_wrap(~ feature)# + geom_line()# + geom_smooth() 
corrEXPsin1x
```


```{r, fig.height=2, fig.width=7, fig.align='center', warning=FALSE, message=FALSE, eval=FALSE}
corrEXPsin37 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(casecontrol=="Severe" & feature=="PVX_085025_1o1_S2.1143" | casecontrol=="Severe" & feature=="PVX_099080_2o2.439" | casecontrol=="Severe" &  feature=="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(reactivity,CumSumSevere7, group=con_exp, colour=con_exp)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature, ncol = 3)# + geom_line()
corrEXPsin37

corrEXPsin310 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(casecontrol=="Severe" & feature=="PVX_085025_1o1_S2.1143" | casecontrol=="Severe" & feature=="PVX_099080_2o2.439" | casecontrol=="Severe" &  feature=="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(reactivity,CumSumSevere10, group=con_exp, colour=con_exp)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature, ncol = 3)# + geom_line()
corrEXPsin310

corrEXPsin310 <- corrEXP %>%
  gather(feature, reactivity, PVX_003775_2o2.150:PVX_085025_1o1_S2.1143) %>%
  mutate(feature=factor(feature)) %>%
  filter(casecontrol=="Severe" & feature=="PVX_085025_1o1_S2.1143" | casecontrol=="Severe" & feature=="PVX_099080_2o2.439" | casecontrol=="Severe" &  feature=="PVX_085590_1o1_S5.814") %>%
  mutate(feature=factor(feature)) %>%
  ggplot(aes(reactivity,CumSumSevere10, group=con_exp, colour=con_exp)) + geom_point() + facet_wrap(~ feature, ncol = 3)# + geom_line()# + geom_smooth(method = "lm", se=FALSE)
corrEXPsin310

#str(cleaned$data)
#str(cleaned)
```




### PlasmoDB protein data
```{r, eval=TRUE}
plasmoDBdiff <- read.csv("EXPOSURE_DiffRve_REDUCED.csv")
#colnames(plasmoDBdiff)

#htmlTable(plasmoDBdiff[,1:4], col.rgroup = c("none", "#F7F7F7"))
kable(plasmoDBdiff[,1:4], align = "c") #TMD and SP
kable(plasmoDBdiff[,c(1:2,5:7)], align = "c") #MaxExpTiming / Stage
kable(plasmoDBdiff[,c(2,8:10)], align = "c") #GOterms
kable(plasmoDBdiff[,c(1:2,11:13)], align = "c") #Orthologs and SNPs
```

## II.2 By Exposure then Severity

Null hypothesis of equal mean per antigen among groups is not rejected, neither by severity between exposed (semi-immune) nor unexposed (naive).

However, a 22-22 comparison between exposed and non-severe against unexposed and severe reject the null hypothesis with fewer antigens (9 antigens) than the complete comparison based on exposure (15 antigens).

### Exp/NoSev vs NoExp/Sev
```{r}
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
#head(pData(eset))
eset <- eset[, phenoData(eset)$con_exp=="Sin exposicion" & phenoData(eset)$casecontrol=="Severe" | 
               phenoData(eset)$con_exp=="Con exposicion" & phenoData(eset)$casecontrol!="Severe" ]
eset$Group <- factor(eset$Group)
#
#library(limma)
# RELEVEL --> the first level is the REFERENCE level
eset$Group <- relevel(eset$Group, ref = "E") # CONTROL: E (NOT-SEVERE) -> CASE: D (severe)
table(eset$Group)
#

# perform the standart limma model
design <- model.matrix(~eset$Group)    # design the experiment
fit <- lmFit(eset, design)                    # fit the model
fit <- eBayes(fit)                               # empirical Bayes for moderate t-statistics
```
**Differentially reactive features**
```{r}
#-----------standard-----------
# DIFFERENTIALLY rve [sorted by p.value] (SAME as default)
#topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description)
kable(format(topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description), digits=3), align = "c")
```


### NoExp: NoSev vs Sev
```{r}
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset <- eset[, phenoData(eset)$con_exp=="Sin exposicion"]
eset$Group <- factor(eset$Group)
#
#library(limma)
# RELEVEL --> the first level is the REFERENCE level
eset$Group <- relevel(eset$Group, ref = "E") # CONTROL: E (NOT-SEVERE) -> CASE: D (severe)
table(eset$Group)
#

# perform the standart limma model
design <- model.matrix(~eset$Group)    # design the experiment
fit <- lmFit(eset, design)                    # fit the model
fit <- eBayes(fit)                               # empirical Bayes for moderate t-statistics
```
**Differentially reactive features**
```{r}
#-----------standard-----------
# DIFFERENTIALLY rve [sorted by p.value] (SAME as default)
#topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description)
kable(format(topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description), digits=3), align = "c")
```

### Exp: NoSev vs Sev
```{r}
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset <- eset[, phenoData(eset)$con_exp=="Con exposicion"]
eset$Group <- factor(eset$Group)
#
library(limma)
# RELEVEL --> the first level is the REFERENCE level
eset$Group <- relevel(eset$Group, ref = "E") # CONTROL: E (NOT-SEVERE) -> CASE: D (severe)
table(eset$Group)
#

# perform the standart limma model
design <- model.matrix(~eset$Group)    # design the experiment
fit <- lmFit(eset, design)                    # fit the model
fit <- eBayes(fit)                               # empirical Bayes for moderate t-statistics
```
**Differentially reactive features**
```{r}
#-----------standard-----------
# DIFFERENTIALLY rve [sorted by p.value] (SAME as default)
#topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description)
kable(format(topTable(fit, coef = 2,sort.by ="P", genelist = fit$genes$Description), digits=3), align = "c")
```

## II.3 Cumulative Sum comparison [***]

Test the null hypothesis of equal mean of cumulative Ab intentensity between Severity groups, using the differentially reactive antigens between Exposure groups.

```{r}

```


## III. Highly reactive features

**Top10 highly reactive features**
```{r-high, eval=TRUE}
# HIGHLY rve [sorted by average expression level (over all arrays) in descending order]
#topTable(fit, coef = 2, sort.by ="AveExpr", genelist = fit$genes$ID)
t <- topTable(fit, coef = 2, sort.by ="AveExpr", genelist = fit$genes$Description)
t <- format(t, digits=3)
#htmlTable(t, col.rgroup = c("none", "#F7F7F7"))
kable(t, align = "c")
#------------------------------
```

**Top10 highly reactive features**
*using `NMF` package*[@Gaujoux2010NMF]
```{r, eval=TRUE, fig.height=5, fig.width=10, fig.align="center"}
# [MODIFIED 16sep2016]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#
levels(eset$con_exp) <- c("CON","SIN") # CHANGE: D: CON exposicion, E: SIN exposicion
#summary(eset$con_exp)

#
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]
#exprs(eset)[rownames(subset(limmares, B>0)),]

eset <- eset[feature_sort_mean,sample_sort_mean] ## REORDER samples and features inside the ExpressionSet

#write.csv(subset(limmares[order(limmares$B, decreasing = TRUE),], B>0), file = "EXPOSURE_DiffRve.csv")

#library(NMF)
aheatmap(exprs(eset)[rownames(subset(limmares[order(limmares$AveExpr, decreasing = TRUE),], abs(AveExpr)>1.738)),], 
         Rowv = NA, Colv = NA, annCol = pData(eset), layout = "_")
```

### Correlations
**Correlation between Magnitud and Exposure**
```{r}
messz <- exprs(eset)[rownames(subset(limmares, abs(AveExpr)>1.738)),]
dim(t(messz))
doom <- pData(eset)#[,c(9:ncol(pData(eset)))]
dim(doom)

corrHIGH <- cbind(t(messz), doom)
#colnames(corrSEV)
corrHIGH$ID <- rownames(corrHIGH)
#str(corrHIGH)
```

```{r, fig.height=6, fig.width=8, fig.align='center', warning=FALSE, message=FALSE}
#library(tidyr)
#library(dplyr)

#12
corrHIGHedad <- corrHIGH %>%
  gather(feature, reactivity, PVX_118705_1o1.263:PVX_083560_2o2.1028) %>%
  ggplot(aes(edad, reactivity)) + geom_point() + geom_smooth() + facet_wrap(~ feature, ncol = 4)# + geom_line()#method = "lm", se=FALSE
corrHIGHedad

corrHIGHedad <- corrHIGH %>%
  gather(feature, reactivity, PVX_118705_1o1.263:PVX_083560_2o2.1028) %>%
  ggplot(aes(edad_CAT, reactivity)) + geom_boxplot() + facet_wrap(~ feature, ncol = 4)# + geom_line()#method = "lm", se=FALSE
corrHIGHedad
```

```{r, fig.height=6, fig.width=9, fig.align='center', warning=FALSE, message=FALSE}
#library(tidyr)
#library(dplyr)

#12
corrHIGHedad2 <- corrHIGH %>%
  gather(feature, reactivity, PVX_118705_1o1.263:PVX_083560_2o2.1028) %>%
  ggplot(aes(edad, reactivity, group=con_exp, colour=con_exp)) + geom_point() + geom_smooth(alpha=.2) + facet_wrap( ~ feature, ncol = 4)# + geom_line()#
corrHIGHedad2

corrHIGHedad2 <- corrHIGH %>%
  gather(feature, reactivity, PVX_118705_1o1.263:PVX_083560_2o2.1028) %>%
  ggplot(aes(edad, reactivity, group=casecontrol, colour=casecontrol)) + geom_point() + geom_smooth(alpha=.2) + facet_wrap( ~ feature, ncol = 4)# + geom_line()#
corrHIGHedad2

#

corrHIGHedad3 <- corrHIGH %>%
  gather(feature, reactivity, PVX_118705_1o1.263:PVX_083560_2o2.1028) %>%
  ggplot(aes(edad, reactivity, group=con_exp, colour=con_exp)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature, ncol = 4)# + geom_line()#
corrHIGHedad3

corrHIGHedad3 <- corrHIGH %>%
  gather(feature, reactivity, PVX_118705_1o1.263:PVX_083560_2o2.1028) %>%
  ggplot(aes(edad, reactivity, group=casecontrol, colour=casecontrol)) + geom_point() + geom_smooth(method = "lm", se=FALSE) + facet_wrap(~ feature, ncol = 4)# + geom_line()#
corrHIGHedad3

#

corrHIGHedad3 <- corrHIGH %>%
  gather(feature, reactivity, PVX_118705_1o1.263:PVX_083560_2o2.1028) %>%
  ggplot(aes(edad_CAT, reactivity, fill=con_exp)) + geom_boxplot() + facet_wrap(~ feature, ncol = 4)# + geom_line()#
corrHIGHedad3

corrHIGHedad3 <- corrHIGH %>%
  gather(feature, reactivity, PVX_118705_1o1.263:PVX_083560_2o2.1028) %>%
  ggplot(aes(edad_CAT, reactivity, fill=casecontrol)) + geom_boxplot() + facet_wrap(~ feature, ncol = 4)# + geom_line()#
corrHIGHedad3


```


### PlasmoDB protein data
```{r, eval=TRUE}
plasmoDBhigh <- read.csv("HIGHLY_Rve_REDUCED.csv")
#colnames(plasmoDBdiff)
#htmlTable(plasmoDBhigh[,1:4], col.rgroup = c("none", "#F7F7F7"))
kable(plasmoDBhigh[,1:4], align = "c") #TMD and SP
kable(plasmoDBhigh[,c(1:2,5:7)], align = "c") #MaxExpTiming / Stage
kable(plasmoDBhigh[,c(2,8:10)], align = "c") #GOterms
kable(plasmoDBhigh[,c(1:2,11:13)], align = "c") #Orthologs and SNPs
```

## IV. Shared top features
**between Symptomatology and Exposure**

Looking at the log-fold change between case vs control ($logFC= log(case)-log(control)$),
significantly over-reactive features in previously-exposed patients, have also a reactivity biased towards to non-severe patients. 
In contrast, 
significantly over-reactive features in non-exposed patients, have also a reactivity biased towards to severe patients
```{r}
edtd <- intersect(rownames(ed),rownames(td))
kable(ed[edtd,], align = "c")
kable(td[edtd,], align = "c")
```

**between Symptomatology and Highly** non

```{r, eval=FALSE}
tdh <- intersect(rownames(td),rownames(t))
kable(td[tdh,], align = "c")
```

**between Exposed and Highly** 
 
```{r}
edh <- intersect(rownames(ed),rownames(t))
kable(ed[edh,], align = "c")
```

**On MSP1**

The differentially reactive protein segment (by symptomatology and exposure, respectively) is different from the highly reactive protein segment
```{r}
kable(rbind(subset(td,td$ID=="major blood-stage surface antigen Pv200"),subset(ed,ed$ID=="major blood-stage surface antigen Pv200"),t[2,]), align = "c")
```

**Heatmap of shared features between Symptomatology Exposure and Highly reactive subsets**

*Row clustered y Columns sorted by Exposure*
```{r, eval=TRUE, echo=TRUE, fig.height=4, fig.width=10, fig.align="center"}
########################### SAMPLE ORDER per group

# [START FROM THE BEGINNNING]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
#eset$Group
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#
eset <- eset[feature_sort_mean,sample_sort_mean_FIRST] ## REORDER samples and features inside the ExpressionSet
#eset$Group
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]

sample_sort_EXPOSURE <- rownames(pData(eset)[order(pData(eset)$malaria_pasado_num, decreasing = FALSE),])
eset <- eset[feature_sort_mean,sample_sort_EXPOSURE] ## REORDER samples and features inside the ExpressionSet
#library(NMF)
aheatmap(exprs(eset)[c(edtd, edh),], Rowv = FALSE, Colv = NA, annCol = pData(eset), layout = "_")
```

```{r, eval=FALSE, echo=FALSE, fig.height=5, fig.width=10, fig.align="center"}
########################### SAMPLE ORDER per group

# [START FROM THE BEGINNNING]
eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
eset$Group <- factor(eset$Group)
#eset$Group
#
pData(eset) <- cbind(pData(eset),t(col.mean.m_dataX))#ncol(pData(eset))
#
eset <- eset[feature_sort_mean,sample_sort_mean_FIRST] ## REORDER samples and features inside the ExpressionSet
#eset$Group
#eset<-normal.log2.raw.eSet.VIVAX.SEV.FILTER
pData(eset) <- pData(eset)[,9:ncol(pData(eset))]

#library(NMF)
aheatmap(exprs(eset), Rowv = NA, Colv = NA, annCol = pData(eset))
```


***
***

# Resume

- Severity
    + No differencially reactive peptides according to severity in P.vivax
- Exposure
    + 15 differencially reactive peptides according to exposure in P.vivax

***
***

# Appendix



```{r, eval=FALSE, echo=FALSE}
par(mfrow=c(2,2))
hist(sampleSEV$edad, breaks=9, col="gray")
hist(sampleSEV$edad, breaks=18, col="gray")
#par(mfrow=c(1,2))
hist(sampleSEV$edad, breaks=9, col="gray", freq = FALSE)
lines(density(sampleSEV$edad), lwd=3)
hist(sampleSEV$edad, breaks=18, col="gray", freq = FALSE)
lines(density(sampleSEV$edad), lwd=3)

#b <- qplot(sampleSEV$edad, binwidth=10)
#a <- qplot(sampleSEV$edad, binwidth=5)


### http://stackoverflow.com/questions/14124373/combine-base-and-ggplot-graphics-in-r-figure-window
t <- c(1:(24*14)) 
P <- 24 
A <- 10 
y <- A*sin(2*pi*t/P)+20

library(gridBase)
par(mfrow=c(2, 2))
plot(y,type = "l",xlab = "Time (hours)",ylab = "Amplitude",main = "Time series")
#
require(biwavelet)
require(grid)
t1 <- cbind(t, y)
wt.t1=wt(t1)
#
plot(wt.t1, plot.cb=FALSE, plot.phase=FALSE,main = "Continuous wavelet transform",
     ylab = "Period (hours)",xlab = "Time (hours)")
spectrum(y,method = "ar",main = "Spectral density function", 
         xlab = "Frequency (cycles per hour)",ylab = "Spectrum")
## the last one is the current plot
plot.new()              ## suggested by @Josh
vps <- baseViewports()
pushViewport(vps$figure) ##   I am in the space of the autocorrelation plot
vp1 <-plotViewport(c(1.8,1,0,1)) ## create new vp with margins, you play with this values 
require(ggplot2)
acz <- acf(y, plot=F)
acd <- data.frame(lag=acz$lag, acf=acz$acf)
p <- ggplot(acd, aes(lag, acf)) + geom_area(fill="grey") +
  geom_hline(yintercept=c(0.05, -0.05), linetype="dashed") +
  theme_bw()+labs(title= "Autocorrelation\n")+
  ## some setting in the title to get something near to the other plots
  theme(plot.title = element_text(size = rel(1.4),face ='bold'))
print(p,vp = vp1)        ## suggested by @bpatiste

```

## Hmisc box plots
```{r}
bpplt()
```


```{r, eval=FALSE}
kable(ta, align = "c")
kable(ea, align = "c")
kable(ha, align = "c")
```


# Computer environment

```{r, echo=FALSE}
sessionInfo()
```

# References

